---
title: "Biomineralization Toolkit Analysis"
author: "Kristen Brown, modified ZD"
date: "1/30/2024"
output: html_document
---

# Biomineralization Toolkit Analysis

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo() #provides list of loaded packages and version of R. 
```

First, load the necessary packages.

```{r load_libraries, include = TRUE}
# load libraries - notes show the install command needed to install (pre installed)
library(goseq)
library(dplyr)
library(forcats)
library(ggplot2)
library(gridExtra)
library(tidyr)
library(grDevices)
library(reshape2)
library(Rmisc)
library(ggpubr)
library(tibble)
library(gridExtra)
library(tidyr)
library(zoo)
library(ComplexHeatmap)
library(circlize)
library(GSEABase)
library(data.table)
library(stringr)
library(GenomicRanges)
library(rtracklayer)
library(rrvgo)
```

## Load in data
```{r}
library(rtracklayer)
gff<-rtracklayer::import("../../data/Pocillopora_acuta_HIv2.genes_fixed.gff3")
gff<-as.data.frame(gff)
dim(gff) # 478988     9
names(gff) 

transcripts <- subset(gff, type == "transcript")
transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information
transcript_lengths <- width(transcripts_gr) #isolate length of each gene
seqnames<-transcripts_gr$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame

dim(transcripts) #33730    13
```

```{r}
kegg <- read.delim("../../data/Pocillopora_acuta_HIv2.genes.KEGG_results.txt",header = FALSE)
kegg <- as.data.frame(kegg)
colnames(kegg)[1] <- "gene_id" 
colnames(kegg)[2] <- "KEGG_new"
head(kegg)
```
```{r}
eggnog <- read.delim("../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt")#this file contains all of the go terms, descriptions, kegg, etc
eggnog <- plyr::rename(eggnog, c("X.query"="gene_id"))
head(eggnog,2)
```

```{r}
gogene <- merge(transcripts, eggnog, by=c("gene_id"), all=T)
gogene <- merge(gogene, kegg, by=c("gene_id"), all=T)
head(gogene,2)
```

```{r}
geneInfo <- read.csv("../../output/WGCNA/WGCNA_ModuleMembership.csv") #this file was generated from the WGCNA analyses and contains the modules of interest
geneInfo<- plyr::rename(geneInfo, c("X"="gene_id"))
dim(geneInfo) # there are 9012 genes in our gene info file
```


```{r}
geneInfo <- merge(gogene, geneInfo, by=c("gene_id"), all=T)
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  GOs column
```{r}
geneInfo$GOs <- gsub(",", ";", geneInfo$GOs)
geneInfo$GOs <- gsub('"', "", geneInfo$GOs)
geneInfo$GOs <- gsub("-", NA, geneInfo$GOs)

geneInfo$KEGG_new[geneInfo$KEGG_new == ""] <- NA
```

```{r}
unique(geneInfo$moduleColor)
```

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
```

## Frequency of GO terms

```{r}
cal_biomin_terms <- read.csv("../../output/Biomineralization_goterms.csv") 
head(cal_biomin_terms)
```

```{r}
cal_up_terms <- read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_calcification_filtered.csv")
cal_up_terms <- cal_up_terms %>%
  mutate(Factor = "Up")
head(cal_up_terms)
```

```{r}
cal_down_terms <-read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_calcification_down_filtered.csv")
cal_down_terms<-cal_down_terms %>%
  mutate(Factor = "Down")
head(cal_down_terms)
```

```{r}
colnames(cal_biomin_terms)
colnames(cal_up_terms)
colnames(cal_down_terms)
```
### Merge biomineralization, up and down-regulation of calcification GOterms

```{r}
all_terms<- merge(cal_up_terms,cal_down_terms, by=c("Factor","GOterm","X.1","X","GOterm","over_represented_pvalue","under_represented_pvalue","numDEInCat","numInCat","term","ontology","bh_adjust","ParentTerm"),all=T)


all_terms<- merge(all_terms,cal_biomin_terms, by=c("Factor","GOterm","X.1","X","GOterm","over_represented_pvalue","under_represented_pvalue","numDEInCat","numInCat","term","ontology","bh_adjust","ParentTerm"),all=T)

all_terms$GOterm<-as.factor(all_terms$GOterm)
head(all_terms)
tail(all_terms)
```

```{r}
goterms_shared <- all_terms %>%
  group_by(GOterm) %>%
  dplyr::summarise(
    ParentTerm = paste(unique(ParentTerm), collapse = ", "),
    Factor = paste(unique(Factor), collapse = ", ")
  )
dim(goterms_shared)
```

```{r}
write.csv(goterms_shared, "../../output/WGCNA/GO_analysis/Merged_GOterms_factor_ParentTerm.csv")
```

```{r}
# How is this file made?
cal_freq_terms <-read.csv("../../output/WGCNA/Kristen_Old/goseq_pattern_calcification_all.csv")
head(cal_freq_terms)
```

##Frequency >10 GO terms upregulation
```{r}
cal_freq_terms_filtered_up <- cal_freq_terms %>%
  filter(Number.of.terms>=10) %>%
  filter(Calcification.direction=="Up")
cal_freq_terms_filtered_up
```
### Figure
```{r}
#counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
#counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
freq_fig_up<-ggplot(cal_freq_terms_filtered_up, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,40))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_up

ggsave(filename="../../output/WGCNA/GO_analysis/freq_fig_up_extra_filtered.png", plot=freq_fig_up, dpi=300, height=6, units="in", limitsize=FALSE)
```

## Frequency >10 GO terms downregulation
```{r}
cal_freq_terms_filtered_down <- cal_freq_terms %>%
  filter(Number.of.terms>=10) %>%
  filter(Calcification.direction=="Down")
cal_freq_terms_filtered_down
```

### Figure
```{r}
freq_fig_down<-ggplot(cal_freq_terms_filtered_down, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  #geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,40))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#d1e5f0","#92c5de","#4393c3","#2166ac"), na.value = "grey98",limits = c(0, 100))+ 
  #scale_fill_gradientn(colours=c("#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(10, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_down

ggsave(filename="../../output/WGCNA/GO_analysis/freq_fig_down_extra_filtered.png", plot=freq_fig_down, dpi=300, height=6, units="in", limitsize=FALSE)
```

```{r}
compare_figs<-cowplot::plot_grid(freq_fig_up, freq_fig_down, nrow=2, align="v")
compare_figs
```

##Frequency all GO terms upregulation
```{r}
cal_freq_terms_filtered_up_all <- cal_freq_terms %>%
  filter(Calcification.direction=="Up")
cal_freq_terms_filtered_up_all
```

###Figure
```{r}
#counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
#counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
freq_fig_up<-ggplot(cal_freq_terms_filtered_up_all, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,40))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_up
```
##Frequency all GO terms downregulation
```{r}
cal_freq_terms_filtered_down_all <- cal_freq_terms %>%
  filter(Calcification.direction=="Down")
cal_freq_terms_filtered_down_all
```

###Figure
```{r}
freq_fig_down<-ggplot(cal_freq_terms_filtered_down_all, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  #geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,40))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#d1e5f0","#92c5de","#4393c3","#2166ac"), na.value = "grey98",limits = c(0, 100))+ 
  #scale_fill_gradientn(colours=c("#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(10, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_down
```
```{r}
compare_figs_all<-cowplot::plot_grid(freq_fig_up, freq_fig_down, ncol=2, align="h")
compare_figs_all
```

## Biomineralization toolkit present in modules

```{r}
biomin <-read.csv("../../output/Biomin_blast_Pocillopora_acuta_best_hit.csv")
```

```{r}
wgcnamod <-read.csv("../../output/WGCNA/WGCNA_ModuleMembership.csv")
wgcnamod <- plyr::rename(wgcnamod, c("X"="Pocillopora_acuta_best_hit"))
```

```{r}
biomin_mod <- merge(biomin, wgcnamod, by=c("Pocillopora_acuta_best_hit"), all=F)
head(biomin_mod)
```

```{r}
glmmseq_exp <-read.csv("../../output/glmmseq/model_expression_prediction_allgenes.csv")
glmmseq_exp <- plyr::rename(glmmseq_exp, c("X"="Pocillopora_acuta_best_hit"))
head(glmmseq_exp)
```

```{r}
biomin_exp <- merge(biomin_mod, glmmseq_exp, by=c("Pocillopora_acuta_best_hit"), all=T)
head(biomin_exp)
```

```{r}
biomin_all <- read.csv("../../output/Biomin_blast_Pocillopora_acuta_best_hit_glmmSeq.csv")
head(biomin_all)
```

```{r}
colnames(biomin_all)
```

```{r}
library(tidyr)
biomin_all_filtered_long <- pivot_longer(biomin_all, cols=30:75, names_to = "Colony", values_to = "Expression")
biomin_all_filtered_long $Colony <- as.factor(biomin_all_filtered_long $Colony)
head(biomin_all_filtered_long)
```

```{r}
biomin_all_filtered_long  <- biomin_all_filtered_long  %>% 
  separate(Colony, into = c('Origin', 'Colony.number'), sep = 2)
head(biomin_all_filtered_long)
```

```{r}
library(stringr)
biomin_all_filtered_long $Colony <- as.numeric(str_extract(biomin_all_filtered_long $Colony.number, "[0-9]+"))
biomin_all_filtered_long <-biomin_all_filtered_long  %>% 
   mutate(Treatment = trimws(str_remove(biomin_all_filtered_long $Colony.number, "(\\s+[A-Za-z]+)?[0-9-]+")))
head(biomin_all_filtered_long)
```

```{r}
biomin_all_filtered_long $Origin <- as.factor(biomin_all_filtered_long $Origin)
biomin_all_filtered_long $Treatment <- as.factor(biomin_all_filtered_long $Treatment)
head(biomin_all_filtered_long)
```

```{r}
biomin_all_filtered_long  <- biomin_all_filtered_long  %>%
  mutate(Treatment2 = ifelse(Treatment == "A" | Treatment == "B", "Variable",
               ifelse(Treatment == "C" | Treatment == "D", "Stable", NA)))
biomin_all_filtered_long $Treatment2 <- as.factor(biomin_all_filtered_long $Treatment2)
head(biomin_all_filtered_long)
```


## MERed signficant genes

```{r}
red_genes <- biomin_mod %>% filter(moduleColor == "red") %>% pull(Pocillopora_acuta_best_hit) %>% unique()

length(red_genes)

red_genes_data <- biomin_all %>% filter(Gene %in% red_genes) 
red_genes_data

red_genes_data_DE_Origin <- red_genes_data %>% filter(Origin < 0.05) %>% dplyr::select(c(Gene, Origin, definition, Ref))
red_genes_data_DE_Origin

unique(red_genes_data_DE_Origin$Gene)
```

### Carbonic Anhydrase (STPCA2-1)
```{r}
biomin_all_filtered_long_g12304 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g12304.t1")
biomin_all_filtered_long_g12304  
```

```{r}
library(nlme)
library(emmeans)
g12304.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g12304 , na.action=na.exclude)
car::Anova(g12304.lme, type=3)
tukey3<- emmeans(g12304.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g12304_sum<-summarySE(biomin_all_filtered_long_g12304 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g12304_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g12304_fig<-ggplot(data=g12304_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g12304,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(paste("Carbonic Anhydrase\n(STPCA2 1)")))+
  ggtitle(~MERed)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g12304_fig
```


### Protein lingerer-like
```{r}
biomin_all_filtered_long_g7908 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7908.t1")
biomin_all_filtered_long_g7908  
```

```{r}
library(nlme)
library(emmeans)
g7908.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g7908 , na.action=na.exclude)
car::Anova(g7908.lme, type=3)
tukey3<- emmeans(g7908.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g7908_sum<-summarySE(biomin_all_filtered_long_g7908 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g7908_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g7908_fig<-ggplot(data=g7908_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g7908,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(Protein~lingerer-like))+
  ggtitle(~MERed)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g7908_fig
```



## MEBrown signficant genes

```{r}
brown_genes <- biomin_mod %>% filter(moduleColor == "brown") %>% pull(Pocillopora_acuta_best_hit) %>% unique()

length(brown_genes)

brown_genes_data <- biomin_all %>% filter(Gene %in% brown_genes) 
brown_genes_data

brown_genes_data_DE_Origin <- brown_genes_data %>% filter(Origin < 0.05) %>% dplyr::select(c(Gene, Origin, definition, Ref))
brown_genes_data_DE_Origin

unique(brown_genes_data_DE_Origin$Gene)
```

### mammalian ependymin-related protein 1-like
```{r}
biomin_all_filtered_long_g25351 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25351.t1")
biomin_all_filtered_long_g25351  
```

```{r}
library(nlme)
library(emmeans)
g25351.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g25351 , na.action=na.exclude)
car::Anova(g25351.lme, type=3)
tukey3<- emmeans(g25351.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g25351_sum<-summarySE(biomin_all_filtered_long_g25351 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g25351_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g25351_fig<-ggplot(data=g25351_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g25351,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(paste("Mammalian ependymin\nrelated protein 1-like")))+
  ggtitle(~MEBrown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g25351_fig
```

### vitellogenin-like protein
```{r}
biomin_all_filtered_long_g13222 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g13222.t1b")
biomin_all_filtered_long_g13222  
```

```{r}
library(nlme)
library(emmeans)
g13222.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g13222 , na.action=na.exclude)
car::Anova(g13222.lme, type=3)
tukey3<- emmeans(g13222.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g13222_sum<-summarySE(biomin_all_filtered_long_g13222 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g13222_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g13222_fig<-ggplot(data=g13222_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g13222,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(Vitellogenin))+
  ggtitle(~MEBrown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g13222_fig
```

### uncharacterized protein LOC111323869
```{r}
biomin_all_filtered_long_g21232 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g21232.t1")
biomin_all_filtered_long_g21232  
```

```{r}
library(nlme)
library(emmeans)
g21232.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g21232 , na.action=na.exclude)
car::Anova(g21232.lme, type=3)
tukey3<- emmeans(g21232.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g21232_sum<-summarySE(biomin_all_filtered_long_g21232 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21232_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g21232_fig<-ggplot(data=g21232_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g21232,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(paste("Skeletal organic matrix\nprotein (LOC111323869)")))+
  ggtitle(~MEBrown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g21232_fig
```

### Cephalotoxin-like protein
```{r}
biomin_all_filtered_long_g5013 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g5013.t1")
biomin_all_filtered_long_g5013  
```

```{r}
library(nlme)
library(emmeans)
g5013.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g5013 , na.action=na.exclude)
car::Anova(g5013.lme, type=3)
tukey3<- emmeans(g5013.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g5013_sum<-summarySE(biomin_all_filtered_long_g5013 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g5013_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g5013_fig<-ggplot(data=g5013_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g5013,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(Cephalotoxin-like~protein))+
  ggtitle(~MEBrown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g5013_fig
```

### thioredoxin reductase 1, cytoplasmic-like
```{r}
biomin_all_filtered_long_g10093 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g10093.t2")
biomin_all_filtered_long_g10093  
```

```{r}
library(nlme)
library(emmeans)
g10093.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g10093 , na.action=na.exclude)
car::Anova(g10093.lme, type=3)
tukey3<- emmeans(g10093.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g10093_sum<-summarySE(biomin_all_filtered_long_g10093 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g10093_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g10093_fig<-ggplot(data=g10093_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g10093,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(Thioredoxin~reductase~1))+
  ggtitle(~MEBrown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g10093_fig
```

### carbonic anhydrase (STPCA2-2)
```{r}
biomin_all_filtered_long_g13824 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g13824.t1")
biomin_all_filtered_long_g13824  
```

```{r}
library(nlme)
library(emmeans)
g13824.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g13824 , na.action=na.exclude)
car::Anova(g13824.lme, type=3)
tukey3<- emmeans(g13824.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g13824_sum<-summarySE(biomin_all_filtered_long_g13824 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g13824_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g13824_fig<-ggplot(data=g13824_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g13824,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(paste("Carbonic Anhydrase\n(STPCA2 2)")))+
  ggtitle(~MEBrown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g13824_fig
```


## Comparison of all Brown and Red sig. genes
```{r}
biomin_compare_figs <- cowplot::plot_grid(g12304_fig,g7908_fig,g25351_fig,g13222_fig,g21232_fig, g5013_fig,g10093_fig,g13824_fig,nrow=3)
biomin_compare_figs

ggsave(filename="../../output/WGCNA/GO_analysis/biomin_Brown_Red_compare_figs.png", plot=biomin_compare_figs, dpi=300, height=8, units="in", limitsize=FALSE)
```

## Non-Module Biomin signficant genes

```{r}
Biomin_DE_Origin <- biomin_all %>% filter(Origin < 0.05) %>% dplyr::select(c(Gene, Origin, definition, Ref))
Biomin_DE_Origin

unique(Biomin_DE_Origin$Gene)
unique(red_genes_data_DE_Origin$Gene) %in% unique(Biomin_DE_Origin$Gene)
unique(brown_genes_data_DE_Origin$Gene) %in% unique(Biomin_DE_Origin$Gene)

unique(Biomin_DE_Origin$Gene) %in% c(unique(red_genes_data_DE_Origin$Gene),unique(brown_genes_data_DE_Origin$Gene))

setdiff(unique(Biomin_DE_Origin$Gene), c(unique(red_genes_data_DE_Origin$Gene),unique(brown_genes_data_DE_Origin$Gene)))

Biomin_DE_Origin %>% filter(Gene == setdiff(unique(Biomin_DE_Origin$Gene), c(unique(red_genes_data_DE_Origin$Gene),unique(brown_genes_data_DE_Origin$Gene)))
)
```

"Pocillopora_acuta_HIv2___RNAseq.g20587.t2" = uncharacterized protein LOC111345150 [Stylophora pistillata]

"Pocillopora_acuta_HIv2___RNAseq.g16715.t1" = Late embryogenesis protein

### uncharacterized protein LOC111345150
```{r}
biomin_all_filtered_long_g20587 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g20587.t2")
biomin_all_filtered_long_g20587  
```

```{r}
library(nlme)
library(emmeans)
g20587.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g20587 , na.action=na.exclude)
car::Anova(g20587.lme, type=3)
tukey3<- emmeans(g20587.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g20587_sum<-summarySE(biomin_all_filtered_long_g20587 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g20587_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g20587_fig<-ggplot(data=g20587_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g20587,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(paste("Skeletal organic matrix\nprotein (LOC111345150)")))+
  ggtitle("")+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12), plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g20587_fig
```

### Late embryogenesis protein
```{r}
biomin_all_filtered_long_g16715 <- biomin_all_filtered_long  %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g16715.t1")
biomin_all_filtered_long_g16715  
```

```{r}
library(nlme)
library(emmeans)
g16715.lme <- lme(Expression~Origin*Treatment2, random = ~1|Colony, data=biomin_all_filtered_long_g16715 , na.action=na.exclude)
car::Anova(g16715.lme, type=3)
tukey3<- emmeans(g16715.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g16715_sum<-summarySE(biomin_all_filtered_long_g16715 , measurevar='Expression', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g16715_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g16715_fig<-ggplot(data=g16715_sum, aes(y=Expression, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=biomin_all_filtered_long_g16715,aes(y=Expression, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Expression-se, ymax=Expression+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_color_manual("Origin", values=c("RS"='#26519B', "RF"= "#FE180C"))+
  scale_y_continuous(expression(paste("Late embryogenesis\nprotein")))+
  ggtitle("")+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12),
        plot.margin = unit(c(1,1,1,1), "lines"))#making the axis title larger 
g16715_fig
```

## Comparison of all Origin sig. genes of the Biomineralization toolkit
```{r}
biomin_compare_figs <- cowplot::plot_grid(g12304_fig,g7908_fig,g25351_fig,g13222_fig,g21232_fig, g5013_fig,g10093_fig,g13824_fig,g20587_fig,g16715_fig,nrow=2)
biomin_compare_figs

ggsave(filename="../../output/WGCNA/GO_analysis/biomin_Brown_Red_nonMod_compare_figs.png", plot=biomin_compare_figs, dpi=300, height=8, width=16, units="in")
```

