---
title: "WGCNA GO Analysis"
author: "Kristen Brown, modified ZD"
date: "1/29/2024"
output: html_document
---

# WGCNA GO Analysis

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo() #provides list of loaded packages and version of R. 
```

First, load the necessary packages.

```{r load_libraries, include = TRUE}
library(goseq)
library(dplyr)
library(forcats)
library(ggplot2)
library(gridExtra)
library(tidyr)
library(grDevices)
library(reshape2)
library(Rmisc)
library(ggpubr)
library(tibble)
library(gridExtra)
library(tidyr)
library(zoo)
library(ComplexHeatmap)
library(circlize)
library(GSEABase)
library(data.table)
library(stringr)
library(GenomicRanges)
library(rtracklayer)
library(rrvgo)
```

## Load in data
```{r}
library(rtracklayer)
gff<-rtracklayer::import("../../data/Pocillopora_acuta_HIv2.genes_fixed.gff3")
gff<-as.data.frame(gff)
dim(gff) # 478988     9
names(gff) 

transcripts <- subset(gff, type == "transcript")
transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information
transcript_lengths <- width(transcripts_gr) #isolate length of each gene
seqnames<-transcripts_gr$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame

dim(transcripts) #33730    13
```

```{r}
kegg <- read.delim("../../data/Pocillopora_acuta_HIv2.genes.KEGG_results.txt",header = FALSE)
kegg <- as.data.frame(kegg)
colnames(kegg)[1] <- "gene_id" 
colnames(kegg)[2] <- "KEGG_new"
head(kegg)
```


```{r}
eggnog<-read.delim("../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt")#this file contains all of the go terms, descriptions, kegg, etc
eggnog<- plyr::rename(eggnog, c("X.query"="gene_id"))
head(eggnog,2)
```

```{r}
gogene <- merge(transcripts, eggnog, by=c("gene_id"), all=T)
gogene <- merge(gogene, kegg, by=c("gene_id"), all=T)
head(gogene,2)
```


```{r}
geneInfo <- read.csv("../../output/WGCNA/WGCNA_ModuleMembership.csv") #this file was generated from the WGCNA analyses and contains the modules of interest
geneInfo<- plyr::rename(geneInfo, c("X"="gene_id"))
dim(geneInfo) # there are 9012 genes in our gene info file
```

```{r}
geneInfo <- merge(gogene, geneInfo, by=c("gene_id"), all=T)
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  GOs column
```{r}
geneInfo$GOs <- gsub(",", ";", geneInfo$GOs)
geneInfo$GOs <- gsub('"', "", geneInfo$GOs)
geneInfo$GOs <- gsub("-", NA, geneInfo$GOs)

geneInfo$KEGG_new[geneInfo$KEGG_new == ""] <- NA
```

```{r}
unique(geneInfo$moduleColor)
```

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
```

## Run the GOSeq function by module color to test for GO term enrichment. Due to high number of enriched GO terms, I am using an adjusted p-value threshold of <0.001 and using `rrvgo` package to reduce redundancy in list of GO terms.   

## Calcification

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```



```{r}
calc_up_mods <- c("brown", "red", "black", "pink", "salmon", "blue")

nrow(geneInfo %>% dplyr::filter(moduleColor=="brown")) #942
nrow(geneInfo %>% dplyr::filter(moduleColor=="red")) #425
nrow(geneInfo %>% filter(moduleColor=="black")) #396
nrow(geneInfo %>% filter(moduleColor=="pink")) #220
nrow(geneInfo %>% filter(moduleColor=="salmon")) #154
nrow(geneInfo %>% filter(moduleColor=="blue")) #1989

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="brown")), nrow(geneInfo %>% dplyr::filter(moduleColor=="red")), nrow(geneInfo %>% filter(moduleColor=="black")), nrow(geneInfo %>% filter(moduleColor=="pink")), nrow(geneInfo %>% filter(moduleColor=="salmon")), nrow(geneInfo %>% filter(moduleColor=="blue")))

# 4126

calc_down_mods <- c("turquoise","magenta","lightcyan")

nrow(geneInfo %>% dplyr::filter(moduleColor=="turquoise")) #2558
nrow(geneInfo %>% dplyr::filter(moduleColor=="magenta")) #219
nrow(geneInfo %>% filter(moduleColor=="lightcyan")) #65

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="turquoise")), nrow(geneInfo %>% dplyr::filter(moduleColor=="magenta")), nrow(geneInfo %>% filter(moduleColor=="lightcyan")))

# 2842

other_mods <- c("green","yellow", "purple", "midnightblue","cyan","tan")

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="green")), nrow(geneInfo %>% dplyr::filter(moduleColor=="yellow")), nrow(geneInfo %>% filter(moduleColor=="purple")), nrow(geneInfo %>% filter(moduleColor=="midnightblue")), nrow(geneInfo %>% filter(moduleColor=="cyan")),nrow(geneInfo %>% filter(moduleColor=="tan")))

# 2044

# 4126 + 2842 + 2044 = 9012, which represents all of our genes
```

4126 genes are in the 6 modules significantly upregulated by calcification.

```{r}
### Generate vector with names in just the module we are analyzing
# ID.vector <- geneInfo %>%
#   filter(moduleColor==c("brown", "red", "black", "pink", "salmon", "green")) %>%
#   #get_rows(.data[[module]]))%>%
#   pull(gene_id)

ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("brown", "red", "black", "pink", "salmon", "blue")) %>%
  pull(gene_id)

length(ID.vector) #4126

##Get a list of GO Terms for each module
GO.terms <- geneInfo %>%
  filter(moduleColor %in% c("brown", "red", "black", "pink", "salmon", "blue")) %>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GOs,gene_id) %>% rename(GOs = "GO.terms")

dim(GO.terms) #4126    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification.csv")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <- read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_calcification.csv")
```

```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
dim(go_results)
head(go_results)
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
dim(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
```

```{r}
write.csv(go_results, "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_filtered.csv")
```


```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_calcification <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_calcification

ggsave(filename="../../output/WGCNA/GO_analysis/GO.plot_calcification.png", plot=GO.plot_calcification, dpi=300, height=100, units="in", limitsize=FALSE)
```

### Count number of GOterms by ParentTerm for the upregulation of calcification
```{r}
library(dplyr)

result <- go_results %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Number.of.terms = n_distinct(term))%>%
  mutate(Calcification.direction = "Up")

print(result)
```

## Calcification - down

2842 genes are in the 4 modules significantly downregulated by calcification.

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor %in% c("turquoise","magenta","lightcyan"))%>%
  pull(gene_id)

length(ID.vector) #2842

##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor %in% c("turquoise","magenta","lightcyan"))%>%
  dplyr::select(GOs,gene_id) %>% rename(GOs = "GO.terms")

dim(GO.terms) #2842    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_down.csv")
```

```{r}
go_results <-read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_calcification_down.csv")
```

```{r}
go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)

dim(go_results)
head(go_results)
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
dim(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
```

```{r}
write.csv(go_results, "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_down_filtered.csv")
```


```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_calcification_down <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_calcification_down


ggsave(filename="../../output/WGCNA/GO_analysis/GO.plot_calcification_down.png", plot=GO.plot_calcification_down, dpi=300, height=100, units="in", limitsize=FALSE)
```

### Count number of GOterms by ParentTerm for the downregulation of calcification

```{r}
library(dplyr)

result_down <- go_results %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Number.of.terms = n_distinct(term))%>%
  mutate(Calcification.direction = "Down")

print(result_down)
```

## code for by module
```{r,eval=FALSE}
library(rrvgo)

# Define the unique module colors
module_colors <- na.omit(unique(geneInfo$moduleColor))

# Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)

# Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)

 
# Loop over each unique module color
for (color in module_colors) {
 # Filter geneInfo based on the current color
  color_filtered <- geneInfo %>% filter(moduleColor == color)

  # Generate vector with names in just the module we are analyzing
  ID.vector <- color_filtered$gene_id
  
  length(ID.vector)

  # Get a list of GO Terms for each module
  GO.terms <- color_filtered %>%
    dplyr::select(GOs, gene_id) %>%
    rename(GOs = "GO.terms")
  
  dim(GO.terms)

  ## Format to have one GO term per row with gene ID repeated
  split <- strsplit(as.character(GO.terms$GO.terms), ";") 
  split2 <- data.frame(v1 = rep.int(GO.terms$gene_id, sapply(split, length)), v2 = unlist(split)) 
  colnames(split2) <- c("gene", "GO.terms")
  GO.terms <- split2

  ## Construct list of genes with 1 for genes in module and 0 for genes not in the module
  gene.vector <- as.integer(ALL.vector %in% ID.vector) 
  names(gene.vector) <- ALL.vector # set names
  # Weight gene vector by bias for length of gene 
  pwf <- nullp(gene.vector, ID.vector, bias.data = LENGTH.vector) 

  # Run goseq using Wallenius method for all categories of GO terms 
  GO.wall <- goseq(pwf, ID.vector, gene2cat = GO.terms, test.cats = c("GO:BP", "GO:MF", "GO:CC"), method = "Wallenius", use_genes_without_cat = TRUE)
  GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
  colnames(GO)[1] <- "GOterm"

  # Adjust p-values 
  GO$bh_adjust <- p.adjust(GO$over_represented_pvalue, method = "BH") 

  # Filtering for p < 0.01
  GO <- GO %>%
    dplyr::filter(bh_adjust < 0.00001) %>%
    dplyr::arrange(., ontology, bh_adjust)
   
  # Write file of results 
  write.csv(GO, file = paste0("../../output/WGCNA/GO_analysis/goseq_pattern_", color, ".csv"))

  go_results <- GO

  go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
  
  #Reduce/collapse GO term set with the rrvgo package 
  simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
  
   #calculate similarity 
  scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
  reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
  
  #keep only the goterms from the reduced list
  go_results <- go_results %>% filter(GOterm %in% reducedTerms$go)
  
  #add in parent terms to list of go terms 
  go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
  
  #plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())

 GO.plot
 
  length(colnames(go_results)[go_results$ParentTerm=="cation transport"])
  length(colnames(go_results)[go_results$ParentTerm=="inorganic ion homeostasis"])
  length(colnames(go_results)[go_results$ParentTerm=="regulation of cellular response to stress"])
}
```

## Specific differentially expressed genes

```{r}
wgcna_counts_filtered<-read.csv("../../output/Filtered_gene_count_matrix.csv", strip.white=T)
wgcna_counts_filtered<- plyr::rename(wgcna_counts_filtered, c("X"="Gene"))
```

```{r}
colnames(wgcna_counts_filtered)
```

```{r}
library(tidyr)
wgcna_counts_filtered_long <- pivot_longer(wgcna_counts_filtered, cols=2:47, names_to = "Colony", values_to = "Counts")
wgcna_counts_filtered_long$Colony <- as.factor(wgcna_counts_filtered_long$Colony)
head(wgcna_counts_filtered_long)
```

```{r}
wgcna_counts_filtered_long <- wgcna_counts_filtered_long %>% 
  separate(Colony, into = c('Origin', 'Colony.number'), sep = 2)
head(wgcna_counts_filtered_long)
```

```{r}
library(stringr)
wgcna_counts_filtered_long$Colony <- as.numeric(str_extract(wgcna_counts_filtered_long$Colony.number, "[0-9]+"))
wgcna_counts_filtered_long<-wgcna_counts_filtered_long %>% 
   mutate(Treatment = trimws(str_remove(wgcna_counts_filtered_long$Colony.number, "(\\s+[A-Za-z]+)?[0-9-]+")))
head(wgcna_counts_filtered_long)
```

```{r}
wgcna_counts_filtered_long$Origin <- as.factor(wgcna_counts_filtered_long$Origin)
wgcna_counts_filtered_long$Treatment <- as.factor(wgcna_counts_filtered_long$Treatment)
```

```{r}
wgcna_counts_filtered_long <- wgcna_counts_filtered_long %>%
  mutate(Treatment2 = ifelse(Treatment == "A" | Treatment == "B", "Variable",
               ifelse(Treatment == "C" | Treatment == "D", "Stable", NA)))
wgcna_counts_filtered_long$Treatment2 <- as.factor(wgcna_counts_filtered_long$Treatment2)
```

## SLC4A7
```{r}
wgcna_counts_filtered_long_SLC4A7<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7402.t1")
```
```{r}
library(nlme)
library(emmeans)
SLC4A7.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_SLC4A7, na.action=na.exclude)
car::Anova(SLC4A7.lme, type=3)
tukey3<- emmeans(SLC4A7.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
SLC4A7_sum<-summarySE(wgcna_counts_filtered_long_SLC4A7, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
SLC4A7_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
SLC4A7_fig<-ggplot(data=SLC4A7_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_SLC4A7,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(SLC4A7~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
SLC4A7_fig
```
##SLC4A3 
```{r}
wgcna_counts_filtered_long_SLC4A3 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g27873.t1")
wgcna_counts_filtered_long_SLC4A3 
```
```{r}
SLC4A3.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_SLC4A3 , na.action=na.exclude)
car::Anova(SLC4A3.lme, type=3)
tukey3<- emmeans(SLC4A3.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
SLC4A3_sum<-summarySE(wgcna_counts_filtered_long_SLC4A3 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
SLC4A3_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
SLC4A3_fig<-ggplot(data=SLC4A3_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_SLC4A3 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(SLC4A3 ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
SLC4A3_fig
```

##NHE3
```{r}
wgcna_counts_filtered_long_NHE3<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g24868.t1")
wgcna_counts_filtered_long_NHE3
```
```{r}
NHE3.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_NHE3, na.action=na.exclude)
car::Anova(NHE3.lme, type=3)
tukey3<- emmeans(NHE3.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
NHE3_sum<-summarySE(wgcna_counts_filtered_long_NHE3, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
NHE3_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
NHE3_fig<-ggplot(data=NHE3_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_NHE3,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(NHE3~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
NHE3_fig
```

##CA1
```{r}
wgcna_counts_filtered_long_CA1<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g12304.t1")
wgcna_counts_filtered_long_CA1
```
```{r}
CA1.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_CA1, na.action=na.exclude)
car::Anova(CA1.lme, type=3)
tukey3<- emmeans(CA1.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA1_sum<-summarySE(wgcna_counts_filtered_long_CA1, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA1_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
CA1_fig<-ggplot(data=CA1_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_CA1,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA1~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA1_fig
```
##CA2
```{r}
wgcna_counts_filtered_long_CA2<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g13824.t1")
wgcna_counts_filtered_long_CA2
```
```{r}
CA2.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_CA2, na.action=na.exclude)
car::Anova(CA2.lme, type=3)
tukey3<- emmeans(CA2.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA2_sum<-summarySE(wgcna_counts_filtered_long_CA2, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA2_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
CA2_fig<-ggplot(data=CA2_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_CA2,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA2~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA2_fig
```
##VHA
```{r}
wgcna_counts_filtered_long_VHA<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g23064.t1")
wgcna_counts_filtered_long_VHA
```
```{r}
VHA.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_VHA, na.action=na.exclude)
car::Anova(VHA.lme, type=3)
tukey3<- emmeans(VHA.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
VHA_sum<-summarySE(wgcna_counts_filtered_long_VHA, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
VHA_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
VHA_fig<-ggplot(data=VHA_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_VHA,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(VHA~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
VHA_fig
```
##HSP90
```{r}
wgcna_counts_filtered_long_HSP90<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g6656.t1")
wgcna_counts_filtered_long_HSP90
```
```{r}
HSP90.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_HSP90, na.action=na.exclude)
car::Anova(HSP90.lme, type=3)
tukey3<- emmeans(HSP90.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
HSP90_sum<-summarySE(wgcna_counts_filtered_long_HSP90, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
HSP90_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
HSP90_fig<-ggplot(data=HSP90_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_HSP90,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(HSP90~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
HSP90_fig
```

##HIF1A 
```{r}
wgcna_counts_filtered_long_HIF1A <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g3039.t1")
wgcna_counts_filtered_long_HIF1A 
```
```{r}
HIF1A.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_HIF1A , na.action=na.exclude)
car::Anova(HIF1A.lme, type=3)
tukey3<- emmeans(HIF1A.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
HIF1A_sum<-summarySE(wgcna_counts_filtered_long_HIF1A , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
HIF1A_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
HIF1A_fig<-ggplot(data=HIF1A_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_HIF1A ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(HIF1A ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
HIF1A_fig
```


## HSP70 
```{r}
wgcna_counts_filtered_long_HSP70 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g10659.t1")
wgcna_counts_filtered_long_HSP70 
```
```{r}
HSP70.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_HSP70 , na.action=na.exclude)
car::Anova(HSP70.lme, type=3)
tukey3<- emmeans(HSP70.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
HSP70_sum<-summarySE(wgcna_counts_filtered_long_HSP70 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
HSP70_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
HSP70_fig<-ggplot(data=HSP70_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_HSP70 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(HSP70 ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
HSP70_fig
```

## PRKCD 
```{r}
wgcna_counts_filtered_long_PRKCD <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25259.t1")
wgcna_counts_filtered_long_PRKCD 
```
```{r}
PRKCD.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_PRKCD , na.action=na.exclude)
car::Anova(PRKCD.lme, type=3)
tukey3<- emmeans(PRKCD.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
PRKCD_sum<-summarySE(wgcna_counts_filtered_long_PRKCD , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
PRKCD_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
PRKCD_fig<-ggplot(data=PRKCD_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(PRKCD ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
PRKCD_fig
```

```{r}
compare_figs<-cowplot::plot_grid(SLC4A7_fig, NHE3_fig, CA1_fig, CA2_fig, HSP70_fig,HIF1A_fig, nrow=3)
compare_figs
```

## Biomineralization toolkit present in modules

```{r}
biomin <-read.csv("../../output/Biomin_blast_Pocillopora_acuta_best_hit.csv")
```

```{r}
wgcnamod <-read.csv("../../output/WGCNA/WGCNA_ModuleMembership.csv")
wgcnamod<- plyr::rename(wgcnamod, c("X"="Pocillopora_acuta_best_hit"))
```

```{r}
biomin_mod <- merge(biomin, wgcnamod, by=c("Pocillopora_acuta_best_hit"), all=F)
head(biomin_mod)
```

```{r}
plyr::count(biomin_mod, "moduleColor")
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  GOs column
```{r}
biomin_mod$GO.terms <- gsub(",", ";", biomin_mod$GO.terms)
biomin_mod$GO.terms <- gsub('"', "", biomin_mod$GO.terms)
biomin_mod$GO.terms <- gsub("-", NA, biomin_mod$GO.terms)
```

### GO terms

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
ID.vector_biomin <- biomin_mod %>%
  #filter(moduleColor=="black")%>%
  #get_rows(.data[[module]]))%>%
  pull(Pocillopora_acuta_best_hit)

##Get a list of GO Terms for each module
GO.terms_biomin <- biomin_mod %>%
  #filter(moduleColor=="black")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,Pocillopora_acuta_best_hit)
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms_biomin$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms_biomin$Pocillopora_acuta_best_hit, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("Pocillopora_acuta_best_hit", "GO.terms")
GO.terms_biomin<-split2
head(GO.terms_biomin)
```

```{r}
#GO.terms_biomin_sub <- GO.terms_biomin%>%
  #filter(Pocillopora_acuta_best_hit==c("Pocillopora_acuta_HIv2___RNAseq.g15280.t1","Pocillopora_acuta_HIv2___RNAseq.g7402.t1"))
#GO.terms_biomin_sub
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms_biomin, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        #dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "../../output/WGCNA/GO_analysis/goseq_pattern_biomin.csv")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_biomin.csv")
```

```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., bh_adjust)

head(go_results)
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
#calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
head(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]

go_results<-go_results %>%
  mutate(Factor = "Biomin")
head(go_results)
```

```{r}
write.csv(go_results, "../../output/Biomineralization_goterms.csv")
```

```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_biomin <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_biomin
```

## ***thioredoxin reductase 1
```{r}
wgcna_counts_filtered_long_g10093 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g10093.t2")
wgcna_counts_filtered_long_g10093  
```

```{r}
g10093.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g10093 , na.action=na.exclude)
car::Anova(g10093.lme, type=3)
tukey3<- emmeans(g10093.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g10093_sum<-summarySE(wgcna_counts_filtered_long_g10093 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g10093_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g10093_fig<-ggplot(data=g10093_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g10093,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Thioredoxin~reductase~1~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g10093_fig
```

## ***Flagellar associated protein
```{r}
wgcna_counts_filtered_long_g11609 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g11609.t1")
wgcna_counts_filtered_long_g11609  
```

```{r}
g11609.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g11609 , na.action=na.exclude)
car::Anova(g11609.lme, type=3)
tukey3<- emmeans(g11609.lme, list(pairwise ~ Treatment2:Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g11609_sum<-summarySE(wgcna_counts_filtered_long_g11609 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g11609_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g11609_fig<-ggplot(data=g11609_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g11609 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Flagellar~associated~protein~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g11609_fig
```

## ***Actin
```{r}
wgcna_counts_filtered_long_g14505 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g14505.t1")
wgcna_counts_filtered_long_g14505  
```

```{r}
g14505.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g14505 , na.action=na.exclude)
car::Anova(g14505.lme, type=3)
tukey3<- emmeans(g14505.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g14505_sum<-summarySE(wgcna_counts_filtered_long_g14505 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g14505_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g14505_fig<-ggplot(data=g14505_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g14505 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Actin~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g14505_fig
```

## ***CA2
```{r}
wgcna_counts_filtered_long_CA2<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g13824.t1")
wgcna_counts_filtered_long_CA2
```

```{r}
CA2.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_CA2, na.action=na.exclude)
car::Anova(CA2.lme, type=3)
tukey3<- emmeans(CA2.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA2_sum<-summarySE(wgcna_counts_filtered_long_CA2, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA2_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
CA2_fig<-ggplot(data=CA2_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_CA2,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA2~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA2_fig
```

## Poly [ADP-ribose] polymerase 11
```{r}
wgcna_counts_filtered_long_g14663 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g14663.t1a")
wgcna_counts_filtered_long_g14663  
```
```{r}
g14663.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g14663 , na.action=na.exclude)
car::Anova(g14663.lme, type=3)
tukey3<- emmeans(g14663.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g14663_sum<-summarySE(wgcna_counts_filtered_long_g14663 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g14663_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g14663_fig<-ggplot(data=g14663_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Poly~"[ADP-ribose]"~polymerase~11~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g14663_fig
```

## ***solute carrier family 4 member gamma
```{r}
wgcna_counts_filtered_long_g15280 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g15280.t1")
wgcna_counts_filtered_long_g15280  
```

```{r}
g15280.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g15280 , na.action=na.exclude)
car::Anova(g15280.lme, type=3)
tukey3<- emmeans(g15280.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g15280_sum<-summarySE(wgcna_counts_filtered_long_g15280 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g15280_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g15280_fig<-ggplot(data=g15280_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g15280 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Solute~carrier~family~4~member~gamma~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g15280_fig
```

## MAGUK p55 subfamily member 7-like
```{r}
wgcna_counts_filtered_long_g15517 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g15517.t1")
wgcna_counts_filtered_long_g15517  
```

```{r}
g15517.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g15517 , na.action=na.exclude)
car::Anova(g15517.lme, type=3)
tukey3<- emmeans(g15517.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g15517_sum<-summarySE(wgcna_counts_filtered_long_g15517 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g15517_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g15517_fig<-ggplot(data=g15517_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(MAGUK~p55~subfamily~member~7-like~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g15517_fig
```

## CARP1
```{r}
wgcna_counts_filtered_long_g16280 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g16280.t1")
wgcna_counts_filtered_long_g16280  
```

```{r}
g16280.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g16280 , na.action=na.exclude)
car::Anova(g16280.lme, type=3)
tukey3<- emmeans(g16280.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g16280_sum<-summarySE(wgcna_counts_filtered_long_g16280 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g16280_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g16280_fig<-ggplot(data=g16280_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CARP1~expression))+
  ggtitle(~tan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g16280_fig
```

## PHD finger protein 21A-like
```{r}
wgcna_counts_filtered_long_g1634 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g1634.t1")
wgcna_counts_filtered_long_g1634  
```

```{r}
g1634.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g1634 , na.action=na.exclude)
car::Anova(g1634.lme, type=3)
tukey3<- emmeans(g1634.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g1634_sum<-summarySE(wgcna_counts_filtered_long_g1634 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g1634_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g1634_fig<-ggplot(data=g1634_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(PHD~finger~protein~"21A"~like~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g1634_fig
```

## digestive cysteine proteinase 1-like
```{r}
wgcna_counts_filtered_long_g18103 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g18103.t1")
wgcna_counts_filtered_long_g18103  
```

```{r}
g18103.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g18103 , na.action=na.exclude)
car::Anova(g18103.lme, type=3)
tukey3<- emmeans(g18103.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g18103_sum<-summarySE(wgcna_counts_filtered_long_g18103 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g18103_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g18103_fig<-ggplot(data=g18103_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g18103 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(digestive~cysteine~proteinase~1-like~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g18103_fig
```

## endothelin-converting enzyme 1-like isoform X2
```{r}
wgcna_counts_filtered_long_g19211 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g19211.t1")
wgcna_counts_filtered_long_g19211  
```

```{r}
g19211.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g19211 , na.action=na.exclude)
car::Anova(g19211.lme, type=3)
tukey3<- emmeans(g19211.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g19211_sum<-summarySE(wgcna_counts_filtered_long_g19211 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g19211_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g19211_fig<-ggplot(data=g19211_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(endothelin-converting~enzyme~1-like~isoform~X2~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g19211_fig
```

## chymotrypsin-like elastase family member 1 
```{r}
wgcna_counts_filtered_long_g19288 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g19288.t1")
wgcna_counts_filtered_long_g19288  
```

```{r}
g19288.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g19288 , na.action=na.exclude)
car::Anova(g19288.lme, type=3)
tukey3<- emmeans(g19288.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g19288_sum<-summarySE(wgcna_counts_filtered_long_g19288 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g19288_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g19288_fig<-ggplot(data=g19288_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(chymotrypsin-like~elastase~family~member~1~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g19288_fig
```

## CUB and peptidase domain-containing protein 2-like 
```{r}
wgcna_counts_filtered_long_g21338 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g21338.t1")
wgcna_counts_filtered_long_g21338  
```

```{r}
g21338.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g21338 , na.action=na.exclude)
car::Anova(g21338.lme, type=3)
tukey3<- emmeans(g21338.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
 
```{r}
library(Rmisc)
g21338_sum<-summarySE(wgcna_counts_filtered_long_g21338 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21338_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g21338_fig<-ggplot(data=g21338_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CUB~and~peptidase~domain-containing~protein~2-like~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g21338_fig
```

##L-type calcium channel alpha-1 subunit
```{r}
wgcna_counts_filtered_long_g21501 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g21501.t1")
wgcna_counts_filtered_long_g21501  
```
```{r}
g21501.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g21501 , na.action=na.exclude)
car::Anova(g21501.lme, type=3)
tukey3<- emmeans(g21501.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g21501_sum<-summarySE(wgcna_counts_filtered_long_g21501 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21501_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g21501_fig<-ggplot(data=g21501_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(L-type~calcium~channel~alpha-1~subunit~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g21501_fig
```

##.Protocadherin
```{r}
wgcna_counts_filtered_long_g21501 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g22388.t1")
wgcna_counts_filtered_long_g21501  
```
```{r}
g21501.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g21501 , na.action=na.exclude)
car::Anova(g21501.lme, type=3)
tukey3<- emmeans(g21501.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g21501_sum<-summarySE(wgcna_counts_filtered_long_g21501 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21501_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g21501_fig<-ggplot(data=g21501_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g21501 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Protocadherin~expression), limits=c(100,200))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g21501_fig
```

##Acropora yongei Na+/Ca2+ exchanger
```{r}
wgcna_counts_filtered_long_g24639 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g24639.t1")
wgcna_counts_filtered_long_g24639  
```
```{r}
g24639.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g24639 , na.action=na.exclude)
car::Anova(g24639.lme, type=3)
tukey3<- emmeans(g24639.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g24639_sum<-summarySE(wgcna_counts_filtered_long_g24639 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g24639_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g24639_fig<-ggplot(data=g24639_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Acropora~yongei~"Na+/Ca2+"~exchanger~expression))+
  ggtitle(~tan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g24639_fig
```
##**mammalian ependymin-related protein 1-like
```{r}
wgcna_counts_filtered_long_g25351 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25351.t1")
wgcna_counts_filtered_long_g25351  
```
```{r}
g25351.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g25351 , na.action=na.exclude)
car::Anova(g25351.lme, type=3)
tukey3<- emmeans(g25351.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g25351_sum<-summarySE(wgcna_counts_filtered_long_g25351 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g25351_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g25351_fig<-ggplot(data=g25351_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g25351,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(mammalian~ependymin-related~protein~1-like~expression), limits=c(2500,12500))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g25351_fig
```
##MAM and LDLr domain-containing protein
```{r}
wgcna_counts_filtered_long_g25935 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25935.t1")
wgcna_counts_filtered_long_g25935  
```
```{r}
g25935.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g25935 , na.action=na.exclude)
car::Anova(g25935.lme, type=3)
tukey3<- emmeans(g25935.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g25935_sum<-summarySE(wgcna_counts_filtered_long_g25935 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g25935_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g25935_fig<-ggplot(data=g25935_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(MAM~and~LDLr~domain-containing~protein~expression))+
  ggtitle(~brown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g25935_fig
```

## SLIT-ROBO Rho GTPase-activating protein 1-like 
```{r}
wgcna_counts_filtered_long_g27376 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g27376.t1")
wgcna_counts_filtered_long_g27376  
```
```{r}
g27376.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27376 , na.action=na.exclude)
car::Anova(g27376.lme, type=3)
tukey3<- emmeans(g27376.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27376_sum<-summarySE(wgcna_counts_filtered_long_g27376 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27376_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g27376_fig<-ggplot(data=g27376_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(SLIT-ROBO~Rho~GTPase-activating~protein~1-like~expression))+
  ggtitle(~grey60)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27376_fig
```

##**Hephaestin-like protein - frontloaded
```{r}
wgcna_counts_filtered_long_g27566 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g27566.t1")
wgcna_counts_filtered_long_g27566  
```
```{r}
g27566.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27566 , na.action=na.exclude)
car::Anova(g27566.lme, type=3)
tukey3<- emmeans(g27566.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27566_sum<-summarySE(wgcna_counts_filtered_long_g27566 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27566_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g27566_fig<-ggplot(data=g27566_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g27566 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Hephaestin-like~protein~expression))+
  ggtitle(~brown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27566_fig
```

## plasma membrane calcium ATPase - frontloaded
```{r}
wgcna_counts_filtered_long_g27976 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g27976.t1")
wgcna_counts_filtered_long_g27976  
```
```{r}
g27976.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27976 , na.action=na.exclude)
car::Anova(g27976.lme, type=3)
tukey3<- emmeans(g27976.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27976_sum<-summarySE(wgcna_counts_filtered_long_g27976 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27976_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g27976_fig<-ggplot(data=g27976_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(plasma~membrane~calcium~ATPase~expression))+
  ggtitle(~brown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27976_fig
```

## von Willebrand factor D and EGF domain-containing protein-like
```{r}
wgcna_counts_filtered_long_g28226 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g28226.t2")
wgcna_counts_filtered_long_g28226  
```
```{r}
g28226.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g28226 , na.action=na.exclude)
car::Anova(g28226.lme, type=3)
tukey3<- emmeans(g28226.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g28226_sum<-summarySE(wgcna_counts_filtered_long_g28226 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g28226_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g28226_fig<-ggplot(data=g28226_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g28226,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(von~Willebrand~factor~D~and~EGF~domain-containing~protein-like~expression), limits=c(0,120))+
  ggtitle(~magenta)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g28226_fig
```

## low-density lipoprotein receptor-related protein 8-like 
```{r}
wgcna_counts_filtered_long_g4085 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g4085.t1")
wgcna_counts_filtered_long_g4085  
```
```{r}
g4085.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g4085 , na.action=na.exclude)
car::Anova(g4085.lme, type=3)
tukey3<- emmeans(g4085.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g4085_sum<-summarySE(wgcna_counts_filtered_long_g4085 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g4085_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g4085_fig<-ggplot(data=g4085_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(low-density~lipoprotein~receptor-related~protein~8-like~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g4085_fig
```

## **Cephalotoxin-like protein
```{r}
wgcna_counts_filtered_long_g5013 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g5013.t1")
wgcna_counts_filtered_long_g5013  
```
```{r}
g5013.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g5013 , na.action=na.exclude)
car::Anova(g5013.lme, type=3)
tukey3<- emmeans(g5013.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g5013_sum<-summarySE(wgcna_counts_filtered_long_g5013 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g5013_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g5013_fig<-ggplot(data=g5013_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g5013 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(cephalotoxin-like~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g5013_fig
```

## **sodium bicarbonate cotransporter 3-like isoform X - SLC4A7
```{r}
wgcna_counts_filtered_long_g7402 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7402.t1")
wgcna_counts_filtered_long_g7402  
```
```{r}
g7402.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g7402, na.action=na.exclude)
car::Anova(g7402.lme, type=3)
tukey3<- emmeans(g7402.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g7402_sum<-summarySE(wgcna_counts_filtered_long_g7402 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g7402_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g7402_fig<-ggplot(data=g7402_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g7402 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(sodium~bicarbonate~cotransporter~3-like~isoform~X2~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g7402_fig
```

##*protein lingerer-like
```{r}
wgcna_counts_filtered_long_g7902 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7908.t1")
wgcna_counts_filtered_long_g7902  
```
```{r}
g7902.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g7902, na.action=na.exclude)
car::Anova(g7902.lme, type=3)
tukey3<- emmeans(g7902.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g7902_sum<-summarySE(wgcna_counts_filtered_long_g7902 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g7902_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g7902_fig<-ggplot(data=g7902_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g7902,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(protein~lingerer-like~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g7902_fig
```

## ***CA1
```{r}
wgcna_counts_filtered_long_CA1<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g12304.t1")
wgcna_counts_filtered_long_CA1
```
```{r}
CA1.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_CA1, na.action=na.exclude)
car::Anova(CA1.lme, type=3)
tukey3<- emmeans(CA1.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA1_sum<-summarySE(wgcna_counts_filtered_long_CA1, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA1_sum
```
#### Figure
```{r}
pd<- position_dodge(0.2)
CA1_fig<-ggplot(data=CA1_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_CA1,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA1~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA1_fig
```

##Uncharacterized skeletal organic matrix protein-6 (USOMP6)
```{r}
wgcna_counts_filtered_long_g22622 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g22622.t1")
wgcna_counts_filtered_long_g22622  
```
```{r}
g22622.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g22622, na.action=na.exclude)
car::Anova(g22622.lme, type=3)
tukey3<- emmeans(g22622.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g22622_sum<-summarySE(wgcna_counts_filtered_long_g22622 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g22622_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g22622_fig<-ggplot(data=g22622_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Uncharacterized~skeletal~organic~matrix~protein-6~(USOMP6)~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g22622_fig
```

##band 3 anion transport protein-like
```{r}
wgcna_counts_filtered_long_g27873 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g27873.t1")
wgcna_counts_filtered_long_g27873  
```
```{r}
g27873.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27873, na.action=na.exclude)
car::Anova(g27873.lme, type=3)
tukey3<- emmeans(g27873.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27873_sum<-summarySE(wgcna_counts_filtered_long_g27873 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27873_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g27873_fig<-ggplot(data=g27873_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(band~3~anion~transport~protein-like~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27873_fig
```

##collagenase 3-like
```{r}
wgcna_counts_filtered_long_g5338 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g5338.t1")
wgcna_counts_filtered_long_g5338  
```
```{r}
g5338.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g5338, na.action=na.exclude)
car::Anova(g5338.lme, type=3)
tukey3<- emmeans(g5338.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g5338_sum<-summarySE(wgcna_counts_filtered_long_g5338 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g5338_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g5338_fig<-ggplot(data=g5338_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(collagenase~3-like~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g5338_fig
```

## Protocadherin
```{r}
wgcna_counts_filtered_long_g6583 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g6583.t1")
wgcna_counts_filtered_long_g6583  
```
```{r}
g6583.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g6583, na.action=na.exclude)
car::Anova(g6583.lme, type=3)
tukey3<- emmeans(g6583.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g6583_sum<-summarySE(wgcna_counts_filtered_long_g6583 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g6583_sum
```

#### Figure
```{r}
pd<- position_dodge(0.2)
g6583_fig<-ggplot(data=g6583_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Protocadherin~expression))+
  ggtitle(~black)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g6583_fig
```

## Comparison of sig genes
```{r}
biomin_compare_figs<-cowplot::plot_grid(g10093_fig,CA2_fig,g25351_fig,g5013_fig,g15280_fig, CA1_fig,g7402_fig,g7902_fig,g27566_fig,g28226_fig,g14505_fig,g11609_fig,g10093_fig, nrow=4)
biomin_compare_figs
```


## Merge the frequency of GOterms for both up- and down-regulation of calcification

```{r}
cal_biomin_terms <-read.csv("../../output/Biomineralization_goterms.csv") 
head(cal_biomin_terms)
```

```{r}
cal_up_terms <-read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_calcification_filtered.csv")
cal_up_terms<- cal_up_terms %>%
  mutate(Factor = "Up")
head(cal_up_terms)
```

```{r}
cal_down_terms <-read.csv("../../output/WGCNA/GO_analysis/goseq_pattern_calcification_down_filtered.csv")
cal_down_terms<-cal_down_terms %>%
  mutate(Factor = "Down")
head(cal_down_terms)
```

```{r}
colnames(cal_biomin_terms)
colnames(cal_up_terms)
colnames(cal_down_terms)
```

### Merge biomineralization, up and down-regulation of calcification GOterms
```{r}
all_terms<- merge(cal_up_terms,cal_down_terms, by=c("Factor","GOterm","X.1","X","GOterm","over_represented_pvalue","under_represented_pvalue","numDEInCat","numInCat","term","ontology","bh_adjust","ParentTerm"),all=T)


all_terms<- merge(all_terms,cal_biomin_terms, by=c("Factor","GOterm","X.1","X","GOterm","over_represented_pvalue","under_represented_pvalue","numDEInCat","numInCat","term","ontology","bh_adjust","ParentTerm"),all=T)

all_terms$GOterm<-as.factor(all_terms$GOterm)
head(all_terms)
tail(all_terms)
```

```{r}
goterms_shared <- all_terms %>%
  group_by(GOterm) %>%
  dplyr::summarise(
    ParentTerm = paste(unique(ParentTerm), collapse = ", "),
    Factor = paste(unique(Factor), collapse = ", ")
  )
dim(goterms_shared)
```

```{r}
write.csv(goterms_shared, "../../output/WGCNA/GO_analysis/Merged_GOterms_factor_ParentTerm.csv")
```

```{r}
result_unique <- goterms_shared %>%
  group_by(ParentTerm,Factor) %>%
  dplyr::summarise(SharedGOterms = n_distinct(GOterm))%>%
  arrange(-SharedGOterms)
```

```{r}
result_filtered_up<- result_unique %>%
  dplyr::filter(Factor=="Biomin, Down, Up" | Factor=="Biomin, Up")
 #dplyr::filter(SharedGOterms>=5) 
```

```{r}
result_filtered_down<- result_unique %>%
  dplyr::filter(Factor=="Biomin, Down, Up" | Factor=="Biomin, Down")
 #dplyr::filter(SharedGOterms>=5) 
```

### Calculate the percentage of shared GOterms upregulation of calcification
```{r}
merged_up <- result_filtered_up %>%
  tidyr::separate_rows(ParentTerm, sep = ", ") %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Sum_of_SharedGOterms = sum(SharedGOterms, na.rm = TRUE))%>%
  left_join(result, by = "ParentTerm") %>%
  mutate(Proportion.of.shared.GO.terms.with.biomineralization.genes = Sum_of_SharedGOterms / Number.of.terms)%>%
  mutate(Percentage.of.shared.GO.terms.with.biomineralization.genes = Proportion.of.shared.GO.terms.with.biomineralization.genes * 100)
merged_up_clean <- na.omit(merged_up)
merged_up_clean
```

### Calculate the percentage of shared GOterms downregulation of calcification
```{r}
merged_down <- result_filtered_down %>%
  tidyr::separate_rows(ParentTerm, sep = ", ") %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Sum_of_SharedGOterms = sum(SharedGOterms, na.rm = TRUE))%>%
  left_join(result_down, by = "ParentTerm") %>%
  mutate(Proportion.of.shared.GO.terms.with.biomineralization.genes = Sum_of_SharedGOterms / Number.of.terms)%>%
  mutate(Percentage.of.shared.GO.terms.with.biomineralization.genes = Proportion.of.shared.GO.terms.with.biomineralization.genes * 100)
merged_down_clean<- na.omit(merged_down)
merged_down_clean
```

###Frequency >10 up
```{r}
cal_freq_terms_filtered_up <- merged_up_clean %>%
  filter(Number.of.terms>=10) %>%
  filter(Calcification.direction=="Up")
cal_freq_terms_filtered_up
```

#### Figure
```{r}
#counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
#counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
freq_fig_up<-ggplot(cal_freq_terms_filtered_up, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,70))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_up

ggsave(filename="../../output/WGCNA/GO_analysis/freq_fig_up.png", plot=freq_fig_up, dpi=300, height=12, units="in", limitsize=FALSE)
```

###Frequency >10 down
```{r}
cal_freq_terms_filtered_down <- merged_down_clean %>%
  filter(Number.of.terms>=10)
  #filter(Calcification.direction=="Down")
cal_freq_terms_filtered_down
```

#### Figure
```{r} 
freq_fig_down<-ggplot(cal_freq_terms_filtered_down, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  #geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,70))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#d1e5f0","#92c5de","#4393c3","#2166ac"), na.value = "grey98",limits = c(0, 100))+ 
  #scale_fill_gradientn(colours=c("#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(10, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_down

ggsave(filename="../../output/WGCNA/GO_analysis/freq_fig_down.png", plot=freq_fig_down, dpi=300, height=12, units="in", limitsize=FALSE)
```
```{r}
compare_figs<-cowplot::plot_grid(freq_fig_up, freq_fig_down, nrow=2, align="v")
compare_figs
```

### All terms up
```{r}
cal_freq_terms_filtered_up_all <- merged_up_clean %>%
  filter(Calcification.direction=="Up")
cal_freq_terms_filtered_up_all
```

#### Figure
```{r}
#counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
#counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
freq_fig_up<-ggplot(cal_freq_terms_filtered_up_all, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,70))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_up

ggsave(filename="../../output/WGCNA/GO_analysis/freq_fig_up_ALL.png", plot=freq_fig_up, dpi=300, height=17, units="in", limitsize=FALSE)
```

### All terms down
```{r}
cal_freq_terms_filtered_down_all <- merged_down_clean %>%
  filter(Calcification.direction=="Down")
cal_freq_terms_filtered_down_all
```

#### Figure
```{r}
freq_fig_down<-ggplot(cal_freq_terms_filtered_down_all, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  #geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,70))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#d1e5f0","#92c5de","#4393c3","#2166ac"), na.value = "grey98",limits = c(0, 100))+ 
  #scale_fill_gradientn(colours=c("#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(10, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_down

ggsave(filename="../../output/WGCNA/GO_analysis/freq_fig_down_ALL.png", plot=freq_fig_down, dpi=300, height=17, units="in", limitsize=FALSE)
```

```{r}
compare_figs<-cowplot::plot_grid(freq_fig_up, freq_fig_down, ncol=2, align="h")
compare_figs
```