---
title: "WGCNA GO Analysis"
author: "Kristen Brown, modified ZD"
date: "1/29/2024"
output: html_document
---

# WGCNA GO Analysis

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo() #provides list of loaded packages and version of R. 
```

```{r}
#BiocManager::install("simplifyEnrichment")
```

First, load the necessary packages.

```{r load_libraries, include = TRUE}
library(goseq)
library(dplyr)
library(forcats)
library(ggplot2)
library(gridExtra)
library(tidyr)
library(grDevices)
library(reshape2)
library(Rmisc)
library(ggpubr)
library(tibble)
library(gridExtra)
library(tidyr)
library(zoo)
library(ComplexHeatmap)
library(circlize)
library(GSEABase)
library(data.table)
library(stringr)
library(GenomicRanges)
library(rtracklayer)
library(rrvgo)
library(scales)
```

## Load in data
```{r}
library(rtracklayer)
gff<-rtracklayer::import("../../data/Pocillopora_acuta_HIv2.genes_fixed.gff3")
gff<-as.data.frame(gff) %>% dplyr::select(-Parent)
dim(gff) # 478988     9
names(gff) 

transcripts <- subset(gff, type == "transcript")
transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information
transcript_lengths <- width(transcripts_gr) #isolate length of each gene
seqnames<-transcripts_gr$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame

dim(transcripts) #33730    13
```

```{r}
kegg <- read.delim("../../data/Pocillopora_acuta_HIv2.genes.KEGG_results.txt",header = FALSE)
kegg <- as.data.frame(kegg)
colnames(kegg)[1] <- "gene_id" 
colnames(kegg)[2] <- "KEGG_new"
head(kegg)
```


```{r}
eggnog<-read.delim("../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt")#this file contains all of the go terms, descriptions, kegg, etc
eggnog<- plyr::rename(eggnog, c("X.query"="gene_id"))
head(eggnog,2)
```

```{r}
gogene <- merge(transcripts, eggnog, by=c("gene_id"), all=T)
gogene <- merge(gogene, kegg, by=c("gene_id"), all=T)
head(gogene,2)
dim(gogene)
```


```{r}
geneInfo <- read.csv("../../output/WGCNA/WGCNA_ModuleMembership.csv") #this file was generated from the WGCNA analyses and contains the modules of interest
geneInfo<- plyr::rename(geneInfo, c("X"="gene_id"))
dim(geneInfo) # there are 9012 genes in our gene info file
```

```{r}
geneInfo <- merge(gogene, geneInfo, by=c("gene_id")) #merging the GO and Kegg info to module membership for the 9012 genes
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  GOs column
```{r}
geneInfo$GOs <- gsub(",", ";", geneInfo$GOs)
geneInfo$GOs <- gsub('"', "", geneInfo$GOs)
geneInfo$GOs <- gsub("-", NA, geneInfo$GOs)

geneInfo$KEGG_new[geneInfo$KEGG_new == ""] <- NA
```

```{r}
unique(geneInfo$moduleColor)
```

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
dim(geneInfo)
write.csv(geneInfo, file = "../../output/WGCNA/GO_analysis/geneInfo_WGCNA.csv") #gene info for reference/supplement
```

## Modules

```{r}
calc_up_mods <- c("brown", "red", "black", "pink", "salmon", "blue")

nrow(geneInfo %>% dplyr::filter(moduleColor=="brown")) #942
nrow(geneInfo %>% dplyr::filter(moduleColor=="red")) #425
nrow(geneInfo %>% filter(moduleColor=="black")) #396
nrow(geneInfo %>% filter(moduleColor=="pink")) #220
nrow(geneInfo %>% filter(moduleColor=="salmon")) #154
nrow(geneInfo %>% filter(moduleColor=="blue")) #1989

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="brown")), nrow(geneInfo %>% dplyr::filter(moduleColor=="red")), nrow(geneInfo %>% filter(moduleColor=="black")), nrow(geneInfo %>% filter(moduleColor=="pink")), nrow(geneInfo %>% filter(moduleColor=="salmon")), nrow(geneInfo %>% filter(moduleColor=="blue")))

# 4126

calc_down_mods <- c("turquoise","magenta","lightcyan")

nrow(geneInfo %>% dplyr::filter(moduleColor=="turquoise")) #2558
nrow(geneInfo %>% dplyr::filter(moduleColor=="magenta")) #219
nrow(geneInfo %>% filter(moduleColor=="lightcyan")) #65

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="turquoise")), nrow(geneInfo %>% dplyr::filter(moduleColor=="magenta")), nrow(geneInfo %>% filter(moduleColor=="lightcyan")))

# 2842

other_mods <- c("green","yellow", "purple", "midnightblue","cyan","tan")

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="green")), nrow(geneInfo %>% dplyr::filter(moduleColor=="yellow")), nrow(geneInfo %>% filter(moduleColor=="purple")), nrow(geneInfo %>% filter(moduleColor=="midnightblue")), nrow(geneInfo %>% filter(moduleColor=="cyan")),nrow(geneInfo %>% filter(moduleColor=="tan")))

# 2044

# 4126 + 2842 + 2044 = 9012, which represents all of our genes
```

## Calcification - Up - By Module

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

#### Brown

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("brown")) %>%
  pull(gene_id)

length(ID.vector) #942

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector #set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_brown <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Brown")

go_results_BP <- GO_05_brown %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

```{r}
#Write file of results 
write.csv(GO_05_brown, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_brown.csv")
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results_BP$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results_BP$over_represented_pvalue), go_results_BP$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
dim(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results_BP_reduced <- go_results_BP %>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results_BP_reduced$ParentTerm <- reducedTerms$parentTerm[match(go_results_BP_reduced$GOterm, reducedTerms$go)]

length(unique(go_results_BP_reduced$ParentTerm))

go_results_BP_reduced <- go_results_BP_reduced %>% group_by(ParentTerm) %>% mutate("N_in_Parent" = n()) %>% ungroup()
```

The reduced list of terms is 157 terms that falls under 30 parent terms.

```{r}
write.csv(go_results_BP_reduced, "../../output/WGCNA/GO_analysis/goseq_brown_filtered.csv")
```

What about for p < 0.01?

```{r}
go_results_BP_reduced_01 <- go_results_BP_reduced %>% filter(over_represented_pvalue < 0.01) 
dim(go_results_BP_reduced_01)
length(unique(go_results_BP_reduced_01$ParentTerm))
```

```{r}
go_results_BP_reduced_Brown <- go_results_BP_reduced_01

freq_fig_brown <- ggplot(go_results_BP_reduced_Brown, aes(y=N_in_Parent,x=reorder(ParentTerm, N_in_Parent)))+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=N_in_Parent)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(limits=c(0,40))+
   scale_x_discrete(labels = label_wrap(30)) +
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100)) +
  labs(title = "Enriched Biological Process Parent GO Terms, Brown Module", x="Parent Term", y="Number of Significant (p < 0.01) GO Terms in Module", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 8), 
              axis.title = element_text(size = 8), 
              plot.title = element_text(hjust = 0.5, size = 8))

freq_fig_brown

ggsave("../../output/WGCNA/GO_analysis/Parent_by_mod/Brown_Parent_filtp01.pdf", freq_fig_brown, width = 5, height = 4, units = c("in"), dpi=300)
```



This is only 41 terms that fall into 12 parent terms

```{r}
go_results_BP_reduced_Brown <- go_results_BP_reduced_01

top_5_parent <- go_results_BP_reduced_Brown %>% dplyr::select(ParentTerm, N_in_Parent) %>% distinct() %>% arrange(-N_in_Parent) %>% head(5)

Top5Plot <- go_results_BP_reduced_Brown %>% 
    filter(ParentTerm %in% top_5_parent$ParentTerm) %>%
    filter(over_represented_pvalue <0.01) %>%
    mutate(hitsPerc=numDEInCat/numInCat) %>% 
    arrange(-N_in_Parent) %>%
  mutate(ParentTerm = factor(ParentTerm, levels = unique(ParentTerm))) %>%
    
    ggplot(aes(x=numDEInCat, 
               y=reorder(term, numDEInCat), 
               fill=over_represented_pvalue)) +
        scale_fill_gradient(low = "black", high = "white", name = "p value",guide=guide_colorbar(reverse=TRUE),limits=c(0,0.01) ) +
        xlim(0, 60) +
        facet_grid(ParentTerm ~ .,scales="free_y", space = "free",labeller = labeller(ParentTerm = label_wrap_gen(width = 25))) + 
        geom_segment(aes(y=term ,yend=term, x=0, xend=numDEInCat), color="black",linewidth=0.3) +
       geom_point(shape=21) +
    scale_y_discrete(labels = label_wrap(30)) + 
        labs(title = "Enriched Biological Process GO Terms p < 0.01,\nFacet by Parent Term (Top 5), Brown Module", x="Count", y="GO term", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 4), 
              axis.title = element_text(size = 10), 
              plot.title = element_text(hjust = 0.5, size = 10),
              strip.text.y = element_text(size = 6,angle = 0,face = "bold"))

Top5Plot

ggsave("../../output/WGCNA/GO_analysis/Top10_BP_by_mod/Brown_allterms_top5parent_filtp01.pdf", Top5Plot, width = 5, height = 5, units = c("in"), dpi=300, limitsize=FALSE)
```


#### Red

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("red")) %>%
  pull(gene_id)

length(ID.vector) #425

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_red <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Red")

go_results_BP <- GO_05_red %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

```{r}
#Write file of results 
write.csv(GO_05_red, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_red.csv")
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results_BP$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results_BP$over_represented_pvalue), go_results_BP$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
dim(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results_BP_reduced <- go_results_BP %>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results_BP_reduced$ParentTerm <- reducedTerms$parentTerm[match(go_results_BP_reduced$GOterm, reducedTerms$go)]

length(unique(go_results_BP_reduced$ParentTerm))

go_results_BP_reduced <- go_results_BP_reduced %>% group_by(ParentTerm) %>% mutate("N_in_Parent" = n()) %>% ungroup()
```

The reduced list of terms is 85 terms that falls under 26 parent terms.

```{r}
write.csv(go_results_BP_reduced, "../../output/WGCNA/GO_analysis/goseq_red_filtered.csv")
```


What about for p < 0.01?

```{r}
go_results_BP_reduced_01 <- go_results_BP_reduced %>% filter(over_represented_pvalue < 0.01) 
dim(go_results_BP_reduced_01)
length(unique(go_results_BP_reduced_01$ParentTerm))
```

This is only 16 terms that fall into 9 parent terms

```{r}
go_results_BP_reduced_Red <- go_results_BP_reduced_01

freq_fig_red <- ggplot(go_results_BP_reduced_Red, aes(y=N_in_Parent,x=reorder(ParentTerm, N_in_Parent)))+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=N_in_Parent)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(limits=c(0,40))+
   scale_x_discrete(labels = label_wrap(30)) +
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100)) +
  labs(title = "Enriched Biological Process Parent GO Terms, Red Module", x="Parent Term", y="Number of Significant (p < 0.01) GO Terms in Module", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 8), 
              axis.title = element_text(size = 8), 
              plot.title = element_text(hjust = 0.5, size = 8))

freq_fig_red

ggsave("../../output/WGCNA/GO_analysis/Parent_by_mod/Red_Parent_filtp01.pdf", freq_fig_red, width = 5, height = 4, units = c("in"), dpi=300)
```

```{r}
go_results_BP_reduced_Red <- go_results_BP_reduced_01

top_5_parent <- go_results_BP_reduced_Red %>% dplyr::select(ParentTerm, N_in_Parent) %>% distinct() %>% arrange(-N_in_Parent) %>% head(5)

Top5Plot <- go_results_BP_reduced_Red %>% 
    filter(ParentTerm %in% top_5_parent$ParentTerm) %>%
    filter(over_represented_pvalue <0.01) %>%
    mutate(hitsPerc=numDEInCat/numInCat) %>% 
    arrange(-N_in_Parent) %>%
  mutate(ParentTerm = factor(ParentTerm, levels = unique(ParentTerm))) %>%
    
    ggplot(aes(x=numDEInCat, 
               y=reorder(term, numDEInCat), 
               fill=over_represented_pvalue)) +
        scale_fill_gradient(low = "black", high = "white", name = "p value",guide=guide_colorbar(reverse=TRUE),limits=c(0,0.01)) +
        xlim(0, 60) +
        facet_grid(ParentTerm ~ .,scales="free_y", space = "free",labeller = labeller(ParentTerm = label_wrap_gen(width = 25))) + 
        geom_segment(aes(y=term ,yend=term, x=0, xend=numDEInCat), color="black",linewidth=0.3) +
       geom_point(shape=21) +
    scale_y_discrete(labels = label_wrap(30)) + 
        labs(title = "Enriched Biological Process GO Terms p < 0.01,\nFacet by Parent Term (Top 5), Red Module", x="Count", y="GO term", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 4), 
              axis.title = element_text(size = 10), 
              plot.title = element_text(hjust = 0.5, size = 10),
              strip.text.y = element_text(size = 6,angle = 0,face = "bold"))

Top5Plot

ggsave("../../output/WGCNA/GO_analysis/Top10_BP_by_mod/Red_allterms_top5parent_filtp01.pdf", Top5Plot, width = 5, height = 5, units = c("in"), dpi=300, limitsize=FALSE)
```

#### Black

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("black")) %>%
  pull(gene_id)

length(ID.vector) #396

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_black <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Black")

go_results_BP <- GO_05_black %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

#### Pink

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("pink")) %>%
  pull(gene_id)

length(ID.vector) #220

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_pink <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Pink")

go_results_BP <- GO_05_pink %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

#### Salmon

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("salmon")) %>%
  pull(gene_id)

length(ID.vector) #154

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_salmon <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Salmon")

go_results_BP <- GO_05_salmon %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

#### Blue

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("blue")) %>%
  pull(gene_id)

length(ID.vector) #1989

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_blue <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Blue")

go_results_BP <- GO_05_blue %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

### Combining all the individual module lists for Calcification - Up

```{r}
GO_05_UP <- rbind(GO_05_brown,GO_05_red,GO_05_black,GO_05_pink,GO_05_salmon,GO_05_blue)

length(GO_05_UP$GOterm) #3810 enriched GO terms
length(unique(GO_05_UP$GOterm)) #3617 enriched GO terms

# Collapse GO terms that are duplicated accross modules - sort by GO term and p-value, then keep the row with the lowest p-value, collapsing the module names so you know which modules were represented

# Identify and collapse duplicates
GO_05_UP <- GO_05_UP %>%
  arrange(GOterm, over_represented_pvalue) %>%
  group_by(GOterm) %>% mutate(Module = paste(unique(Module), collapse = ";")) %>%
  group_by(GOterm) %>%
  slice_min(order_by = over_represented_pvalue, n = 1) %>%
  ungroup()

length(GO_05_UP$GOterm) #3617 enriched GO terms
length(unique(GO_05_UP$GOterm)) #3617 enriched GO terms, all are unique

#Write file of results 
write.csv(GO_05_UP, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_up_bymod.csv")
```

## Calcification - Down - By Module

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

#### Turquoise

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("turquoise")) %>%
  pull(gene_id)

length(ID.vector) #2558

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector #set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_turquoise <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Turquoise")

go_results_BP <- GO_05_turquoise %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```


```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results_BP$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results_BP$over_represented_pvalue), go_results_BP$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
dim(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results_BP_reduced <- go_results_BP %>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results_BP_reduced$ParentTerm <- reducedTerms$parentTerm[match(go_results_BP_reduced$GOterm, reducedTerms$go)]

length(unique(go_results_BP_reduced$ParentTerm))

go_results_BP_reduced <- go_results_BP_reduced %>% group_by(ParentTerm) %>% mutate("N_in_Parent" = n()) %>% ungroup()
```

The reduced list of terms is 291 terms that falls under 46 parent terms.

```{r}
write.csv(go_results_BP_reduced, "../../output/WGCNA/GO_analysis/goseq_turquoise_filtered.csv")
```


What about for p < 0.01?

```{r}
go_results_BP_reduced_01 <- go_results_BP_reduced %>% filter(over_represented_pvalue < 0.01) 
dim(go_results_BP_reduced_01)
length(unique(go_results_BP_reduced_01$ParentTerm))
```

This is only 155 terms that fall into 28 parent terms

```{r}
go_results_BP_reduced_Turquoise <- go_results_BP_reduced_01

freq_fig_turquoise <- ggplot(go_results_BP_reduced_Turquoise, aes(y=N_in_Parent,x=reorder(ParentTerm, N_in_Parent)))+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=N_in_Parent)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(limits=c(0,40))+
   scale_x_discrete(labels = label_wrap(30)) +
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100)) +
  labs(title = "Enriched Biological Process Parent GO Terms, Turquoise Module", x="Parent Term", y="Number of Significant (p < 0.01) GO Terms in Module", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 8), 
              axis.title = element_text(size = 8), 
              plot.title = element_text(hjust = 0.5, size = 8))

freq_fig_turquoise

ggsave("../../output/WGCNA/GO_analysis/Parent_by_mod/Turquoise_Parent_filtp01.pdf", freq_fig_turquoise, width = 5, height = 8, units = c("in"), dpi=300)


go_results_BP_reduced_Turquoise_001 <- go_results_BP_reduced_Turquoise %>% filter(over_represented_pvalue <0.001)
  
freq_fig_turquoise_001 <- ggplot(go_results_BP_reduced_Turquoise_001, aes(y=N_in_Parent,x=reorder(ParentTerm, N_in_Parent)))+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=N_in_Parent)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(limits=c(0,40))+
   scale_x_discrete(labels = label_wrap(30)) +
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100)) +
  labs(title = "Enriched Biological Process Parent GO Terms, Turquoise Module", x="Parent Term", y="Number of Significant (p < 0.001) GO Terms in Module", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 5), 
              axis.title = element_text(size = 8), 
              plot.title = element_text(hjust = 0.5, size = 8))

freq_fig_turquoise_001

ggsave("../../output/WGCNA/GO_analysis/Parent_by_mod/Turquoise_Parent_filtp001.pdf", freq_fig_turquoise_001, width = 5, height = 4, units = c("in"), dpi=300)
```

```{r}
go_results_BP_reduced_Turquoise <- go_results_BP_reduced_01

top_5_parent <- go_results_BP_reduced_Turquoise %>% dplyr::select(ParentTerm, N_in_Parent) %>% distinct() %>% arrange(-N_in_Parent) %>% head(5)

Top5Plot <- go_results_BP_reduced_Turquoise %>% 
    filter(ParentTerm %in% top_5_parent$ParentTerm) %>%
    filter(over_represented_pvalue <0.001) %>%
    mutate(hitsPerc=numDEInCat/numInCat) %>% 
    arrange(-N_in_Parent) %>%
  mutate(ParentTerm = factor(ParentTerm, levels = unique(ParentTerm))) %>%
    
    ggplot(aes(x=numDEInCat, 
               y=reorder(term, numDEInCat), 
               fill=over_represented_pvalue)) +
        scale_fill_gradient(low = "black", high = "white", name = "p value",guide=guide_colorbar(reverse=TRUE),limits=c(0,0.01)) +
        expand_limits(x=1.3) +
        facet_grid(ParentTerm ~ .,scales="free_y", space = "free",labeller = labeller(ParentTerm = label_wrap_gen(width = 25))) + 
        geom_segment(aes(y=term ,yend=term, x=0, xend=numDEInCat), color="black",linewidth=0.3) +
       geom_point(shape=21) +
    scale_y_discrete(labels = label_wrap(30)) + 
        labs(title = "Enriched Biological Process GO Terms p < 0.001,\nFacet by Parent Term (Top 5), Turquoise Module", x="Count", y="GO term", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 4), 
              axis.title = element_text(size = 10), 
              plot.title = element_text(hjust = 0.5, size = 10),
              strip.text.y = element_text(size = 6,angle = 0,face = "bold"))

Top5Plot

ggsave("../../output/WGCNA/GO_analysis/Top10_BP_by_mod/Turquoise_allterms_top5parent_filtp001.pdf", Top5Plot, width = 5, height = 7, units = c("in"), dpi=300, limitsize=FALSE)
```


#### Magenta

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("magenta")) %>%
  pull(gene_id)

length(ID.vector) #219

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_magenta <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Magenta")

go_results_BP <- GO_05_magenta %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

```{r}
#Write file of results 
write.csv(GO_05_magenta, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_magenta.csv")
```


```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results_BP$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results_BP$over_represented_pvalue), go_results_BP$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
dim(reducedTerms)
```

```{r}
#keep only the goterms from the reduced list
go_results_BP_reduced <- go_results_BP %>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results_BP_reduced$ParentTerm <- reducedTerms$parentTerm[match(go_results_BP_reduced$GOterm, reducedTerms$go)]

length(unique(go_results_BP_reduced$ParentTerm))

go_results_BP_reduced <- go_results_BP_reduced %>% group_by(ParentTerm) %>% mutate("N_in_Parent" = n()) %>% ungroup()
```

The reduced list of terms is 79 terms that falls under 23 parent terms.

```{r}
write.csv(go_results_BP_reduced, "../../output/WGCNA/GO_analysis/goseq_magenta_filtered.csv")
```


What about for p < 0.01?

```{r}
go_results_BP_reduced_01 <- go_results_BP_reduced %>% filter(over_represented_pvalue < 0.01) 
dim(go_results_BP_reduced_01)
length(unique(go_results_BP_reduced_01$ParentTerm))
```

This is only 14 terms that fall into 6 parent terms

```{r}
go_results_BP_reduced_Magenta <- go_results_BP_reduced_01

freq_fig_magenta <- ggplot(go_results_BP_reduced_Magenta, aes(y=N_in_Parent,x=reorder(ParentTerm, N_in_Parent)))+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=N_in_Parent)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(limits=c(0,40))+
   scale_x_discrete(labels = label_wrap(30)) +
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100)) +
  labs(title = "Enriched Biological Process Parent GO Terms, Magenta Module", x="Parent Term", y="Number of Significant (p < 0.01) GO Terms in Module", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 8), 
              axis.title = element_text(size = 8), 
              plot.title = element_text(hjust = 0.5, size = 8))

freq_fig_red

ggsave("../../output/WGCNA/GO_analysis/Parent_by_mod/Magenta_Parent_filtp01.pdf", freq_fig_magenta, width = 5, height = 4, units = c("in"), dpi=300)
```

```{r}
go_results_BP_reduced_Magenta <- go_results_BP_reduced_01

top_5_parent <- go_results_BP_reduced_Magenta %>% dplyr::select(ParentTerm, N_in_Parent) %>% distinct() %>% arrange(-N_in_Parent) %>% head(10)

Top5Plot <- go_results_BP_reduced_Magenta %>% 
    filter(ParentTerm %in% top_5_parent$ParentTerm) %>%
    filter(over_represented_pvalue <0.01) %>%
    mutate(hitsPerc=numDEInCat/numInCat) %>% 
    arrange(-N_in_Parent) %>%
  mutate(ParentTerm = factor(ParentTerm, levels = unique(ParentTerm))) %>%
    
    ggplot(aes(x=numDEInCat, 
               y=reorder(term, numDEInCat), 
               fill=over_represented_pvalue)) +
        scale_fill_gradient(low = "black", high = "white", name = "p value",guide=guide_colorbar(reverse=TRUE),limits=c(0,0.01)) +
        xlim(0, 60) +
        facet_grid(ParentTerm ~ .,scales="free_y", space = "free",labeller = labeller(ParentTerm = label_wrap_gen(width = 25))) + 
        geom_segment(aes(y=term ,yend=term, x=0, xend=numDEInCat), color="black",linewidth=0.3) +
       geom_point(shape=21) +
    scale_y_discrete(labels = label_wrap(30)) + 
        labs(title = "Enriched Biological Process GO Terms p < 0.01,\nFacet by Parent Term (Top 5), Magenta Module", x="Count", y="GO term", colour="p-value") +
        theme_bw() + 
        theme(axis.text.y = element_text(size = 4), 
              axis.title = element_text(size = 10), 
              plot.title = element_text(hjust = 0.5, size = 10),
              strip.text.y = element_text(size = 6,angle = 0,face = "bold"))

Top5Plot

ggsave("../../output/WGCNA/GO_analysis/Top10_BP_by_mod/Magenta_allterms_top5parent_filtp01.pdf", Top5Plot, width = 5, height = 5, units = c("in"), dpi=300, limitsize=FALSE)
```


#### Lightcyan

```{r}
ID.vector <- geneInfo %>%
  filter(moduleColor %in% c("lightcyan")) %>%
  pull(gene_id)

length(ID.vector) #65

##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% rename(GOs = "GO.terms")

dim(GO.terms) #9012    2
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ";") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2

##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 

#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#Filtering for p < 0.05
GO_05_lightcyan <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue) %>% mutate(Module = "Lightcyan")

go_results_BP <- GO_05_lightcyan %>%
      filter(ontology=="BP")%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)

dim(go_results_BP)
```

### Combining all the individual module lists for Calcification - Down

```{r}
GO_05_DOWN <- rbind(GO_05_turquoise,GO_05_magenta,GO_05_lightcyan)

length(GO_05_DOWN$GOterm) #1072 enriched GO terms
length(unique(GO_05_DOWN$GOterm)) #1070 enriched GO terms

# Collapse GO terms that are duplicated accross modules - sort by GO term and p-value, then keep the row with the lowest p-value, collapsing the module names so you know which modules were represented

# Identify and collapse duplicates
GO_05_DOWN <- GO_05_DOWN %>%
  arrange(GOterm, over_represented_pvalue) %>%
  group_by(GOterm) %>% mutate(Module = paste(unique(Module), collapse = ";")) %>%
  group_by(GOterm) %>%
  slice_min(order_by = over_represented_pvalue, n = 1) %>%
  ungroup()

#Write file of results 
write.csv(GO_05_DOWN, file = "../../output/WGCNA/GO_analysis/goseq_pattern_calcification_down_bymod.csv")
```

