---
title: "R Notebook"
output:
  pdf_document: default
  html_notebook: default
---
```{r}
rm(list=ls())
```

```{r load_libraries, include = TRUE}
# load libraries - notes show the install command needed to install (pre installed)
library(goseq)#
library(dplyr)
library(forcats)
library(ggplot2)
library(gridExtra)
library(tidyr)
library(grDevices)
library(reshape2)
library(Rmisc)
library(ggpubr)
library(tibble)
#library(hrbrthemes)#
library(gridExtra)
library(tidyr)
library(zoo)
library(ComplexHeatmap)
library(circlize)
library(GSEABase)#
library(data.table)
library(stringr)
library(GenomicRanges)
library(rrvgo)
```

#Load in data
```{r}
library(rtracklayer)
gff<-rtracklayer::import("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/Pocillopora_acuta_HIv2.genes_fixed.gff3")
gff<-as.data.frame(gff)
dim(gff) # 478988     9
names(gff) 
gff
transcripts <- subset(gff, type == "transcript")
transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information
transcript_lengths <- width(transcripts_gr) #isolate length of each gene
seqnames<-transcripts_gr$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame
transcripts
```
```{r}
kegg<-read.delim("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/Pocillopora_acuta_HIv2.genes.KEGG_results (1).txt.gz")
kegg<-as.data.frame(kegg)
kegg<- plyr::rename(kegg, c("Pocillopora_acuta_HIv2___RNAseq.g24143.t1a"="KEGG_new"))
kegg <- cbind(rownames(kegg),kegg)
rownames(kegg) <- NULL
colnames(kegg) <- c(names(kegg)) #to not write all the column names
colnames(kegg)[1] <- "gene_id" 
names(kegg)
kegg
```


```{r}
eggnog<-read.delim("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/Pocillopora_acuta_HIv2.genes.EggNog_results.txt")#this file contains all of the go terms, descriptions, kegg, etc
eggnog<- plyr::rename(eggnog, c("X.query"="gene_id"))
eggnog
```

```{r}
gogene<- merge(transcripts, eggnog, by=c("gene_id"), all=T)
gogene<- merge(gogene, kegg, by=c("gene_id"), all=T)
gogene
```

```{r}
geneInfo<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/WGCNA_ModulMembership.csv") #this file was generated from the WGCNA analyses and contains the modules of interest
geneInfo<- plyr::rename(geneInfo, c("X"="gene_id"))
dim(geneInfo)
geneInfo # there are 9012 genes in our gene info file
```
```{r}
geneInfo<- merge(gogene, geneInfo, by=c("gene_id"), all=T)
geneInfo
```



```{r}
unique(geneInfo$moduleColor)
```

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
geneInfo
```

#Run the GOSeq function by module color to test for GO term enrichment. Due to high number of enriched GO terms, I am using an adjusted p-value threshold of <0.001 and using `rrvgo` package to reduce redundancy in list of GO terms.      
##Calcification

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor==c("blue","pink","magenta","lightcyan","purple","brown"))%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor==c("blue","pink","magenta","lightcyan","purple","brown"))%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_calcification.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_calcification.csv")
go_results
```

```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```

```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```

```{r}
write.csv(go_results, "goseq_pattern_calcification_filtered.csv")
```


```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_calcification <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_calcification
```
###Count number of GOterms by ParentTerm for the upregulation of calcification
```{r}
library(dplyr)

result <- go_results %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Number.of.terms = n_distinct(term))%>%
  mutate(Calcification.direction = "Up")

print(result)
```

##Calcification - down

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor==c("black","red"))%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor==c("black","red"))%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_calcification_down.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification_down","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_calcification_down.csv")
go_results
```

```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```

```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```

```{r}
write.csv(go_results, "goseq_pattern_calcification_down_filtered.csv")
```


```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_calcification_down <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_calcification_down
```

###Count number of GOterms by ParentTerm for the downregulation of calcification
```{r}
library(dplyr)

result_down <- go_results %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Number.of.terms = n_distinct(term))%>%
  mutate(Calcification.direction = "Down")

print(result_down)
```

###Merge the frequency of GOterms for both up- and down-regulation of calcification
```{r}
cal_freq_terms<- merge(result,result_down, by=c("ParentTerm","Number.of.terms","Calcification.direction"),all=T)
cal_freq_terms
```

```{r}
cal_biomin_terms <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/Biomineralizationgo terms.csv")
cal_biomin_terms
```
```{r}
cal_up_terms <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_pattern_calcification_filtered.csv")
cal_up_terms<- cal_up_terms %>%
  mutate(Factor = "Up")
cal_up_terms
```

```{r}
cal_down_terms <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_pattern_calcification_down_filtered.csv")
cal_down_terms<-cal_down_terms %>%
  mutate(Factor = "Down")
cal_down_terms
```
```{r}
colnames(cal_biomin_terms)
```
###Merge biomineralization, up and down-regulation of calcification GOterms
```{r}
all_terms<- merge(cal_up_terms,cal_down_terms, by=c("Factor","GOterm","X.1","X","GOterm","over_represented_pvalue","under_represented_pvalue","numDEInCat","numInCat","term","ontology","bh_adjust","ParentTerm"),all=T)
all_terms<- merge(all_terms,cal_biomin_terms, by=c("Factor","GOterm","X.1","X","GOterm","over_represented_pvalue","under_represented_pvalue","numDEInCat","numInCat","term","ontology","bh_adjust","ParentTerm"),all=T)
all_terms$GOterm<-as.factor(all_terms$GOterm)
all_terms
```
```{r}
goterms_shared <- all_terms %>%
  group_by(GOterm) %>%
  dplyr::summarise(
    ParentTerm = paste(unique(ParentTerm), collapse = ", "),
    Factor = paste(unique(Factor), collapse = ", ")
  )
goterms_shared
```
```{r}
#write.csv(goterms_shared, "Merged GOterms by factor and ParentTerm.csv")
```

```{r}
result_unique <- goterms_shared %>%
  group_by(ParentTerm,Factor) %>%
  dplyr::summarise(SharedGOterms = n_distinct(GOterm))%>%
  arrange(-SharedGOterms)
result_unique
```
```{r}
result_filtered_up<- result_unique %>%
  dplyr::filter(Factor=="Biomin, Down, Up" | Factor=="Biomin, Up")
 #dplyr::filter(SharedGOterms>=5) 
result_filtered_up
```

```{r}
result_filtered_down<- result_unique %>%
  dplyr::filter(Factor=="Biomin, Down, Up" | Factor=="Biomin, Down")
 #dplyr::filter(SharedGOterms>=5) 
result_filtered_down
```

###Calculate the percentage of shared GOterms upregulation of calcification
```{r}
merged_up <- result_filtered_up %>%
  tidyr::separate_rows(ParentTerm, sep = ", ") %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Sum_of_SharedGOterms = sum(SharedGOterms, na.rm = TRUE))%>%
  left_join(result, by = "ParentTerm") %>%
  mutate(Proportion.of.shared.GO.terms.with.biomineralization.genes = Sum_of_SharedGOterms / Number.of.terms)%>%
  mutate(Percentage.of.shared.GO.terms.with.biomineralization.genes = Proportion.of.shared.GO.terms.with.biomineralization.genes * 100)
merged_up_clean<- na.omit(merged_up)
merged_up_clean
```
###Calculate the percentage of shared GOterms downregulation of calcification
```{r}
merged_down <- result_filtered_down %>%
  tidyr::separate_rows(ParentTerm, sep = ", ") %>%
  dplyr::group_by(ParentTerm) %>%
  dplyr::summarize(Sum_of_SharedGOterms = sum(SharedGOterms, na.rm = TRUE))%>%
  left_join(result_down, by = "ParentTerm") %>%
  mutate(Proportion.of.shared.GO.terms.with.biomineralization.genes = Sum_of_SharedGOterms / Number.of.terms)%>%
  mutate(Percentage.of.shared.GO.terms.with.biomineralization.genes = Proportion.of.shared.GO.terms.with.biomineralization.genes * 100)
merged_down_clean<- na.omit(merged_down)
merged_down_clean
```

###Frequency >10 up
```{r}
cal_freq_terms_filtered_up <- merged_up_clean %>%
  filter(Number.of.terms>=10) %>%
  filter(Calcification.direction=="Up")
cal_freq_terms_filtered_up
```
####Figure
```{r}
#counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
#counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
freq_fig_up<-ggplot(cal_freq_terms_filtered_up, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,50))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_up
```
###Frequency >10 down
```{r}
cal_freq_terms_filtered_down <- merged_down_clean %>%
  filter(Number.of.terms>=10)
  #filter(Calcification.direction=="Down")
cal_freq_terms_filtered_down
```
####Figure
```{r}
freq_fig_down<-ggplot(cal_freq_terms_filtered_down, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  #geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,50))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#d1e5f0","#92c5de","#4393c3","#2166ac"), na.value = "grey98",limits = c(0, 100))+ 
  #scale_fill_gradientn(colours=c("#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(10, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_down
```
```{r}
compare_figs<-cowplot::plot_grid(freq_fig_up, freq_fig_down, nrow=2, align="v")
compare_figs
```
###All terms up
```{r}
cal_freq_terms_filtered_up_all <- merged_up_clean %>%
  filter(Calcification.direction=="Up")
cal_freq_terms_filtered_up_all
```
####Figure
```{r}
#counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
#counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
freq_fig_up<-ggplot(cal_freq_terms_filtered_up_all, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,50))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_up
```

###All terms down
```{r}
cal_freq_terms_filtered_down_all <- merged_down_clean %>%
  filter(Calcification.direction=="Down")
cal_freq_terms_filtered_down_all
```
####Figure
```{r}
freq_fig_down<-ggplot(cal_freq_terms_filtered_down_all, aes(y=Number.of.terms,x=reorder(ParentTerm, Number.of.terms), fill=Percentage.of.shared.GO.terms.with.biomineralization.genes))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=Number.of.terms)) +
  #geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,50))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_gradientn(colours=c("white","#d1e5f0","#92c5de","#4393c3","#2166ac"), na.value = "grey98",limits = c(0, 100))+ 
  #scale_fill_gradientn(colours=c("#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(10, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig_down
```
```{r}
compare_figs<-cowplot::plot_grid(freq_fig_up, freq_fig_down, ncol=2, align="h")
compare_figs
```
##Black

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="black")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="black")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_black.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_black.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_black <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_black
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="cation transport"])
length(colnames(go_results)[go_results$ParentTerm=="inorganic ion homeostasis"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cellular response to stress"])
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="black")%>%
  dplyr::select(gene_id, KEGG_new)
KO.terms
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_black.csv")
    
```

```{r}
ko_results_black<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_black.csv") 
ko_results_black
```
```{r}
ko_results_black<-ko_results_black%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_black
```

```{r}
#plot significantly enriched KO terms 
KO.plot.black <-  ggplot(ko_results_black, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.black
```
###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="black")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_black.csv")
    
```

```{r}
ko_results_black<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_black.csv") 
ko_results_black
```
```{r}
ko_results_black<-ko_results_black%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_black
```

```{r}
#plot significantly enriched KO terms 
KO.plot.black <-  ggplot(ko_results_black, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.black
```

##Blue

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="blue")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="blue")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.0001
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_blue.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis", "regulation", "developmental", "localization", "multicellular","response to stimulus")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_blue.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_blue <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 20, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_blue
```

```{r}
length(colnames(go_results)[go_results$ParentTerm=="organic substance transport"])
length(colnames(go_results)[go_results$ParentTerm=="response to stress"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cell death"])
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="blue")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_blue.csv")
    
```

```{r}
ko_results_blue<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_blue.csv") 
ko_results_blue
```
```{r}
ko_results_blue<-ko_results_blue%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_blue
```

```{r}
#plot significantly enriched KO terms 
KO.plot.blue <-  ggplot(ko_results_blue, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.blue
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="blue")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_blue.csv")
    
```

```{r}
ko_results_blue<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_blue.csv") 
ko_results_blue
```
```{r}
ko_results_blue<-ko_results_blue%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_blue
```

```{r}
#plot significantly enriched KO terms 
KO.plot.blue <-  ggplot(ko_results_blue, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "blue"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.blue
```
##Brown

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="brown")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="brown")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_brown.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_brown.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_brown <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_brown
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="organic substance transport"])
length(colnames(go_results)[go_results$ParentTerm=="response to stress"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cell death"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cell communication"])
```


###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="brown")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_brown.csv")
    
```

```{r}
ko_results_brown<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_brown.csv") 
ko_results_brown
```
```{r}
ko_results_brown<-ko_results_brown%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_brown
```

```{r}
#plot significantly enriched KO terms 
KO.plot.brown <-  ggplot(ko_results_brown, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "brown"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.brown
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="brown")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_brown.csv")
    
```

```{r}
ko_results_brown<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_brown.csv") 
ko_results_brown
```
```{r}
ko_results_brown<-ko_results_brown%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_brown
```

```{r}
#plot significantly enriched KO terms 
KO.plot.brown <-  ggplot(ko_results_brown, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "brown"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.brown
```

##Cyan

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="cyan")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="cyan")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_cyan.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_cyan.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_cyan <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_cyan
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="regulation of neuron death"])
length(colnames(go_results)[go_results$ParentTerm=="glucose homeostasis"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of transmembrane transport"])
```


###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="cyan")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_cyan.csv")
    
```

```{r}
ko_results_cyan<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_cyan.csv") 
ko_results_cyan
```
```{r}
ko_results_cyan<-ko_results_cyan%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_cyan
```

```{r}
#plot significantly enriched KO terms 
KO.plot.cyan <-  ggplot(ko_results_cyan, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "cyan"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.cyan
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="cyan")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_cyan.csv")
    
```

```{r}
ko_results_cyan<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_cyan.csv") 
ko_results_cyan
```
```{r}
ko_results_cyan<-ko_results_cyan%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_cyan
```

```{r}
#plot significantly enriched KO terms 
KO.plot.cyan <-  ggplot(ko_results_cyan, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "cyan"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.cyan
```
##Green

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="green")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="green")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_green.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_green.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_green <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_green
```
###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="green")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_green.csv")
    
```

```{r}
ko_results_green<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_green.csv") 
ko_results_green
```
```{r}
ko_results_green<-ko_results_green%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_green
```

```{r}
#plot significantly enriched KO terms 
KO.plot.green <-  ggplot(ko_results_green, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "green"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.green
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="green")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_green.csv")
    
```

```{r}
ko_results_green<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_green.csv") 
ko_results_green
```
```{r}
ko_results_green<-ko_results_green%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_green
```

```{r}
#plot significantly enriched KO terms 
KO.plot.green <-  ggplot(ko_results_green, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "green"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.green
```

##Greenyellow

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="greenyellow")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="greenyellow")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_greenyellow.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_greenyellow.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_greenyellow <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_greenyellow
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="regulation of cell death"])
length(colnames(go_results)[go_results$ParentTerm=="positive regulation of cell communication"])
length(colnames(go_results)[go_results$ParentTerm=="response to stress"])
```


###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="greenyellow")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_greenyellow.csv")
    
```

```{r}
ko_results_greenyellow<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_greenyellow.csv") 
ko_results_greenyellow
```
```{r}
ko_results_greenyellow<-ko_results_greenyellow%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_greenyellow
```

```{r}
#plot significantly enriched KO terms 
KO.plot.greenyellow <-  ggplot(ko_results_greenyellow, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "greenyellow"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.greenyellow
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="greenyellow")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_greenyellow.csv")
    
```

```{r}
ko_results_greenyellow<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_greenyellow.csv") 
ko_results_greenyellow
```
```{r}
ko_results_greenyellow<-ko_results_greenyellow%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_greenyellow
```

```{r}
#plot significantly enriched KO terms 
KO.plot.greenyellow <-  ggplot(ko_results_greenyellow, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "greenyellow"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.greenyellow
```
##Lightcyan

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="lightcyan")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="lightcyan")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_lightcyan.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_lightcyan.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_lightcyan <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_lightcyan
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="response to stress"])
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="lightcyan")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_lightcyan.csv")
    
```

```{r}
ko_results_lightcyan<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_lightcyan.csv") 
ko_results_lightcyan
```
```{r}
ko_results_lightcyan<-ko_results_lightcyan%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_lightcyan
```

```{r}
#plot significantly enriched KO terms 
KO.plot.lightcyan <-  ggplot(ko_results_lightcyan, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "lightcyan"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.lightcyan
```
###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="lightcyan")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_lightcyan.csv")
    
```

```{r}
ko_results_lightcyan<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_lightcyan.csv") 
ko_results_lightcyan
```
```{r}
ko_results_lightcyan<-ko_results_lightcyan%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_lightcyan
```

```{r}
#plot significantly enriched KO terms 
KO.plot.lightcyan <-  ggplot(ko_results_lightcyan, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "lightcyan"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.lightcyan
```
##Magenta

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="magenta")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="magenta")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_magenta.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_magenta.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_magenta <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_magenta
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="cell death"])
length(colnames(go_results)[go_results$ParentTerm=="cellular ion homeostasis"])
length(colnames(go_results)[go_results$ParentTerm=="ion transport"])
```


###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="magenta")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_magenta.csv")
    
```

```{r}
ko_results_magenta<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_magenta.csv") 
ko_results_magenta
```
```{r}
ko_results_magenta<-ko_results_magenta%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_magenta
```

```{r}
#plot significantly enriched KO terms 
KO.plot.magenta <-  ggplot(ko_results_magenta, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "magenta"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.magenta
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="magenta")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_magenta.csv")
    
```

```{r}
ko_results_magenta<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_magenta.csv") 
ko_results_magenta
```
```{r}
ko_results_magenta<-ko_results_magenta%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_magenta
```

```{r}
#plot significantly enriched KO terms 
KO.plot.magenta <-  ggplot(ko_results_magenta, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "magenta"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.magenta
```
##Midnightblue

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="midnightblue")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="midnightblue")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_midnightblue.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_midnightblue.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_midnightblue <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_midnightblue
```

```{r}
length(colnames(go_results)[go_results$ParentTerm=="response to stress"])
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="midnightblue")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_midnightblue.csv")
    
```

```{r}
ko_results_midnightblue<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_midnightblue.csv") 
ko_results_midnightblue
```
```{r}
ko_results_midnightblue<-ko_results_midnightblue%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_midnightblue
```

```{r}
#plot significantly enriched KO terms 
KO.plot.midnightblue <-  ggplot(ko_results_midnightblue, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "midnightblue"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.midnightblue
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="midnightblue")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_midnightblue.csv")
    
```

```{r}
ko_results_midnightblue<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_midnightblue.csv") 
ko_results_midnightblue
```
```{r}
ko_results_midnightblue<-ko_results_midnightblue%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_midnightblue
```

```{r}
#plot significantly enriched KO terms 
KO.plot.midnightblue <-  ggplot(ko_results_midnightblue, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "midnightblue"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.midnightblue
```
##Pink

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="pink")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="pink")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_pink.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_pink.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_pink <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_pink
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="cell death"])
length(colnames(go_results)[go_results$ParentTerm=="chemical homeostasis"])
length(colnames(go_results)[go_results$ParentTerm=="ion transmembrane transport"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cellular response to stress"])
```


###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="pink")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_pink.csv")
    
```

```{r}
ko_results_pink<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_pink.csv") 
ko_results_pink
```
```{r}
ko_results_pink<-ko_results_pink%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_pink
```

```{r}
#plot significantly enriched KO terms 
KO.plot.pink <-  ggplot(ko_results_pink, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "pink"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.pink
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="pink")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_pink.csv")
    
```

```{r}
ko_results_pink<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_pink.csv") 
ko_results_pink
```
```{r}
ko_results_pink<-ko_results_pink%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_pink
```

```{r}
#plot significantly enriched KO terms 
KO.plot.pink <-  ggplot(ko_results_pink, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "pink"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.pink
```
##Purple

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="purple")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="purple")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.00001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_purple.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_purple.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>100)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_purple <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_purple
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="cellular cation homeostasis"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cell death"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of response to stress"])
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="purple")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_purple.csv")
    
```

```{r}
ko_results_purple<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_purple.csv") 
ko_results_purple
```
```{r}
ko_results_purple<-ko_results_purple%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_purple
```

```{r}
#plot significantly enriched KO terms 
KO.plot.purple <-  ggplot(ko_results_purple, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "purple"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.purple
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="purple")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_purple.csv")
    
```

```{r}
ko_results_purple<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_purple.csv") 
ko_results_purple
```
```{r}
ko_results_purple<-ko_results_purple%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_purple
```

```{r}
#plot significantly enriched KO terms 
KO.plot.purple <-  ggplot(ko_results_purple, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "purple"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.purple
```
##Red

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="red")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="red")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_red.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_red.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_red <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_red
```
```{r}
length(colnames(go_results)[go_results$ParentTerm=="regulation of neuron death"])
length(colnames(go_results)[go_results$ParentTerm=="cellular calcium ion homeostasis"])
length(colnames(go_results)[go_results$ParentTerm=="ion transmembrane transport"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of metal ion transport"])
length(colnames(go_results)[go_results$ParentTerm=="receptor protein tyrosine kinase signaling"])
```


###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="red")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_red.csv")
    
```

```{r}
ko_results_red<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_red.csv") 
ko_results_red
```
```{r}
ko_results_red<-ko_results_red%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_red
```

```{r}
#plot significantly enriched KO terms 
KO.plot.red <-  ggplot(ko_results_red, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "red"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.red
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="red")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_red.csv")
    
```

```{r}
ko_results_red<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_red.csv") 
ko_results_red
```
```{r}
ko_results_red<-ko_results_red%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_red
```

```{r}
#plot significantly enriched KO terms 
KO.plot.red <-  ggplot(ko_results_red, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "red"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.red
```

##Salmon

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="salmon")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="salmon")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_salmon.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_salmon.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
go_results
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_salmon <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_salmon
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="salmon")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_salmon.csv")
    
```

```{r}
ko_results_salmon<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_salmon.csv") 
ko_results_salmon
```
```{r}
ko_results_salmon<-ko_results_salmon%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_salmon
```

```{r}
#plot significantly enriched KO terms 
KO.plot.salmon <-  ggplot(ko_results_salmon, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "salmon"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.salmon
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="salmon")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_salmon.csv")
    
```

```{r}
ko_results_salmon<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_salmon.csv") 
ko_results_salmon
```
```{r}
ko_results_salmon<-ko_results_salmon%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_salmon
```

```{r}
#plot significantly enriched KO terms 
KO.plot.salmon <-  ggplot(ko_results_salmon, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "salmon"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.salmon
```
##Tan

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="tan")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="tan")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_tan.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_tan.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_tan <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_tan
```
###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="tan")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_tan.csv")
    
```

```{r}
ko_results_tan<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_tan.csv") 
ko_results_tan
```

```{r}
ko_results_tan<-ko_results_tan%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_tan
```

```{r}
#plot significantly enriched KO terms 
KO.plot.tan <-  ggplot(ko_results_tan, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "tan"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.tan
```


###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="tan")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_tan.csv")
    
```

```{r}
ko_results_tan<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_tan.csv") 
ko_results_tan
```

```{r}
ko_results_tan<-ko_results_tan%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_tan
```

```{r}
#plot significantly enriched KO terms 
KO.plot.tan <-  ggplot(ko_results_tan, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "tan"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.tan
```


##Turquoise

```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)
```

```{r}
### Generate length vector for all genes 
LENGTH.vector <- as.integer(geneInfo$Length)
```

```{r}
### Generate vector with names in just the module we are analyzing
ID.vector <- geneInfo%>%
  filter(moduleColor=="turquoise")%>%
  #get_rows(.data[[module]]))%>%
  pull(gene_id)
##Get a list of GO Terms for each module
GO.terms <- geneInfo%>%
  filter(moduleColor=="turquoise")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,gene_id)
GO.terms
```
```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("gene", "GO.terms")
GO.terms<-split2
```


```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        dplyr::filter(bh_adjust<0.0001) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_turquoise.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_turquoise.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```
```{r}
simMatrix
```


```{r}
 #calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```
```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_turquoise <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_turquoise
```

```{r}
length(colnames(go_results)[go_results$ParentTerm=="organic substance transport"])
length(colnames(go_results)[go_results$ParentTerm=="regulation of cell death"])
length(colnames(go_results)[go_results$ParentTerm=="response to stress"])
```

###KEGG KO
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="turquoise")%>%
  dplyr::select(gene_id, KEGG_new)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_turquoise.csv")
    
```

```{r}
ko_results_turquoise<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_turquoise.csv") 
ko_results_turquoise
```
```{r}
ko_results_turquoise<-ko_results_turquoise%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_turquoise
```

```{r}
#plot significantly enriched KO terms 
KO.plot.turquoise <-  ggplot(ko_results_turquoise, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "turquoise"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.turquoise
```

###KEGG Pathway
```{r}
 ##Get a list of GO Terms for each module
KO.terms <- geneInfo%>%
  filter(moduleColor=="turquoise")%>%
  dplyr::select(gene_id, KEGG_Pathway)
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
    gene.vector=as.integer(ALL.vector %in% ID.vector) 
    names(gene.vector)<-ALL.vector#set names
    #weight gene vector by bias for length of gene 
    pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
    
    #run goseq using Wallenius method for all categories of GO terms 
    KO.wall<-goseq(pwf, ID.vector, gene2cat=KO.terms, test.cats=c("KEGG"), method="Wallenius", use_genes_without_cat=TRUE)
    KO <- KO.wall[order(KO.wall$over_represented_pvalue),]
    colnames(KO)[1] <- "KEGGterm"
    
    #adjust p-values 
    KO$bh_adjust <-  p.adjust(KO$over_represented_pvalue, method="BH") #add adjusted p-values
  
    #Filtering for p < 0.01
    KO <- KO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., bh_adjust)
    
    #Write file of results 
    write.csv(KO, "goseq_KEGG_pattern_turquoise.csv")
    
```

```{r}
ko_results_turquoise<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/goseq_KEGG_pattern_turquoise.csv") 
ko_results_turquoise
```
```{r}
ko_results_turquoise<-ko_results_turquoise%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat >10)%>%
      arrange(., bh_adjust)
ko_results_turquoise
```

```{r}
#plot significantly enriched KO terms 
KO.plot.turquoise <-  ggplot(ko_results_turquoise, aes(x = X, y = KEGGterm)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse")+
    theme_bw() + 
  theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "turquoise"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank()) 
    #axis.ticks.x = element_blank(), 
    #axis.text.x = element_blank())
KO.plot.turquoise
```

```{r}
# GOslim
slim <- getOBOCollection("http://current.geneontology.org/ontology/subsets/goslim_generic.obo") #get GO database - # call goslim_generic.obo terms as 'slim'
```


#Frequency of GO terms within slim categories
```{r}
counts<-read.csv("/Users/imkri/Desktop/Penn postdoc/Heron experiment/Significant GO term counts by module .csv", strip.white=T)
counts$Module <-as.factor(counts$Module)
counts$Direction.of.flat.origin <-as.factor(counts$Direction.of.flat.origin)
counts$Slim.term <-as.factor(counts$Slim.term)
counts
```
```{r}
unique(counts$Module)
```

```{r}
counts$Direction.of.flat.origin<- factor(counts$Direction.of.flat.origin, levels =c("up","no pattern","down"))
counts$Module<- factor(counts$Module, levels=c("Blue","Brown","Greenyellow","Cyan","Pink","Magenta","Lightcyan","Midnight blue","Purple","Turquiose","Red","Black"))
test_fig<-ggplot(counts, aes(y=Number, x = Slim.term, color=Direction.of.flat.origin, fill=Direction.of.flat.origin,group=1))+
  facet_wrap(~Module, nrow = 1)+
  geom_point(size=5, alpha=0.9, pch=21,color="black")+
  geom_segment(aes(x=Slim.term, xend=Slim.term, y=0, yend=Number)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
 scale_y_continuous(expression(Significant~GO~term~counts),limits=c(0,35))+
  scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(0, 40))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_text(vjust=0.5, size=12),#making the axis title larger 
        axis.title.y = element_text(vjust=0.5, size=12),
        strip.text = element_text(size=12))#making the axis title larger 
test_fig
```
#Specific differentially expressed genes

```{r}
wgcna_counts_filtered<-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/Counts filtered.csv", strip.white=T)
wgcna_counts_filtered<- plyr::rename(wgcna_counts_filtered, c("X"="Gene"))
wgcna_counts_filtered
```
```{r}
colnames(wgcna_counts_filtered)
```

```{r}
library(tidyr)
wgcna_counts_filtered_long <- pivot_longer(wgcna_counts_filtered, cols=2:47, names_to = "Colony", values_to = "Counts")
wgcna_counts_filtered_long$Colony <- as.factor(wgcna_counts_filtered_long$Colony)
wgcna_counts_filtered_long
```
```{r}
wgcna_counts_filtered_long <- wgcna_counts_filtered_long %>% 
  separate(Colony, into = c('Origin', 'Colony.number'), sep = 2)
wgcna_counts_filtered_long
```

```{r}
library(stringr)
wgcna_counts_filtered_long$Colony <- as.numeric(str_extract(wgcna_counts_filtered_long$Colony.number, "[0-9]+"))
wgcna_counts_filtered_long<-wgcna_counts_filtered_long %>% 
   mutate(Treatment = trimws(str_remove(wgcna_counts_filtered_long$Colony.number, "(\\s+[A-Za-z]+)?[0-9-]+")))
wgcna_counts_filtered_long
```

```{r}
wgcna_counts_filtered_long$Origin <- as.factor(wgcna_counts_filtered_long$Origin)
wgcna_counts_filtered_long$Treatment <- as.factor(wgcna_counts_filtered_long$Treatment)
wgcna_counts_filtered_long
```

```{r}
wgcna_counts_filtered_long <- wgcna_counts_filtered_long %>%
  mutate(Treatment2 = ifelse(Treatment == "A" | Treatment == "B", "Variable",
               ifelse(Treatment == "C" | Treatment == "D", "Stable", NA)))
wgcna_counts_filtered_long$Treatment2 <- as.factor(wgcna_counts_filtered_long$Treatment2)
wgcna_counts_filtered_long
```
##SLC4A7
```{r}
wgcna_counts_filtered_long_SLC4A7<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7402.t1")
wgcna_counts_filtered_long_SLC4A7
```
```{r}
library(nlme)
SLC4A7.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_SLC4A7, na.action=na.exclude)
car::Anova(SLC4A7.lme, type=3)
tukey3<- emmeans(SLC4A7.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
SLC4A7_sum<-summarySE(wgcna_counts_filtered_long_SLC4A7, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
SLC4A7_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
SLC4A7_fig<-ggplot(data=SLC4A7_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_SLC4A7,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(SLC4A7~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
SLC4A7_fig
```
##SLC4A3 
```{r}
wgcna_counts_filtered_long_SLC4A3 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g27873.t1")
wgcna_counts_filtered_long_SLC4A3 
```
```{r}
SLC4A3.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_SLC4A3 , na.action=na.exclude)
car::Anova(SLC4A3.lme, type=3)
tukey3<- emmeans(SLC4A3.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
SLC4A3_sum<-summarySE(wgcna_counts_filtered_long_SLC4A3 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
SLC4A3_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
SLC4A3_fig<-ggplot(data=SLC4A3_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_SLC4A3 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(SLC4A3 ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
SLC4A3_fig
```

##NHE3
```{r}
wgcna_counts_filtered_long_NHE3<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g24868.t1")
wgcna_counts_filtered_long_NHE3
```
```{r}
NHE3.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_NHE3, na.action=na.exclude)
car::Anova(NHE3.lme, type=3)
tukey3<- emmeans(NHE3.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
NHE3_sum<-summarySE(wgcna_counts_filtered_long_NHE3, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
NHE3_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
NHE3_fig<-ggplot(data=NHE3_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_NHE3,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(NHE3~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
NHE3_fig
```

##CA1
```{r}
wgcna_counts_filtered_long_CA1<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g12304.t1")
wgcna_counts_filtered_long_CA1
```
```{r}
CA1.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_CA1, na.action=na.exclude)
car::Anova(CA1.lme, type=3)
tukey3<- emmeans(CA1.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA1_sum<-summarySE(wgcna_counts_filtered_long_CA1, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA1_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
CA1_fig<-ggplot(data=CA1_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_CA1,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA1~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA1_fig
```
##CA2
```{r}
wgcna_counts_filtered_long_CA2<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g13824.t1")
wgcna_counts_filtered_long_CA2
```
```{r}
CA2.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_CA2, na.action=na.exclude)
car::Anova(CA2.lme, type=3)
tukey3<- emmeans(CA2.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA2_sum<-summarySE(wgcna_counts_filtered_long_CA2, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA2_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
CA2_fig<-ggplot(data=CA2_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_CA2,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA2~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA2_fig
```
##VHA
```{r}
wgcna_counts_filtered_long_VHA<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g23064.t1")
wgcna_counts_filtered_long_VHA
```
```{r}
VHA.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_VHA, na.action=na.exclude)
car::Anova(VHA.lme, type=3)
tukey3<- emmeans(VHA.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
VHA_sum<-summarySE(wgcna_counts_filtered_long_VHA, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
VHA_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
VHA_fig<-ggplot(data=VHA_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_VHA,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(VHA~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
VHA_fig
```
##HSP90
```{r}
wgcna_counts_filtered_long_HSP90<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g6656.t1")
wgcna_counts_filtered_long_HSP90
```
```{r}
HSP90.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_HSP90, na.action=na.exclude)
car::Anova(HSP90.lme, type=3)
tukey3<- emmeans(HSP90.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
HSP90_sum<-summarySE(wgcna_counts_filtered_long_HSP90, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
HSP90_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
HSP90_fig<-ggplot(data=HSP90_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_HSP90,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(HSP90~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
HSP90_fig
```

##HIF1A 
```{r}
wgcna_counts_filtered_long_HIF1A <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g3039.t1")
wgcna_counts_filtered_long_HIF1A 
```
```{r}
HIF1A.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_HIF1A , na.action=na.exclude)
car::Anova(HIF1A.lme, type=3)
tukey3<- emmeans(HIF1A.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
HIF1A_sum<-summarySE(wgcna_counts_filtered_long_HIF1A , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
HIF1A_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
HIF1A_fig<-ggplot(data=HIF1A_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_HIF1A ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(HIF1A ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
HIF1A_fig
```


##HSP70 
```{r}
wgcna_counts_filtered_long_HSP70 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g10659.t1")
wgcna_counts_filtered_long_HSP70 
```
```{r}
HSP70.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_HSP70 , na.action=na.exclude)
car::Anova(HSP70.lme, type=3)
tukey3<- emmeans(HSP70.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
HSP70_sum<-summarySE(wgcna_counts_filtered_long_HSP70 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
HSP70_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
HSP70_fig<-ggplot(data=HSP70_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_HSP70 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(HSP70 ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
HSP70_fig
```

##PRKCD 
```{r}
wgcna_counts_filtered_long_PRKCD <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25259.t1")
wgcna_counts_filtered_long_PRKCD 
```
```{r}
PRKCD.lme <- lme(Counts~Origin*Treatment, random = ~1|Colony, data=wgcna_counts_filtered_long_PRKCD , na.action=na.exclude)
car::Anova(PRKCD.lme, type=3)
tukey3<- emmeans(PRKCD.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
PRKCD_sum<-summarySE(wgcna_counts_filtered_long_PRKCD , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
PRKCD_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
PRKCD_fig<-ggplot(data=PRKCD_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(PRKCD ~expression))+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
PRKCD_fig
```


```{r}
compare_figs<-cowplot::plot_grid(SLC4A7_fig, NHE3_fig, CA1_fig, CA2_fig, HSP70_fig,HIF1A_fig, nrow=3)
compare_figs
```
#Biomineralization toolkit present in modules

```{r}
biomin <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/Biomin_blast_Pocillopora_acuta_best_hit.csv")
biomin
```

```{r}
wgcnamod <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/WGCNA_ModulMembership.csv")
wgcnamod<- plyr::rename(wgcnamod, c("X"="Pocillopora_acuta_best_hit"))
wgcnamod
```
```{r}
biomin_mod<- merge(biomin, wgcnamod, by=c("Pocillopora_acuta_best_hit"), all=T)
biomin_mod
```
```{r}
biomin_mod <- biomin_mod %>%
  # recode empty strings "" by NAs
  na_if("") %>%
  # remove NAs
  na.omit
biomin_mod
```
```{r}
#write.csv(biomin_mod, "Biomineralization toolkit with modules.csv")
```


```{r}
unique(biomin_mod$Pocillopora_acuta_best_hit)
```

```{r}
plyr::count(biomin_mod, "moduleColor")
```

##GO terms

```{r}
ID.vector_biomin <- biomin_mod%>%
  #filter(moduleColor=="black")%>%
  #get_rows(.data[[module]]))%>%
  pull(Pocillopora_acuta_best_hit)
##Get a list of GO Terms for each module
GO.terms_biomin <- biomin_mod%>%
  #filter(moduleColor=="black")%>%
  #filter(get_rows(.data[[module]]))%>%
  dplyr::select(GO.terms,Pocillopora_acuta_best_hit)
GO.terms_biomin
```

```{r}
##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms_biomin$GO.terms), ",") 
split2 <- data.frame(v1 = rep.int(GO.terms_biomin$Pocillopora_acuta_best_hit, sapply(split, length)), v2 = unlist(split)) 
colnames(split2) <- c("Pocillopora_acuta_best_hit", "GO.terms")
GO.terms_biomin<-split2
GO.terms_biomin
```

```{r}
#GO.terms_biomin_sub <- GO.terms_biomin%>%
  #filter(Pocillopora_acuta_best_hit==c("Pocillopora_acuta_HIv2___RNAseq.g15280.t1","Pocillopora_acuta_HIv2___RNAseq.g7402.t1"))
#GO.terms_biomin_sub
```

```{r}
##Construct list of genes with 1 for genes in module and 0 for genes not in the module
gene.vector=as.integer(ALL.vector %in% ID.vector) 
names(gene.vector)<-ALL.vector#set names
#weight gene vector by bias for length of gene 
pwf<-nullp(gene.vector, ID.vector, bias.data=LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall<-goseq(pwf, ID.vector, gene2cat=GO.terms_biomin, test.cats=c("GO:BP", "GO:MF", "GO:CC"), method="Wallenius", use_genes_without_cat=TRUE)
GO <- GO.wall[order(GO.wall$over_represented_pvalue),]
colnames(GO)[1] <- "GOterm"
```
```{r}
GO
```

```{r}
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values
```

```{r}
#Filtering for p < 0.01
GO <- GO %>%
        #dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)
   
#Write file of results 
write.csv(GO, file = "/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_biomin.csv")
```

```{r}
module_vector<-c(levels(geneInfo$moduleColor))
ontologies<-c("BP", "MF")
```

```{r}
#add vector for terms of interest to reduce number of GO terms - NOT using this to look at individual modules for exploratory purposes
keywords<-c("metabolism", "carbon","bicarbonate", "apoptosis", "death", "symbiosis", "regulation of cell communication", "trans membrane transport", "transmembrane",  "organic substance transport", "inorganic substance transport","response to stress", "antioxidant", "calcification","biomineralization", "heat","transporters","proton transport","ion transport","acid-base", "homeostasis")
```

```{r}
go_results <-read.csv("/Users/imkri/Desktop/Penn Postdoc/Heron experiment/R figures/goseq_pattern_biomin.csv")
go_results
```
```{r}
 go_results<-go_results%>%
      filter(ontology=="BP")%>%
      filter(bh_adjust != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., bh_adjust)
go_results
```

```{r}
library(rrvgo)
#Reduce/collapse GO term set with the rrvgo package 
simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont="BP",
                                method="Rel")
```

```{r}
#calculate similarity 
scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
reducedTerms
```
```{r}
#keep only the goterms from the reduced list
go_results<-go_results%>%
  filter(GOterm %in% reducedTerms$go)
 #add in parent terms to list of go terms 
go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
go_results
```

```{r}
#write.csv(go_results, "Biomineralizationgo terms.csv")
```

```{r}
go_results<-go_results%>%
      filter(grepl(paste(keywords, collapse="|"), ParentTerm))
```

```{r}
#plot significantly enriched GO terms by Slim Category faceted by slim term 
 GO.plot_biomin <-  ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
GO.plot_biomin
```
##***thioredoxin reductase 1
```{r}
wgcna_counts_filtered_long_g10093 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g10093.t2")
wgcna_counts_filtered_long_g10093  
```
```{r}
g10093.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g10093 , na.action=na.exclude)
car::Anova(g10093.lme, type=3)
tukey3<- emmeans(g10093.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g10093_sum<-summarySE(wgcna_counts_filtered_long_g10093 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g10093_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
g10093_fig<-ggplot(data=g10093_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g10093,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Thioredoxin~reductase~1~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g10093_fig
```

##***Flagellar associated protein
```{r}
wgcna_counts_filtered_long_g11609 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g11609.t1")
wgcna_counts_filtered_long_g11609  
```
```{r}
g11609.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g11609 , na.action=na.exclude)
car::Anova(g11609.lme, type=3)
tukey3<- emmeans(g11609.lme, list(pairwise ~ Treatment2:Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
g11609_sum<-summarySE(wgcna_counts_filtered_long_g11609 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g11609_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
g11609_fig<-ggplot(data=g11609_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g11609 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Flagellar~associated~protein~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g11609_fig
```

##***Actin
```{r}
wgcna_counts_filtered_long_g14505 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g14505.t1")
wgcna_counts_filtered_long_g14505  
```
```{r}
g14505.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g14505 , na.action=na.exclude)
car::Anova(g14505.lme, type=3)
tukey3<- emmeans(g14505.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g14505_sum<-summarySE(wgcna_counts_filtered_long_g14505 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g14505_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g14505_fig<-ggplot(data=g14505_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g14505 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Actin~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g14505_fig
```

##***CA2
```{r}
wgcna_counts_filtered_long_CA2<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g13824.t1")
wgcna_counts_filtered_long_CA2
```
```{r}
CA2.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_CA2, na.action=na.exclude)
car::Anova(CA2.lme, type=3)
tukey3<- emmeans(CA2.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA2_sum<-summarySE(wgcna_counts_filtered_long_CA2, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA2_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
CA2_fig<-ggplot(data=CA2_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_CA2,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA2~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA2_fig
```

##Poly [ADP-ribose] polymerase 11
```{r}
wgcna_counts_filtered_long_g14663 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g14663.t1a")
wgcna_counts_filtered_long_g14663  
```
```{r}
g14663.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g14663 , na.action=na.exclude)
car::Anova(g14663.lme, type=3)
tukey3<- emmeans(g14663.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g14663_sum<-summarySE(wgcna_counts_filtered_long_g14663 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g14663_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g14663_fig<-ggplot(data=g14663_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Poly~"[ADP-ribose]"~polymerase~11~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g14663_fig
```
##***solute carrier family 4 member gamma
```{r}
wgcna_counts_filtered_long_g15280 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g15280.t1")
wgcna_counts_filtered_long_g15280  
```
```{r}
g15280.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g15280 , na.action=na.exclude)
car::Anova(g15280.lme, type=3)
tukey3<- emmeans(g15280.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g15280_sum<-summarySE(wgcna_counts_filtered_long_g15280 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g15280_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g15280_fig<-ggplot(data=g15280_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g15280 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Solute~carrier~family~4~member~gamma~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g15280_fig
```
##MAGUK p55 subfamily member 7-like
```{r}
wgcna_counts_filtered_long_g15517 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g15517.t1")
wgcna_counts_filtered_long_g15517  
```
```{r}
g15517.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g15517 , na.action=na.exclude)
car::Anova(g15517.lme, type=3)
tukey3<- emmeans(g15517.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g15517_sum<-summarySE(wgcna_counts_filtered_long_g15517 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g15517_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g15517_fig<-ggplot(data=g15517_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(MAGUK~p55~subfamily~member~7-like~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g15517_fig
```

##CARP1
```{r}
wgcna_counts_filtered_long_g16280 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g16280.t1")
wgcna_counts_filtered_long_g16280  
```
```{r}
g16280.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g16280 , na.action=na.exclude)
car::Anova(g16280.lme, type=3)
tukey3<- emmeans(g16280.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g16280_sum<-summarySE(wgcna_counts_filtered_long_g16280 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g16280_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g16280_fig<-ggplot(data=g16280_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CARP1~expression))+
  ggtitle(~tan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g16280_fig
```

##PHD finger protein 21A-like
```{r}
wgcna_counts_filtered_long_g1634 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g1634.t1")
wgcna_counts_filtered_long_g1634  
```
```{r}
g1634.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g1634 , na.action=na.exclude)
car::Anova(g1634.lme, type=3)
tukey3<- emmeans(g1634.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g1634_sum<-summarySE(wgcna_counts_filtered_long_g1634 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g1634_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g1634_fig<-ggplot(data=g1634_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(PHD~finger~protein~"21A"~like~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g1634_fig
```

##digestive cysteine proteinase 1-like
```{r}
wgcna_counts_filtered_long_g18103 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g18103.t1")
wgcna_counts_filtered_long_g18103  
```
```{r}
g18103.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g18103 , na.action=na.exclude)
car::Anova(g18103.lme, type=3)
tukey3<- emmeans(g18103.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g18103_sum<-summarySE(wgcna_counts_filtered_long_g18103 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g18103_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g18103_fig<-ggplot(data=g18103_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g18103 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(digestive~cysteine~proteinase~1-like~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g18103_fig
```

##.endothelin-converting enzyme 1-like isoform X2
```{r}
wgcna_counts_filtered_long_g19211 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g19211.t1")
wgcna_counts_filtered_long_g19211  
```
```{r}
g19211.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g19211 , na.action=na.exclude)
car::Anova(g19211.lme, type=3)
tukey3<- emmeans(g19211.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g19211_sum<-summarySE(wgcna_counts_filtered_long_g19211 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g19211_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g19211_fig<-ggplot(data=g19211_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(endothelin-converting~enzyme~1-like~isoform~X2~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g19211_fig
```

##chymotrypsin-like elastase family member 1 
```{r}
wgcna_counts_filtered_long_g19288 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g19288.t1")
wgcna_counts_filtered_long_g19288  
```
```{r}
g19288.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g19288 , na.action=na.exclude)
car::Anova(g19288.lme, type=3)
tukey3<- emmeans(g19288.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g19288_sum<-summarySE(wgcna_counts_filtered_long_g19288 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g19288_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g19288_fig<-ggplot(data=g19288_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(chymotrypsin-like~elastase~family~member~1~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g19288_fig
```

##CUB and peptidase domain-containing protein 2-like 
```{r}
wgcna_counts_filtered_long_g21338 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g21338.t1")
wgcna_counts_filtered_long_g21338  
```
```{r}
g21338.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g21338 , na.action=na.exclude)
car::Anova(g21338.lme, type=3)
tukey3<- emmeans(g21338.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g21338_sum<-summarySE(wgcna_counts_filtered_long_g21338 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21338_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g21338_fig<-ggplot(data=g21338_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CUB~and~peptidase~domain-containing~protein~2-like~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g21338_fig
```

##L-type calcium channel alpha-1 subunit
```{r}
wgcna_counts_filtered_long_g21501 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g21501.t1")
wgcna_counts_filtered_long_g21501  
```
```{r}
g21501.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g21501 , na.action=na.exclude)
car::Anova(g21501.lme, type=3)
tukey3<- emmeans(g21501.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g21501_sum<-summarySE(wgcna_counts_filtered_long_g21501 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21501_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g21501_fig<-ggplot(data=g21501_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(L-type~calcium~channel~alpha-1~subunit~expression))+
  ggtitle(~turquoise)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g21501_fig
```

##.Protocadherin
```{r}
wgcna_counts_filtered_long_g21501 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g22388.t1")
wgcna_counts_filtered_long_g21501  
```
```{r}
g21501.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g21501 , na.action=na.exclude)
car::Anova(g21501.lme, type=3)
tukey3<- emmeans(g21501.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g21501_sum<-summarySE(wgcna_counts_filtered_long_g21501 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g21501_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g21501_fig<-ggplot(data=g21501_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g21501 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Protocadherin~expression), limits=c(100,200))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g21501_fig
```

##Acropora yongei Na+/Ca2+ exchanger
```{r}
wgcna_counts_filtered_long_g24639 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g24639.t1")
wgcna_counts_filtered_long_g24639  
```
```{r}
g24639.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g24639 , na.action=na.exclude)
car::Anova(g24639.lme, type=3)
tukey3<- emmeans(g24639.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g24639_sum<-summarySE(wgcna_counts_filtered_long_g24639 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g24639_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g24639_fig<-ggplot(data=g24639_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Acropora~yongei~"Na+/Ca2+"~exchanger~expression))+
  ggtitle(~tan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g24639_fig
```
##**mammalian ependymin-related protein 1-like
```{r}
wgcna_counts_filtered_long_g25351 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25351.t1")
wgcna_counts_filtered_long_g25351  
```
```{r}
g25351.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g25351 , na.action=na.exclude)
car::Anova(g25351.lme, type=3)
tukey3<- emmeans(g25351.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g25351_sum<-summarySE(wgcna_counts_filtered_long_g25351 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g25351_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g25351_fig<-ggplot(data=g25351_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g25351,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(mammalian~ependymin-related~protein~1-like~expression), limits=c(2500,12500))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g25351_fig
```
##MAM and LDLr domain-containing protein
```{r}
wgcna_counts_filtered_long_g25935 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g25935.t1")
wgcna_counts_filtered_long_g25935  
```
```{r}
g25935.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g25935 , na.action=na.exclude)
car::Anova(g25935.lme, type=3)
tukey3<- emmeans(g25935.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g25935_sum<-summarySE(wgcna_counts_filtered_long_g25935 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g25935_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g25935_fig<-ggplot(data=g25935_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(MAM~and~LDLr~domain-containing~protein~expression))+
  ggtitle(~brown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g25935_fig
```

#SLIT-ROBO Rho GTPase-activating protein 1-like 
```{r}
wgcna_counts_filtered_long_g27376 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g27376.t1")
wgcna_counts_filtered_long_g27376  
```
```{r}
g27376.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27376 , na.action=na.exclude)
car::Anova(g27376.lme, type=3)
tukey3<- emmeans(g27376.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27376_sum<-summarySE(wgcna_counts_filtered_long_g27376 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27376_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g27376_fig<-ggplot(data=g27376_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(SLIT-ROBO~Rho~GTPase-activating~protein~1-like~expression))+
  ggtitle(~grey60)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27376_fig
```

##**Hephaestin-like protein - frontloaded
```{r}
wgcna_counts_filtered_long_g27566 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g27566.t1")
wgcna_counts_filtered_long_g27566  
```
```{r}
g27566.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27566 , na.action=na.exclude)
car::Anova(g27566.lme, type=3)
tukey3<- emmeans(g27566.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27566_sum<-summarySE(wgcna_counts_filtered_long_g27566 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27566_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g27566_fig<-ggplot(data=g27566_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g27566 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Hephaestin-like~protein~expression))+
  ggtitle(~brown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27566_fig
```

#plasma membrane calcium ATPase - frontloaded
```{r}
wgcna_counts_filtered_long_g27976 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g27976.t1")
wgcna_counts_filtered_long_g27976  
```
```{r}
g27976.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27976 , na.action=na.exclude)
car::Anova(g27976.lme, type=3)
tukey3<- emmeans(g27976.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27976_sum<-summarySE(wgcna_counts_filtered_long_g27976 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27976_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g27976_fig<-ggplot(data=g27976_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(plasma~membrane~calcium~ATPase~expression))+
  ggtitle(~brown)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27976_fig
```

#von Willebrand factor D and EGF domain-containing protein-like
```{r}
wgcna_counts_filtered_long_g28226 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g28226.t2")
wgcna_counts_filtered_long_g28226  
```
```{r}
g28226.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g28226 , na.action=na.exclude)
car::Anova(g28226.lme, type=3)
tukey3<- emmeans(g28226.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g28226_sum<-summarySE(wgcna_counts_filtered_long_g28226 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g28226_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g28226_fig<-ggplot(data=g28226_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g28226,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(von~Willebrand~factor~D~and~EGF~domain-containing~protein-like~expression), limits=c(0,120))+
  ggtitle(~magenta)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g28226_fig
```

#low-density lipoprotein receptor-related protein 8-like 
```{r}
wgcna_counts_filtered_long_g4085 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g4085.t1")
wgcna_counts_filtered_long_g4085  
```
```{r}
g4085.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g4085 , na.action=na.exclude)
car::Anova(g4085.lme, type=3)
tukey3<- emmeans(g4085.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g4085_sum<-summarySE(wgcna_counts_filtered_long_g4085 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g4085_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g4085_fig<-ggplot(data=g4085_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(low-density~lipoprotein~receptor-related~protein~8-like~expression))+
  ggtitle(~cyan)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g4085_fig
```

#**Cephalotoxin-like protein
```{r}
wgcna_counts_filtered_long_g5013 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g5013.t1")
wgcna_counts_filtered_long_g5013  
```
```{r}
g5013.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g5013 , na.action=na.exclude)
car::Anova(g5013.lme, type=3)
tukey3<- emmeans(g5013.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g5013_sum<-summarySE(wgcna_counts_filtered_long_g5013 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g5013_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g5013_fig<-ggplot(data=g5013_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g5013 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(cephalotoxin-like~expression))+
  ggtitle(~blue)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g5013_fig
```

#**sodium bicarbonate cotransporter 3-like isoform X - SLC4A7
```{r}
wgcna_counts_filtered_long_g7402 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7402.t1")
wgcna_counts_filtered_long_g7402  
```
```{r}
g7402.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g7402, na.action=na.exclude)
car::Anova(g7402.lme, type=3)
tukey3<- emmeans(g7402.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g7402_sum<-summarySE(wgcna_counts_filtered_long_g7402 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g7402_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g7402_fig<-ggplot(data=g7402_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g7402 ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(sodium~bicarbonate~cotransporter~3-like~isoform~X2~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g7402_fig
```

##*protein lingerer-like
```{r}
wgcna_counts_filtered_long_g7902 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___RNAseq.g7908.t1")
wgcna_counts_filtered_long_g7902  
```
```{r}
g7902.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g7902, na.action=na.exclude)
car::Anova(g7902.lme, type=3)
tukey3<- emmeans(g7902.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g7902_sum<-summarySE(wgcna_counts_filtered_long_g7902 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g7902_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g7902_fig<-ggplot(data=g7902_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_g7902,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(protein~lingerer-like~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g7902_fig
```

##***CA1
```{r}
wgcna_counts_filtered_long_CA1<- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g12304.t1")
wgcna_counts_filtered_long_CA1
```
```{r}
CA1.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_CA1, na.action=na.exclude)
car::Anova(CA1.lme, type=3)
tukey3<- emmeans(CA1.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```

```{r}
library(Rmisc)
CA1_sum<-summarySE(wgcna_counts_filtered_long_CA1, measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
CA1_sum
```
####Figure
```{r}
pd<- position_dodge(0.2)
CA1_fig<-ggplot(data=CA1_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  geom_point(data=wgcna_counts_filtered_long_CA1,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(CA1~expression))+
  ggtitle(~pink)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
CA1_fig
```

##Uncharacterized skeletal organic matrix protein-6 (USOMP6)
```{r}
wgcna_counts_filtered_long_g22622 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g22622.t1")
wgcna_counts_filtered_long_g22622  
```
```{r}
g22622.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g22622, na.action=na.exclude)
car::Anova(g22622.lme, type=3)
tukey3<- emmeans(g22622.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g22622_sum<-summarySE(wgcna_counts_filtered_long_g22622 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g22622_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g22622_fig<-ggplot(data=g22622_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Uncharacterized~skeletal~organic~matrix~protein-6~(USOMP6)~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g22622_fig
```

##band 3 anion transport protein-like
```{r}
wgcna_counts_filtered_long_g27873 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g27873.t1")
wgcna_counts_filtered_long_g27873  
```
```{r}
g27873.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g27873, na.action=na.exclude)
car::Anova(g27873.lme, type=3)
tukey3<- emmeans(g27873.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g27873_sum<-summarySE(wgcna_counts_filtered_long_g27873 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g27873_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g27873_fig<-ggplot(data=g27873_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(band~3~anion~transport~protein-like~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g27873_fig
```

##collagenase 3-like
```{r}
wgcna_counts_filtered_long_g5338 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g5338.t1")
wgcna_counts_filtered_long_g5338  
```
```{r}
g5338.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g5338, na.action=na.exclude)
car::Anova(g5338.lme, type=3)
tukey3<- emmeans(g5338.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g5338_sum<-summarySE(wgcna_counts_filtered_long_g5338 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g5338_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g5338_fig<-ggplot(data=g5338_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(collagenase~3-like~expression))+
  ggtitle(~green)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g5338_fig
```

##Protocadherin
```{r}
wgcna_counts_filtered_long_g6583 <- wgcna_counts_filtered_long %>%
  filter(Gene == "Pocillopora_acuta_HIv2___TS.g6583.t1")
wgcna_counts_filtered_long_g6583  
```
```{r}
g6583.lme <- lme(Counts~Origin*Treatment2, random = ~1|Colony, data=wgcna_counts_filtered_long_g6583, na.action=na.exclude)
car::Anova(g6583.lme, type=3)
tukey3<- emmeans(g6583.lme, list(pairwise ~ Origin), adjust = "tukey")
tukey3
```
```{r}
library(Rmisc)
g6583_sum<-summarySE(wgcna_counts_filtered_long_g6583 , measurevar='Counts', groupvars=c('Origin', 'Treatment2'), na.rm=TRUE, conf.interval = 0.95)
g6583_sum
```

####Figure
```{r}
pd<- position_dodge(0.2)
g6583_fig<-ggplot(data=g6583_sum, aes(y=Counts, x=Treatment2, color=Origin),group = interaction(Origin))+
  #geom_point(data=wgcna_counts_filtered_long_PRKCD ,aes(y=Counts, x=Treatment2, color=Origin), alpha=0.4, position = pd)+
  geom_line(aes(group = interaction(Origin), stat="identity"),position=position_dodge(0.2))+
  geom_point(size=3, stat="identity", position = pd)+
  geom_errorbar(aes(ymin=Counts-se, ymax=Counts+se), stat="identity",width=0.2, position = pd)+
  scale_fill_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_color_manual("Origin", values=c("RS"='paleturquoise3', "RF"= "indianred"))+
  scale_y_continuous(expression(Protocadherin~expression))+
  ggtitle(~black)+
  theme_classic()+
  theme(axis.text.x=element_text(vjust=0.5,size=12),#angling the labels on the x-axis
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),#telling it where to position our plot title
        panel.background= element_rect(fill=NA, color='black'),#this is making the black box around the graph
        #strip.background = element_blank(), 
        #strip.text = element_blank(),
        legend.title = element_text(vjust=0.5,size=12),
        legend.position="none",
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_text(size=12))#making the axis title larger 
g6583_fig
```
#Comparison of sig genes
```{r}
biomin_compare_figs<-cowplot::plot_grid(g10093_fig,CA2_fig,g25351_fig,g5013_fig,g15280_fig, CA1_fig,g7402_fig,g7902_fig,g27566_fig,g28226_fig,g14505_fig,g11609_fig,g10093_fig, nrow=4)
biomin_compare_figs
```