---
title: "DEG_Enrich after glmmSeq, Slope as Base Level"
author: "Zoe Dellaert"
date: "4/24/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is a Rmd file analyzing our raw count data by the glmmSeq package as described in the [vignette](https://cran.r-project.org/web/packages/glmmSeq/vignettes/glmmSeq.html) and [manual](https://cran.r-project.org/web/packages/glmmSeq/glmmSeq.pdf).

```{r}
sessionInfo() #provides list of loaded packages and version of R. I still have version 4.1 for now.
```

```{r install packages}
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('goseq') 
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed

library(dplyr)
library(ggplot2)
library(goseq)
library(rtracklayer)
library(rrvgo)
library(GO.db)
library(tidyr)
library(forcats)
library(simplifyEnrichment)
sessionInfo() #list of packages after library-ing these packages
```

```{r}
DEGs <- read.csv(file="../../../output/glmmseq/signif_genes_normcts.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

#NOTE! This is not a file only with differentially expressed genes, this contains all of the genes in our dataset but also contains p-value information and fold change information to help determine which genes are signficant DEGs based on our model in glmmSeq

rownames(DEGs) <- DEGs$Gene

dim(DEGs)

Origin_DEGs <- DEGs %>%  dplyr::filter(Origin < 0.05)

nrow(Origin_DEGs)
```

```{r}
genes <- rownames(DEGs)
```

## **Generate files for Functional Enrichment**  

Based off of [Ariana's script](https://github.com/AHuffmyer/EarlyLifeHistory_Energetics/blob/master/Mcap2020/Scripts/TagSeq/Genome_V3/1_WGCNA_Mcap_V3.Rmd)

Functional annotation file obtained from: http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.EggNog_results.txt.gz on April 3, 2023.  


```{r}
Pacuta.annot <- read.delim("../../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt") %>% dplyr::rename("query" = X.query)

dim(Pacuta.annot)
head(Pacuta.annot)
```

```{r}
genes2annot = match(genes, Pacuta.annot$query) #match genes in DEGs (all genes after filtering) to genes in annotation file

sum(is.na(genes2annot)) #number of genes without EggNog annotation
missing<-as.data.frame(genes[which(is.na(genes2annot))]) #dataframe of genes without EggNog annotation
head(missing)
```

2812/9011 genes without eggnog annotation

```{r}
names(Pacuta.annot)

geneInfo0 = data.frame(gene_id = genes, #add gene id
  Accession = Pacuta.annot$seed_ortholog[genes2annot], #add accession number
  Bitscore = Pacuta.annot$score[genes2annot], #add bitscore
  eValue = Pacuta.annot$evalue[genes2annot], #add e value
  Description = Pacuta.annot$Description[genes2annot], #add description of gene
  Annotation.GO.ID = Pacuta.annot$GOs[genes2annot], #add GO ID's
  q_Origin = DEGs$Origin, #add Origin adjusted p-value
  q_Treatment = DEGs$Treatment, #add Treatment adjusted p-value
  q_Interaction = DEGs$Treatment.Origin, #add Treatment:Origin adjusted p-value
  Stable_OriginFC = DEGs$Stable_OriginFC, #add fold change for Slope vs Flat in the stable treatment
  Variable_OriginFC = DEGs$Variable_OriginFC, #add fold change for Slope vs Flat in the variable treatment
  maxGroupFC = DEGs$maxGroupFC, #add max group fold change (was FC bigger in stable of variable treatment)
  col = DEGs$col) #add qualitative significance info

dim(geneInfo0)
head(geneInfo0,2)
```

Add KEGG annotation information.  Downloaded from: http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.KEGG_results.txt.gz on April 3, 2023.  

```{r}
kegg<-read.table("../../../data/Pocillopora_acuta_HIv2.genes.KEGG_results.txt", sep="", quote="", na.strings=c("","NA"), blank.lines.skip = FALSE, header=FALSE)

dim(kegg)
head(kegg)
```

Add KEGG annotations to each gene.  
```{r}
geneInfo0$KEGG <- kegg$V2[match(geneInfo0$gene_id, kegg$V1)]

sum(is.na(geneInfo0$KEGG)) #number of genes without KEGG annotation

missing_KEGG <- as.data.frame(genes[which(is.na(geneInfo0$KEGG))]) #dataframe of genes without EggNog annotation
head(missing_KEGG)
```

Order geneInfo0 by significance of Origin, q_Origin (adjusted p value)
```{r}
geneInfo <- geneInfo0[order(geneInfo0[, 'q_Origin']), ]

write.csv(geneInfo, file = "../../../output/glmmseq/geneInfo.csv") #gene info for reference/supplement
```


## Now onto Enrichment!
```{r}
#geneInfo<-read.csv("../../../output/glmmseq/geneInfo.csv")

dim(geneInfo)

```

Get gene length information.  
```{r}
#import file
gff <- rtracklayer::import("../../../data/Pocillopora_acuta_HIv2.genes_fixed.gff3") #if this doesn't work, restart R and try again 
transcripts <- subset(gff, type == "transcript") #keep only transcripts , not CDS or exons
transcript_lengths <- width(transcripts) #isolate length of each gene
seqnames<-transcripts$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame

geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)] #Add in length to geneInfo
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  Annotation.GO.ID column
```{r}
geneInfo$Annotation.GO.ID <- gsub(",", ";", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub('"', "", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub("-", NA, geneInfo$Annotation.GO.ID)
```


```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)

### Generate length vector for all genes
LENGTH.vector <- as.integer(geneInfo$Length)

### Generate vector with names of the 840 significant DEGs by Origin
ID.vector <- geneInfo %>%
  filter(q_Origin < 0.05) %>%
  pull(gene_id)

##Get a list of GO Terms for the all 9011 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, Annotation.GO.ID)

##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$Annotation.GO.ID), ";")
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split))
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms <- split2

dim(GO.terms)

##Construct list of genes with 1 for genes in that are significant for Origin and 0 for those that are not
gene.vector = as.integer(ALL.vector %in% ID.vector) #since we ordered geneInfo by q_Origin, this puts a "1" for the first 840 genes, which are the same genes in ID.vector

names(gene.vector) <- ALL.vector #set names

#weight gene vector by bias for length of gene
pwf <- nullp(gene.vector, ID.vector, bias.data = LENGTH.vector) 
```

```{r}
#how many go terms from the 840 genes

##Get a list of GO Terms for the 840 DE genes
GO.terms_DE <- geneInfo %>% filter(q_Origin < 0.05) %>%
  dplyr::select(gene_id, Annotation.GO.ID)

##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms_DE$Annotation.GO.ID), ";")
split2 <- data.frame(v1 = rep.int(GO.terms_DE$gene, sapply(split, length)), v2 = unlist(split))
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms_DE <- split2

dim(GO.terms_DE)
```


```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall <- goseq(pwf, ID.vector, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)

GO <- GO.wall[order(GO.wall$over_represented_pvalue),]

colnames(GO)[1] <- "GOterm"

#Write file of results 
write.csv(GO, "../../../output/glmmseq/GOSeq/GOseq_DEG_Origin.csv")

#Filtering for p < 0.05
GO_05 <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

#Filtering for p < 0.05
GO_001 <- GO %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)


dim(GO_05)
head(GO_05)

dim(GO_001)
GO_001
```


Plotting!

```{r}
ontologies<-c("BP", "CC", "MF")

for (category in ontologies) {
    
    go_results <- GO_05
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$over_represented_pvalue), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
    
    filename <-  paste0("../../../output/glmmseq/GOSeq/GOseq_DEG_Origin_filtered_P05_", category, ".csv")
    write.csv(go_results, filename)
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=over_represented_pvalue)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot)
  
ggsave(filename=paste0("../../../output/glmmseq/GOSeq/OriginDEGs_P05", "_", category, ".png"), plot=GO.plot, dpi=300, height=10, units="in", limitsize=FALSE)
}
```

Collapsing with simplifyEnrichment package

```{r}
library(simplifyEnrichment)
BP_terms <- GO_05 %>% dplyr::filter(ontology=="BP") %>% dplyr::select(GOterm) %>% unlist()

Mat <- GO_similarity(BP_terms, ont = "BP", db = "org.Ce.eg.db")

pdf(file = "../../../output/glmmseq/GOSeq/OriginDEGs_P05_GOSEM.pdf", width = 7, height = 5)
simplifyGO(Mat, word_cloud_grob_param = list(max_width = 50), method = "kmeans")
dev.off()
```

### Combining MF, CC, and BP into one plot and order by pvalue, based on [Jill's script](https://github.com/JillAshey/SedimentStress/blob/master/RAnalysis/pacuta/pacuta_GOSeq.Rmd)

```{r}
#ordered by p-value
GO_plot_all_pval <- GO_001 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=1, aes(colour = ontology)) +
  coord_flip() +
  #ylim(0,305) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("Number of Differentially Expressed Genes in GO Term") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes

GO_plot_all_pval

ggsave("../../../output/glmmseq/GOSeq/OriginDEGs_P001_orderedpval.pdf", GO_plot_all_pval)
```

```{r}
#ordered by p-value
GO_plot_all_pval <- GO_05 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=1, aes(colour = ontology)) +
  coord_flip() +
  #ylim(0,305) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("Number of Differentially Expressed Genes in GO Term") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes

GO_plot_all_pval

ggsave("../../../output/glmmseq/GOSeq/OriginDEGs_P05_orderedpval.pdf", GO_plot_all_pval, width = 12, height = 24, units = c("in"), dpi=300, limitsize=FALSE)
```

### KEGG enrichment
```{r}
##Get a list of KEGG Terms for all 9011 genes
KO.terms <- geneInfo %>%
  dplyr::select(gene_id, KEGG)


#run goseq using Wallenius method for all categories of KEGG terms

KO.wall <-
  goseq(
    pwf,
    ID.vector,
    gene2cat = KO.terms,
    test.cats = c("KEGG"),
    method = "Wallenius",
    use_genes_without_cat = TRUE
  )


KO <- KO.wall[order(KO.wall$over_represented_pvalue), ]
colnames(KO)[1] <- "KEGGterm"

#Filtering for p < 0.05
KO_05 <- KO %>%
  dplyr::filter(over_represented_pvalue < 0.05) %>%
  dplyr::arrange(., over_represented_pvalue)

head(KO_05)
dim(KO_05)

#Write file of results 
write.csv(KO, "../../../output/glmmseq/GOSeq/GOseq_KEGG_Origin.csv")
```

Over-represented KEGG terms, p = 0.05:
"K01052" "K01539" "K05658" "K05619" "K11789" "K01363" "K13348" "K01345" "K24436" "K17592" "K25493" "K24125" "K12825" "K08059" "K05929" "K19721" "K14480" "K03671" "K06711" "K13443" "K18405"


## Part 2: Frequency based graphs

```{r}
freq_fig <- ggplot(go_results, aes(y=numInCat,x=reorder(ParentTerm, numInCat)
                                   #, fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1
                                   ))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=numInCat)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,95))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig

ggsave(filename="../../../output/glmmseq/GOSeq/OriginDEGs_P05_BP_numInCat.png", plot=freq_fig, dpi=300, height=10, units="in", limitsize=FALSE)
```

```{r}
freq_fig <- ggplot(go_results, aes(y=numDEInCat,x=reorder(ParentTerm, numDEInCat)
                                   #, fill=Percentage.of.shared.GO.terms.with.biomineralization.genes,group=1
                                   ))+
  #facet_wrap(~Calcification.direction, nrow = 1)+
  geom_point(size=5, alpha=1, pch=21,color="black")+
  geom_segment(aes(x=ParentTerm, xend=ParentTerm, y=0, yend=numDEInCat)) +
  geom_hline(yintercept = 0, linetype="solid", color = 'black', size=0.5, show.legend = TRUE)+
  coord_flip()+
  scale_y_continuous(expression(GO~term~counts),limits=c(0,20))+
  #scale_color_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_manual("Direction.of.flat.origin",values= c("up"="#b2182b","no pattern"="grey","down" ="#67a9cf"))+
  #scale_fill_gradientn(colours=c("white","#fddbc7","#f4a582","#d6604d","#b2182b"), na.value = "grey98",limits = c(0, 100))+ 
 #scale_color_gradientn(colours=c("#b2182b","#fddbc7","white","#d1e5f0","#67a9cf", "#67a9cf", "#2166ac"), na.value = "grey98",limits = c(-0, 40))+ 
  theme_classic()+
   theme(axis.text.x=element_text(vjust=0.5, hjust=0.95,size=12),
        plot.title = element_text(margin = margin(t = 10, b = 10), hjust=0.5),
        panel.background= element_rect(fill=NA, color='black'),
        legend.title = element_blank(),
        axis.text.y = element_text(vjust=0.5, size=12), #making the axis text larger 
        axis.title.x = element_blank(),#making the axis title larger 
        axis.title.y = element_blank(),
        strip.text = element_text(size=12))#making the axis title larger 
freq_fig

ggsave(filename="../../../output/glmmseq/GOSeq/OriginDEGs_P05_BP_numDEInCat.png", plot=freq_fig, dpi=300, height=10, units="in", limitsize=FALSE)
```


## Part 3: Direction of FC

### What about the direction of FC: Genes significant by Origin and upregulated in Slope origin (positive FC) for both treatments (we should also look into the ones that are positive in one treatment but not the other)

```{r}
### Generate vector with names of the 346 significant DEGs by Origin that are upregulated in the Slope origin in both treatments
ID.vector_upSlope <- geneInfo %>%
  filter(q_Origin < 0.05) %>%
  filter(Stable_OriginFC > 0 & Variable_OriginFC > 0) %>%
  pull(gene_id)

##Construct list of genes with 1 for genes in that are significant for Origin and upregulated in Slope and 0 for those that are not

gene.vector_upSlope = as.integer(ALL.vector %in% ID.vector_upSlope) 

names(gene.vector_upSlope) <- ALL.vector #set names

#weight gene vector by bias for length of gene
pwf_upSlope <- nullp(gene.vector_upSlope, ID.vector_upSlope, bias.data = LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall_upSlope <- goseq(pwf_upSlope, ID.vector_upSlope, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)

GO_upSlope <- GO.wall_upSlope[order(GO.wall_upSlope$over_represented_pvalue),]

colnames(GO_upSlope)[1] <- "GOterm"
    
#Write file of results 
write.csv(GO_upSlope, "../../../output/glmmseq/GOSeq/GOseq_DEG_Origin_upSlope.csv")

#Filtering for p < 0.05
GO_upSlope_05 <- GO_upSlope %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

#Filtering for p < 0.05
GO_upSlope_001 <- GO_upSlope %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

dim(GO_upSlope_05)
dim(GO_upSlope_001)

```

There are 422 over-represented GO terms (by p <0.05) in the 346 significant DEGs by Origin that are upregulated in the Slope origin in both treatments

Plotting!

```{r}
ontologies<-c("BP", "CC", "MF")

for (category in ontologies) {
    
    go_results <- GO_upSlope_05
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(over_represented_pvalue != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$over_represented_pvalue), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot_upslope <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=over_represented_pvalue)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot_upslope)
  
ggsave(filename=paste0("../../../output/glmmseq/GOSeq/OriginDEGs_upslope_P05", "_", category, ".png"), plot=GO.plot_upslope, dpi=100, width=12, height=48, units="in", limitsize=FALSE)
}
```

### What about the direction of FC: Genes significant by Origin and upregulated in Flat origin (negative FC) for both treatments (we should also look into the ones that are positive in one treatment but not the other)

```{r}
### Generate vector with names of the 481 significant DEGs by Origin that are upregulated in the Flat origin in both treatments

ID.vector_upFlat <- geneInfo %>%
  filter(q_Origin < 0.05) %>%
  filter(Stable_OriginFC < 0 & Variable_OriginFC < 0) %>%
  pull(gene_id)

##Construct list of genes with 1 for genes in that are significant for Origin and upregulated in Flat and 0 for those that are not

gene.vector_upFlat = as.integer(ALL.vector %in% ID.vector_upFlat) 

names(gene.vector_upFlat) <- ALL.vector #set names

#weight gene vector by bias for length of gene
pwf_upFlat <- nullp(gene.vector_upFlat, ID.vector_upFlat, bias.data = LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall_upFlat <- goseq(pwf_upFlat, ID.vector_upFlat, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)

GO_upFlat <- GO.wall_upFlat[order(GO.wall_upFlat$over_represented_pvalue),]

colnames(GO_upFlat)[1] <- "GOterm"

#Write file of results 
write.csv(GO_upFlat, "../../../output/glmmseq/GOSeq/GOseq_DEG_Origin_upFlat.csv")

#Filtering for p < 0.05
GO_upFlat_05 <- GO_upFlat %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

#Filtering for p < 0.05
GO_upFlat_001 <- GO_upFlat %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

dim(GO_upFlat_05)
dim(GO_upFlat_001)
```

There are 243 over-represented GO terms (by p <0.05) in the 481 significant DEGs by Origin that are upregulated in the Flat origin in both treatments

Plotting!

```{r}
ontologies<-c("BP", "CC", "MF")

for (category in ontologies) {
    
    go_results <- GO_upFlat_05
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$over_represented_pvalue), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot_upFlat <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=over_represented_pvalue)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot_upFlat)
  
ggsave(filename=paste0("../../../output/glmmseq/GOSeq/OriginDEGs_upFlat_P05", "_", category, ".png"), plot=GO.plot_upFlat, dpi=100, width=12, height=12, units="in", limitsize=FALSE)
}
```


### Genes significant by Origin and upregulated in Slope in the stable treatment but down in Slope (up in Flat) in the Variable treatment

Stable_OriginFC > 0 & Variable_OriginFC < 0

```{r}
### Generate vector with names of the 11 significant DEGs by Origin that are upregulated in the Slope origin in only the stable treatment
ID.vector_Slope_upStable_downVariable <- geneInfo %>%
  filter(q_Origin < 0.05) %>%
  filter(Stable_OriginFC > 0 & Variable_OriginFC < 0) %>%
  pull(gene_id)

##Construct list of genes with 1 for genes in that are significant for Origin and upregulated in Slope and 0 for those that are not

gene.vector_updownSlope = as.integer(ALL.vector %in% ID.vector_Slope_upStable_downVariable) 

names(gene.vector_updownSlope) <- ALL.vector #set names

#weight gene vector by bias for length of gene
pwf_updownSlope <- nullp(gene.vector_updownSlope, ID.vector_Slope_upStable_downVariable, bias.data = LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall_updownSlope <- goseq(pwf_updownSlope, ID.vector_Slope_upStable_downVariable, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)

GO_updownSlope <- GO.wall_updownSlope[order(GO.wall_updownSlope$over_represented_pvalue),]

colnames(GO_updownSlope)[1] <- "GOterm"
    
#Write file of results 
write.csv(GO_updownSlope, "../../../output/glmmseq/GOSeq/GOseq_DEG_Origin_Slope_upStable_downVariable.csv")

#Filtering for p < 0.05
GO_updownSlope_05 <- GO_updownSlope %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

#Filtering for p < 0.001
GO_updownSlope_001 <- GO_updownSlope %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

dim(GO_updownSlope_05)
dim(GO_updownSlope_001)

```

There are 151 over-represented GO terms (by p <0.05) in the 11 significant DEGs by Origin that are upregulated in the Slope origin in Stable treatment and downregulated in the variable treatment

Plotting!

```{r}
ontologies<-c("BP", "CC", "MF")

for (category in ontologies) {
    
    go_results <- GO_updownSlope_05
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(over_represented_pvalue != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$over_represented_pvalue), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot_updownSlope <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=over_represented_pvalue)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot_updownSlope)
  
ggsave(filename=paste0("../../../output/glmmseq/GOSeq/OriginDEGs_updownSlope_P05", "_", category, ".png"), plot=GO.plot_updownSlope, dpi=100, width=12, height=24, units="in", limitsize=FALSE)
}
```


### Genes significant by Origin and upregulated in Slope in the Vaiable treatment but downregulated in Slope (up in Flat) in the Stable treatment

Stable_OriginFC < 0 & Variable_OriginFC > 0

2 genes: one is unannotated ("Pocillopora_acuta_HIv2___RNAseq.g5107.t1a") and one is: 

Pocillopora_acuta_HIv2___RNAseq.g7761.t1
6087.XP_002155939.2
Belongs to the DHHC palmitoyltransferase family

```{r}
### Generate vector with names of the 2 significant DEGs by Origin that are upregulated in the Slope origin in both treatments
ID.vector_Slope_downStable_upVariable <- geneInfo %>%
  filter(q_Origin < 0.05) %>%
  filter(Stable_OriginFC < 0 & Variable_OriginFC > 0) %>%
  pull(gene_id)

##Construct list of genes with 1 for genes in that are significant for Origin and upregulated in Slope and 0 for those that are not

gene.vector_downupSlope = as.integer(ALL.vector %in% ID.vector_Slope_downStable_upVariable) 

names(gene.vector_downupSlope) <- ALL.vector #set names

#weight gene vector by bias for length of gene
pwf_downupSlope <- nullp(gene.vector_downupSlope, ID.vector_Slope_downStable_upVariable, bias.data = LENGTH.vector) 
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall_downupSlope <- goseq(pwf_downupSlope, ID.vector_downupSlope, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)

GO_downupSlope <- GO.wall_downupSlope[order(GO.wall_downupSlope$over_represented_pvalue),]

colnames(GO_downupSlope)[1] <- "GOterm"
    
#Write file of results 
write.csv(GO_downupSlope, "../../../output/glmmseq/GOSeq/GOseq_DEG_Origin_Slope_downStable_upVariable.csv")

#Filtering for p < 0.05
GO_downupSlope_05 <- GO_downupSlope %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

#Filtering for p < 0.001
GO_downupSlope_001 <- GO_downupSlope %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

dim(GO_downupSlope_05)
dim(GO_downupSlope_001)

```

There are 11 over-represented GO terms (by p <0.05) in the 2 significant DEGs by Origin that are downregulated in the Slope origin in Stable treatment and upregulated in the variable treatment

Plotting!

```{r}
ontologies<-c("BP", "MF")

for (category in ontologies) {
    
    go_results <- GO_downupSlope_05
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(over_represented_pvalue != "NA") %>%
      #filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$over_represented_pvalue), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot_downupSlope <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=over_represented_pvalue)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot_downupSlope)
  
ggsave(filename=paste0("../../../output/glmmseq/GOSeq/OriginDEGs_downupSlope_P05", "_", category, ".png"), plot=GO.plot_downupSlope)
}
```