---
title: "Fix GFF format"
author: "Ariana S Huffmyer (edited by Zoe Dellaert)"
date: "2/4/2023"
output: html_document
editor_options: 
  chunk_output_type: console
---
This script adds transcript and gene id into GFF file for alignment.  

Here, I'll be adding transcript_id= and gene_id= to 'gene' column because we needs that label to map our TagSeq data  

Load libraries and data. 
```{r}
#Load libraries
library(tidyverse)
library(R.utils)
```

Load  gene gff file

```{r}
setwd("~/Documents/URI/Heron-Pdam-gene-expression/BioInf")
gff <- read.csv(file="Pocillopora_acuta_HIv2.genes.gff3", header=FALSE, sep="\t") 
```

Rename columns 

```{r}
colnames(gff) <- c("scaffold", "Gene.Predict", "id", "gene.start","gene.stop", "pos1", "pos2","pos3", "gene")
head(gff)

gff_old <- gff
```

Create transcript ID  
```{r}
gff$transcript_id <- sub(";.*", "", gff$gene)
gff$transcript_id <- gsub("ID=", "", gff$transcript_id) #remove ID= 
gff$transcript_id <- gsub("Parent=", "", gff$transcript_id) #remove ID= 
head(gff)
```

Create Parent ID 
```{r}
gff$parent_id <- sub(".*Parent=", "", gff$gene)
gff$parent_id <- sub(";.*", "", gff$parent_id)
gff$parent_id <- gsub("ID=", "", gff$parent_id) #remove ID= 
head(gff)
```

Now add these values into the gene column separated by semicolons.  

```{r}
gff <- gff %>% 
  mutate(gene = ifelse(id != "gene", paste0(gene, ";transcript_id=", gff$transcript_id, ";gene_id=", gff$parent_id),  paste0(gene)))
head(gff)
```

Now remove the transcript and parent ID separate columns.  

```{r}
gff<-gff %>%
  select(!transcript_id)%>%
  select(!parent_id)

head(gff)
```

Save file. Then upload this to Andromeda for use in bioinformatic steps.  

```{r}
write.table(gff, file="Pocillopora_acuta_HIv2.genes_fixed.gff3", sep="\t", col.names = FALSE, row.names=FALSE, quote=FALSE)

#gzip the file 
gzip("Pocillopora_acuta_HIv2.genes_fixed.gff3")
```

`scp /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/Pocillopora_acuta_HIv2.genes_fixed.gff3.gz zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/references/`

or 

`wget https://github.com/imkristenbrown/Heron-Pdam-gene-expression/tree/master/BioInf/Pocillopora_acuta_HIv2.genes_fixed.gff3.gz`  
