---
title: "BLAST-ing our genes to the Biomineralization Toolkit"
author: "Zoe Dellaert"
date: "8/24/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

# BLAST-ing our genes to the Biomineralization Toolkit

On andromeda

Download protein file from genome

```{bash eval=FALSE}
cd /data/putnamlab/zdellaert/Pdam-TagSeq/references

wget http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.pep.faa.gz

gunzip Pocillopora_acuta_HIv2.genes.pep.faa.gz

cd ..

mkdir blast
cd blast
```

On personal computer

```{bash eval=FALSE}
scp  /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/data/Biomineralization_Toolkit_FScucchia/Biomineralization_Toolkit_FScucchia.fasta zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/blast/
Biomineralization_Toolkit_FScucchia.fasta
```

On andromeda

```{bash eval=FALSE}
nano Biomineralization_blast.sh
```


```{bash eval=FALSE}
#!/bin/bash
#SBATCH --job-name="Pacuta_TRP_blast"
#SBATCH -t 240:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL #email you when job starts, stops and/or fails
#SBATCH --mail-user=zdellaert@uri.edu #your email to send notifications
#SBATCH --mem=100GB
#SBATCH --error="blast_out_error"
#SBATCH --output="blast_out"
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/zdellaert/Pdam-TagSeq/blast/
#SBATCH --nodes=1 --ntasks-per-node=20

module load BLAST+/2.9.0-iimpi-2019b

makeblastdb -in ../references/Pocillopora_acuta_HIv2.genes.pep.faa -out Pacuta_prot -dbtype prot

blastp -query Biomineralization_Toolkit_FScucchia.fasta -db Pacuta_prot -out Biomineralization_blast_results.txt -outfmt 0

blastp -query Biomineralization_Toolkit_FScucchia.fasta -db Pacuta_prot -out Biomineralization_blast_results_tab.txt -outfmt 6 -max_target_seqs 1
```


```{bash eval=FALSE}
sbatch Biomineralization_blast.sh
```

Errors:

Warning: [blastp] Examining 5 or more matches is recommended
FASTA-Reader: Ignoring invalid residues at position(s): On line 2741: 378, 383, 386-390, 401, 417, 420-422, 431, 437-439, 443, 459-461
Warning: [blastp] Query_168 Gene: g13552, N.. : One or more O characters replaced by X for alignment score calculations at positions 382, 390, 392, 422

On personal computer:

```{bash eval=FALSE}
scp  zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/blast/Biomineralization_blast_results.txt /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/output

scp  zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/blast/Biomineralization_blast_results_tab.txt /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/output
```

Now, will take the best Pacuta alignment for each Biomineralization Gene and match to the name of that gene and make the dataframe into a format to match with differentially expressed/frontloaded genes or modules.

 
```{r}
sessionInfo() #provides list of loaded packages and version of R. 
```

```{r}
library(dplyr)
```

```{r}
library(readxl)
Biomin_genes <- read_excel("~/Documents/URI/Heron-Pdam-gene-expression/BioInf/data/Biomineralization_Toolkit_FScucchia/Biomineralization_Toolkit_FScucchia.xlsx")

Biomin_genes <- Biomin_genes %>% select(-`blasted protein in Stylophora`)
```


```{r}
Biomin_blast_results <- read.delim("~/Documents/URI/Heron-Pdam-gene-expression/BioInf/output/Biomineralization_blast_results_tab.txt", header=FALSE) 

Biomin_blast_results <- Biomin_blast_results %>% select(V1, V2) %>% distinct()
```

```{r}
# Merge data frames based on accessionnumber/geneID
merged_data <- Biomin_genes %>%
  inner_join(Biomin_blast_results, by = c("accessionnumber/geneID" = "V1")) %>% rename("Pocillopora_acuta_best_hit" = "V2")

write.csv(merged_data, "~/Documents/URI/Heron-Pdam-gene-expression/BioInf/output/Biomin_blast_Pocillopora_acuta_best_hit.csv", row.names = F)
```


How many of our 9011 genes are represented in the Biomineralization genes?

```{r}
DEGs <- read.csv(file="../../../output/Slope_Base/signif_genes_normcts.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

#NOTE! This is not a file only with differentially expressed genes, this contains all of the genes in our dataset but also contains p-value information and fold change information to help determine which genes are signficant DEGs based on our model in glmmSeq

rownames(DEGs) <- DEGs$Gene

dim(DEGs)

Biomin_genes <- DEGs %>%
  inner_join(merged_data, by = c("Gene" = "Pocillopora_acuta_best_hit"))

Biomin_genes$definition

length(Biomin_genes$definition)

Biomin_genes_names <- unique(Biomin_genes$Gene)

length(Biomin_genes_names)

Biomin_genes %>% select(Gene, `accessionnumber/geneID`, definition, Ref) 

#Biomin_genes %>% select(Gene, `accessionnumber/geneID`, definition, Ref, Origin, Treatment, Treatment.Origin) %>% View()


write.csv(Biomin_genes, "~/Documents/URI/Heron-Pdam-gene-expression/BioInf/output/Biomin_blast_Pocillopora_acuta_best_hit_glmmSeq.csv", row.names = F)
```

127/172 of the Biomineralization Genes are represented in our dataset of 9011 genes, matching to 65/9011 genes


Differentially expressed genes: are any of these Biomineralization genes?

```{r}
Origin_DEGs <- DEGs %>%  dplyr::filter(Origin < 0.05)

nrow(Origin_DEGs)

Treatment_DEGs <- DEGs %>%  dplyr::filter(Treatment < 0.05)

nrow(Treatment_DEGs)

Interaction_DEGs <- DEGs %>%  dplyr::filter(Treatment.Origin < 0.05)

nrow(Interaction_DEGs)
```

Setting up for plotting genes, loading in results from glmmseq
```{r}
library(glmmSeq)

results <- readRDS(file = "glmmSeq.rds") #load in RDS from previous step / previous iteration
results <- glmmQvals(results)
source(file = "../Factor_ggmodelPlot.R")

plotColours <- c("skyblue","mediumseagreen")
modColours <- c("dodgerblue3","seagreen4")
```


```{r}
Biomin_Origin_DEGs <- Origin_DEGs %>%
  inner_join(merged_data, by = c("Gene" = "Pocillopora_acuta_best_hit"))

Biomin_Origin_DEGs$definition

length(Biomin_Origin_DEGs$definition)

Biomin_Origin_DEG_names <- unique(Biomin_Origin_DEGs$Gene)

length(Biomin_Origin_DEG_names)

Biomin_Origin_DEGs %>% select(Gene, `accessionnumber/geneID`, definition, Ref) 
```

16/172 of the Biomineralization Genes are represented in the Origin DEGS, and these are 10 Pocillopora genes (some of the 10 have matches to multiple Biomineralization Genes) out of the 65 that are matching to Biomineralization Genes (10/65)

Pocillopora_acuta_HIv2___TS.g13222.t1b is a best match for:
  - Gene:g15294.t1	Annotated: Vitellogenin
  - P24_g15888	clone g15888 vitellogenin-like protein gene
  - P26_g1441	clone g1441 vitellogenin-like protein gene
  - XP_022779720.1	vitellogenin-like [Stylophora pistillata]
  
Pocillopora_acuta_HIv2___TS.g12304.t1 is a best match for:
  - ACE95141.1	carbonic anhydrase [Stylophora pistillata]
  - EU532164.1	carbonic anhydrase 2
  - Gene:g29033.t1	Annotated: Carbonic Anhydrase (STPCA2-1)
  - Gene:g29034.t1	Annotated: CarbonicAnhyrase


```{r}
for (i in Biomin_Origin_DEG_names) {print(Factor_ggmodelPlot(results,
            geneName = i,
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            title = i,
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3))}
```


```{r}
Biomin_Treatment_DEGs <- Treatment_DEGs %>%
  inner_join(merged_data, by = c("Gene" = "Pocillopora_acuta_best_hit"))

Biomin_Treatment_DEGs$definition
```

0/172 of the Biomineralization Genes are represented in the Treatment DEGS


```{r}
Biomin_Interaction_DEGs <- Interaction_DEGs %>%
  inner_join(merged_data, by = c("Gene" = "Pocillopora_acuta_best_hit"))

Biomin_Interaction_DEGs$definition
```

0/172 of the Biomineralization Genes are represented in the Interaction DEGS


Frontloaded genes!

```{r}
FRONTs <- read.csv(file="../../../output/Slope_Base/frontloaded_genes.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

Biomin_FRONTs <- FRONTs %>%
  inner_join(merged_data, by = c("Gene" = "Pocillopora_acuta_best_hit"))

Biomin_FRONTs$definition

length(Biomin_FRONTs$definition)

Biomin_FRONTs_names <- unique(Biomin_FRONTs$Gene)

length(Biomin_FRONTs_names)

Biomin_FRONTs %>% select(Gene, `accessionnumber/geneID`, definition, Ref) 
```

64/172 of the Biomineralization Genes are represented in the Frontloaded genes

This is 29 genes, some of which are mapping to multiple Biomineralization genes, out of the 65 that are matching to Biomineralization Genes (29/65)

```{r}
for (i in Biomin_FRONTs_names) {print(Factor_ggmodelPlot(results,
            geneName = i,
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            title = i,
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3))}
```

```{r}
READY <- read.csv(file="../../../output/Slope_Base/frontloaded_genes_plotting.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))


READY$color <- rep('gray', nrow(READY))
#These are "frontloaded, need a different color:
READY$color[READY$yall > 1 & READY$xall_1 < 1] <- 'gray40'
READY$color[READY$Gene %in% merged_data$Pocillopora_acuta_best_hit] <- 'green4'
READY$color[READY$yall > 1 & READY$xall_1 < 1 & READY$Gene %in% merged_data$Pocillopora_acuta_best_hit] <- 'green'

READY_cutoff <- READY %>% dplyr::filter(yall < 6) %>% dplyr::filter(xall_1 < 6)

P <- READY_cutoff %>% 
        ggplot(aes(x=xall_1, y=yall)) +
        #geom_point(colour = READY_cutoff$color, alpha=0.8) +
        geom_point(data = subset(READY_cutoff, READY_cutoff$color != "green4"), colour = subset(READY_cutoff$color, READY_cutoff$color != "green4"), alpha = 0.7) +
        geom_point(data = subset(READY_cutoff, READY_cutoff$color == "green4"), colour = subset(READY_cutoff$color, READY_cutoff$color == "green4"), alpha = 1) +
        geom_point(data = subset(READY_cutoff, READY_cutoff$color == "green"), colour = subset(READY_cutoff$color, READY_cutoff$color == "green"), alpha = 1) +
        theme_classic() + 
        stat_smooth(method = "lm", formula = y ~ x + poly(x, 2) - 1) +
        geom_vline(xintercept=1, linetype="dotted") + 
        geom_hline(yintercept=1, linetype="dotted") + 
        labs(y= "Flat to Slope (Conditioned to naive) control ratio", 
             x = "Flat to Slope (Conditioned to naive) foldchange ratio",
             title = "Frontloaded genes") + 
        scale_x_continuous(limits = c(0,6.1),expand = c(0, 0)) + scale_y_continuous(limits = c(0,6.1), expand = c(0, 0)) + 
        annotate("rect", xmin = 0, xmax = 1, ymin = 1, ymax = 6.1, alpha = .2) + 
        annotate("rect", xmin = 0, xmax = 1, ymin = 0, ymax = 1,alpha = .5)

P
```


Output lists of frontloaded genes with Biomineralization gene info

```{r}
Biomin_FRONTs_info <- merged_data %>% filter(merged_data$Pocillopora_acuta_best_hit %in% Biomin_FRONTs_names)


write.csv(Biomin_FRONTs_info, "~/Documents/URI/Heron-Pdam-gene-expression/BioInf/output/Biomin_frontloaded.csv", row.names = F)

```


-----

VST reaction norm for just frontloaded biomineralization genes 

```{r}
#load in vst normalized gene expression

vst <- read.csv(file="../../../output/Slope_Base/vst.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))
```


Lets also pull in our metadata

```{r}
coldata <- read.csv("../../../../RAnalysis/Data/RNA Submission Sample List metadata.csv") #read in metadata file
coldata <- plyr::rename(coldata, c("Sample.Name"="Coral_ID")) #Make a column that represents the colonies
coldata$Colony <- gsub("A", "", coldata$Coral_ID) #ID which sample is from which colony by remobing letters after colony name
coldata$Colony <- gsub("B", "", coldata$Colony) #ID which sample is from which colony by remobing letters after colony name
coldata$Colony <- gsub("C", "", coldata$Colony) #ID which sample is from which colony by remobing letters after colony name
coldata$Colony <- gsub("D", "", coldata$Colony) #ID which sample is from which colony by remobing letters after colony name

coldata$Origin <- factor(coldata$Origin) #set variables to factors
coldata$Colony <- factor(coldata$Colony) #set variables to factors
coldata$Treatment <- factor(coldata$Treatment) #set variables to factors

coldata <- coldata %>% filter(Coral_ID != c("RF16A","RF16C")) #removed outlier rows from metadata
head(coldata)
```

```{r}
#  reaction norm data config and plotting 
vst_front  <- vst[which(vst$Gene %in% Biomin_FRONTs_names),]

vst_front  <- vst_front %>%  reshape2::melt(id.var = 'Gene') %>% 
                    dplyr::rename(Coral_ID = variable)

vst_front_Merge <- merge(vst_front, coldata, by = 'Coral_ID') %>% 
    mutate(group = paste(Treatment, Origin, sep = "_"))
  

vst_summary <- vst_front_Merge %>%
  group_by(Gene, Treatment, Origin, group) %>%
  summarise(mean_val = mean(value),
            sd_val = sd(value),
            se_val = sd_val/sqrt(n())) %>%
  ungroup()

group_summary <- vst_summary %>%
  group_by(Treatment, Origin) %>%
  summarise(meanExp = mean(mean_val),
            sdExp = sd(mean_val),
            n = n(),
            se = sdExp/sqrt(n))

# anova
ANOVA_moderate <- aov(lm(mean_val~as.character(group), data = vst_summary))
summary(ANOVA_moderate) # as.character(group)      3    1.4  0.4666   0.154  0.927
TukeyHSD(ANOVA_moderate)

#                                    diff        lwr       upr     p adj
#Stable_Slope-Stable_Flat     -0.28439865 -1.4761351 0.9073378 0.9247110
#Variable_Flat-Stable_Flat    -0.05615529 -1.2478918 1.1355812 0.9993328
#Variable_Slope-Stable_Flat   -0.17559153 -1.3673280 1.0161449 0.9806102
#Variable_Flat-Stable_Slope    0.22824336 -0.9634931 1.4199798 0.9590235
#Variable_Slope-Stable_Slope   0.10880712 -1.0829293 1.3005436 0.9952236
#Variable_Slope-Variable_Flat -0.11943624 -1.3111727 1.0723002 0.9937106


#Chi squre test
library(lsmeans)

model <- glm(mean_val ~ group, data = vst_summary)

lsmeans_results <- lsmeans(model, pairwise ~ group)

posthoc_results <- pairs(lsmeans_results)
posthoc_results

 # contrast                       estimate    SE  df t.ratio p.value
 # Stable_Flat - Stable_Slope       0.2844 0.457 112   0.622  0.9247
 # Stable_Flat - Variable_Flat      0.0562 0.457 112   0.123  0.9993
 # Stable_Flat - Variable_Slope     0.1756 0.457 112   0.384  0.9806
 # Stable_Slope - Variable_Flat    -0.2282 0.457 112  -0.499  0.9590
 # Stable_Slope - Variable_Slope   -0.1088 0.457 112  -0.238  0.9952
 # Variable_Flat - Variable_Slope   0.1194 0.457 112   0.261  0.9937

#######

ggplot(group_summary, aes(x = Treatment, y = meanExp, color = Origin)) +
  geom_errorbar(aes(ymin = meanExp -  se, ymax = meanExp + se, color = Origin), width = 0.05) +
  geom_line(aes(group = Origin), color = "darkgrey") +
   geom_point(size = 3) + 
  labs(x = "Treatment", y = "VST Mean Gene Expression (mean +- SE)") +
  theme_minimal()


ggplot(group_summary, aes(x = Treatment, y = meanExp, fill = Origin, group = Origin)) +
  geom_line(aes(linetype = Origin), size = 0.5) +
  geom_pointrange(aes(ymin=meanExp-se, ymax=meanExp+se, shape=Origin, fill=Origin)) + 
                      scale_shape_manual(values=c(1, 16))+
                      scale_fill_manual(values=c('black','white')) +
                      theme_classic() +
  scale_linetype_manual(values = c("Slope" = "solid", "Flat" = "dashed"))  +
                      labs(y= "mean SE VST expression", 
                           x = "Treatment",
                           title = "Mean VST Expression of Frontloaded Biomineralization-Related Genes") + 
   theme(text = element_text(size=12)) +
                      #scale_y_continuous(limits = c(7.28,7.48), breaks = seq(7.2,7.5, by = .1)) +
                        #annotate(geom="text", x=0.94, y=7.31, label="b", color="black",size=4) +
                        #annotate(geom="text", x=0.94, y=7.46,  label="a", color="black",size=4) +
                        #annotate(geom="text", x=2.06, y=7.39, label="c", color="black",size=4) +
                        #annotate(geom="text", x=2.09, y=7.42,  label="a,c", color="black",size=4) +
                        annotate(geom="text", x = 2.3, y = 7.9, label=paste("N = ",group_summary$n[1]), size = 5) + guides(fill = FALSE, shape = FALSE)

```



```{r}
# Normality test
shapiro.test(vst_summary$mean_val[vst_summary$Treatment == "Stable" & vst_summary$Origin == "Slope"])
shapiro.test(vst_summary$mean_val[vst_summary$Treatment == "Variable" & vst_summary$Origin == "Slope"])
shapiro.test(vst_summary$mean_val[vst_summary$Treatment == "Stable" & vst_summary$Origin == "Flat"])
shapiro.test(vst_summary$mean_val[vst_summary$Treatment == "Variable" & vst_summary$Origin == "Flat"])

# Data is not normally distributed
```


```{r}
# Perform Kruskal-Wallis test
kruskal_test <- kruskal.test(mean_val ~ group, data = vst_summary)

# Check the Kruskal-Wallis test results
print(kruskal_test)


library(dunn.test)

# Conduct Dunn's post-hoc test
posthoc_dunn <- dunn.test(vst_summary$mean_val, g = vst_summary$group, method = "bonferroni")

# Check the post-hoc test results
print(posthoc_dunn)
```



