# BLAST-ing our genes to the Biomineralization Toolkit


On andromeda

Download protein file from genome

```{bash}
cd /data/putnamlab/zdellaert/Pdam-TagSeq/references

wget http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.pep.faa.gz

gunzip Pocillopora_acuta_HIv2.genes.pep.faa.gz

cd ..

mkdir blast
cd blast
```

On personal computer

```{bash}
scp  /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/data/Biomineralization_Toolkit_FScucchia/Biomineralization_Toolkit_FScucchia.fasta zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/blast/
Biomineralization_Toolkit_FScucchia.fasta
```

On andromeda

```{bash}
nano Biomineralization_blast.sh
```


```{bash}
#!/bin/bash
#SBATCH --job-name="Pacuta_TRP_blast"
#SBATCH -t 240:00:00
#SBATCH --export=NONE
#SBATCH --mail-type=BEGIN,END,FAIL #email you when job starts, stops and/or fails
#SBATCH --mail-user=zdellaert@uri.edu #your email to send notifications
#SBATCH --mem=100GB
#SBATCH --error="blast_out_error"
#SBATCH --output="blast_out"
#SBATCH --account=putnamlab
#SBATCH -D /data/putnamlab/zdellaert/Pdam-TagSeq/blast/
#SBATCH --nodes=1 --ntasks-per-node=20

module load BLAST+/2.9.0-iimpi-2019b

makeblastdb -in ../references/Pocillopora_acuta_HIv2.genes.pep.faa -out Pacuta_prot -dbtype prot

blastp -query Biomineralization_Toolkit_FScucchia.fasta -db Pacuta_prot -out Biomineralization_blast_results.txt -outfmt 0

blastp -query Biomineralization_Toolkit_FScucchia.fasta -db Pacuta_prot -out Biomineralization_blast_results_tab.txt -outfmt 6 -max_target_seqs 1
```


```{bash}
sbatch Biomineralization_blast.sh
```

Errors:

Warning: [blastp] Examining 5 or more matches is recommended
FASTA-Reader: Ignoring invalid residues at position(s): On line 2741: 378, 383, 386-390, 401, 417, 420-422, 431, 437-439, 443, 459-461
Warning: [blastp] Query_168 Gene: g13552, N.. : One or more O characters replaced by X for alignment score calculations at positions 382, 390, 392, 422

On personal computer:

```{bash}
scp  zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/blast/Biomineralization_blast_results.txt /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/output

scp  zdellaert@ssh3.hac.uri.edu:/data/putnamlab/zdellaert/Pdam-TagSeq/blast/Biomineralization_blast_results_tab.txt /Users/zoedellaert/Documents/URI/Heron-Pdam-gene-expression/BioInf/output
```

Now, will take the best Pacuta alignment for each Biomineralization Gene and match to the name of that gene and make the dataframe into a format to match with differentially expressed/frontloaded genes or modules.

 
```{r}
sessionInfo() #provides list of loaded packages and version of R. 
```

```{r}
library(dplyr)
```

```{r}
library(readxl)
Biomin_genes <- read_excel("~/Documents/URI/Heron-Pdam-gene-expression/BioInf/data/Biomineralization_Toolkit_FScucchia/Biomineralization_Toolkit_FScucchia.xlsx")

Biomin_genes <- Biomin_genes %>% select(-`blasted protein in Stylophora`)
```


```{r}
Biomin_blast_results <- read.delim("~/Documents/URI/Heron-Pdam-gene-expression/BioInf/output/Biomineralization_blast_results_tab.txt", header=FALSE) 

Biomin_blast_results <- Biomin_blast_results %>% select(V1, V2) %>% distinct()
```

```{r}
# Merge data frames based on accessionnumber/geneID
merged_data <- Biomin_genes %>%
  inner_join(Biomin_blast_results, by = c("accessionnumber/geneID" = "V1")) %>% rename("Pocillopora_acuta_best_hit" = "V2")

write.csv(merged_data, "~/Documents/URI/Heron-Pdam-gene-expression/BioInf/output/Biomin_blast_Pocillopora_acuta_best_hit.csv", row.names = F)
```









