---
title: "Frontloaded genes, Enrich after glmmSeq, Slope as Base Level"
author: "Zoe Dellaert"
date: "4/24/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is a Rmd file analyzing our raw count data by the glmmSeq package as described in the [vignette](https://cran.r-project.org/web/packages/glmmSeq/vignettes/glmmSeq.html) and [manual](https://cran.r-project.org/web/packages/glmmSeq/glmmSeq.pdf).

```{r}
sessionInfo() #provides list of loaded packages and version of R. I still have version 4.1 for now.
```

```{r install packages}
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")
if ("goseq" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install('goseq') 
if ("rrvgo" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rrvgo")
if ("GO.db" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("GO.db")
#BiocManager::install("org.Ce.eg.db", force=TRUE) #install if needed

library(dplyr)
library(ggplot2)
library(goseq)
library(rtracklayer)
library(rrvgo)
library(GO.db)
library(tidyr)
library(forcats)
sessionInfo() #list of packages after library-ing these packages
```


2631 genes are frontloaded, with a yall ratio larger than 1 and a xall_1 ratio less than 1.

```{r}
DEGs <- read.csv(file="../../../output/Slope_Base/signif_genes_normcts.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

#NOTE! This is not a file only with differentially expressed genes, this contains all of the genes in our dataset but also contains p-value information and fold change information to help determine which genes are signficant DEGs based on our model in glmmSeq

rownames(DEGs) <- DEGs$Gene

dim(DEGs)
```

```{r}
genes <- rownames(DEGs)
```

## **Generate files for Functional Enrichment**  

Based off of [Ariana's script](https://github.com/AHuffmyer/EarlyLifeHistory_Energetics/blob/master/Mcap2020/Scripts/TagSeq/Genome_V3/1_WGCNA_Mcap_V3.Rmd)

Functional annotation file obtained from: http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.EggNog_results.txt.gz on April 3, 2023.  


```{r}
Pacuta.annot <- read.delim("../../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt") %>% dplyr::rename("query" = X.query)

dim(Pacuta.annot)
head(Pacuta.annot)
```

```{r}
genes2annot = match(genes, Pacuta.annot$query) #match genes in DEGs (all genes after filtering) to genes in annotation file

sum(is.na(genes2annot)) #number of genes without EggNog annotation
missing<-as.data.frame(genes[which(is.na(genes2annot))]) #dataframe of genes without EggNog annotation
head(missing)
```

2812/9011 genes without eggnog annotation

```{r}
names(Pacuta.annot)

geneInfo0 = data.frame(gene_id = genes, #add gene id
  Accession = Pacuta.annot$seed_ortholog[genes2annot], #add accession number
  Bitscore = Pacuta.annot$score[genes2annot], #add bitscore
  eValue = Pacuta.annot$evalue[genes2annot], #add e value
  Description = Pacuta.annot$Description[genes2annot], #add description of gene
  Annotation.GO.ID = Pacuta.annot$GOs[genes2annot], #add GO ID's
  q_Origin = DEGs$Origin, #add Origin adjusted p-value
  q_Treatment = DEGs$Treatment, #add Treatment adjusted p-value
  q_Interaction = DEGs$Treatment.Origin, #add Treatment:Origin adjusted p-value
  Stable_OriginFC = DEGs$Stable_OriginFC, #add fold change for Slope vs Flat in the stable treatment
  Variable_OriginFC = DEGs$Variable_OriginFC, #add fold change for Slope vs Flat in the variable treatment
  maxGroupFC = DEGs$maxGroupFC, #add max group fold change (was FC bigger in stable of variable treatment)
  col = DEGs$col) #add qualitative significance info

dim(geneInfo0)
head(geneInfo0,2)
```

Add KEGG annotation information.  Downloaded from: http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.KEGG_results.txt.gz on April 3, 2023.  

```{r}
kegg<-read.table("../../../data/Pocillopora_acuta_HIv2.genes.KEGG_results.txt", sep="", quote="", na.strings=c("","NA"), blank.lines.skip = FALSE, header=FALSE)

dim(kegg)
head(kegg)
```

Add KEGG annotations to each gene.  
```{r}
geneInfo0$KEGG <- kegg$V2[match(geneInfo0$gene_id, kegg$V1)]
```

Order geneInfo0 by significance of Origin, q_Origin (adjusted p value)
```{r}
geneInfo <- geneInfo0[order(geneInfo0[, 'q_Origin']), ]

write.csv(geneInfo, file = "../../../output/Slope_Base/geneInfo.csv") #gene info for reference/supplement
```


## Now onto Enrichment!
```{r}
#geneInfo<-read.csv("../../../output/Slope_Base/geneInfo.csv")

dim(geneInfo)

```

Get gene length information.  
```{r}
#import file
gff <- rtracklayer::import("../../../data/Pocillopora_acuta_HIv2.genes_fixed.gff3") #if this doesn't work, restart R and try again 
transcripts <- subset(gff, type == "transcript") #keep only transcripts , not CDS or exons
transcript_lengths <- width(transcripts) #isolate length of each gene
seqnames<-transcripts$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame

geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)] #Add in length to geneInfo
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  Annotation.GO.ID column
```{r}
geneInfo$Annotation.GO.ID <- gsub(",", ";", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub('"', "", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub("-", NA, geneInfo$Annotation.GO.ID)
```


```{r}
### Generate vector with names of all genes 
ALL.vector <- c(geneInfo$gene_id)

### Generate length vector for all genes
LENGTH.vector <- as.integer(geneInfo$Length)


#### THIS IS WHERE THE FRONTLOADING NEEDS TO COME IN
#2631 genes are frontloaded, with a yall ratio larger than 1 and a xall_1 ratio less than 1.


FRONTs <- read.csv(file="../../../output/Slope_Base/frontloaded_genes.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

### Generate vector with names of the 2631 frontloaded genes
ID.vector <- geneInfo %>%
  filter(gene_id %in% FRONTs$Gene) %>%
  pull(gene_id)

##Get a list of GO Terms for the 2631 frontloaded genes
GO.terms <- geneInfo %>%
  filter(gene_id %in% FRONTs$Gene) %>%
  dplyr::select(gene_id, Annotation.GO.ID)

##Format to have one goterm per row with gene ID repeated
split <- strsplit(as.character(GO.terms$Annotation.GO.ID), ";")
split2 <- data.frame(v1 = rep.int(GO.terms$gene, sapply(split, length)), v2 = unlist(split))
colnames(split2) <- c("gene", "Annotation.GO.ID")
GO.terms <- split2

##Construct list of genes with 1 for genes in that are frontloaded and 0 for those that are not
gene.vector = as.integer(ALL.vector %in% ID.vector) 

names(gene.vector) <- ALL.vector #set names

#weight gene vector by bias for length of gene
pwf <- nullp(gene.vector, bias.data = LENGTH.vector)
```

```{r}
#run goseq using Wallenius method for all categories of GO terms 
GO.wall <- goseq(pwf, gene2cat=GO.terms, method="Wallenius", use_genes_without_cat=TRUE)

GO <- GO.wall[order(GO.wall$over_represented_pvalue),]

colnames(GO)[1] <- "GOterm"
    
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values

#Write file of results 
write.csv(GO, "../../../output/Slope_Base/Frontloaded_GOSeq/GOseq_Frontloaded.csv")

#Filtering for p < 0.05
GO_05 <- GO %>%
        dplyr::filter(bh_adjust<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

#Filtering for p < 0.05
GO_001 <- GO %>%
        dplyr::filter(bh_adjust<0.001) %>%
        dplyr::arrange(., ontology, bh_adjust)

#Filtering for p < 0.05
GO_052 <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, bh_adjust)

#Filtering for p < 0.05
GO_0012 <- GO %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, bh_adjust)
```

Plotting!

```{r}
ontologies<-c("BP", "CC", "MF")

for (category in ontologies) {
    
    go_results <- GO_05
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(bh_adjust != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., bh_adjust)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$bh_adjust), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=bh_adjust)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot)
  
ggsave(filename=paste0("../../../output/Slope_Base/Frontloaded_GOSeq/FRONTs_P05", "_", category, ".png"), plot=GO.plot, dpi=100, width=12, height=100, units="in", limitsize=FALSE)
}
```

### Combining MF, CC, and BP into one plot and order by pvalue, based on [Jill's script](https://github.com/JillAshey/SedimentStress/blob/master/RAnalysis/pacuta/pacuta_GOSeq.Rmd)

```{r}
GO_plot_all <- GO_001 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, numDEInCat)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  #geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=1, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,305) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes

GO_plot_all

ggsave("../../../output/Slope_Base/Frontloaded_GOSeq/FRONTs_P001.pdf", GO_plot_all, width = 40, height = 500, units = c("in"), dpi=100, limitsize=FALSE)

#ordered by p-value
GO_plot_all_pval <- GO_001 %>% drop_na(ontology) %>% mutate(term = fct_reorder(term, over_represented_pvalue)) %>%
  mutate(term = fct_reorder(term, ontology)) %>%
  ggplot( aes(x=term, y=numDEInCat) ) +
  geom_segment( aes(x=term ,xend=term, y=0, yend=numDEInCat), color="grey") +
  #geom_text(aes(label = over_represented_pvalue), hjust = -1, vjust = 0, size = 2) +
  geom_point(size=1, aes(colour = ontology)) +
  coord_flip() +
  ylim(0,305) +
  theme(
    panel.grid.minor.y = element_blank(),
    panel.grid.major.y = element_blank(),
    legend.position="bottom"
  ) +
  xlab("") +
  ylab("") +
  theme_bw() + #Set background color 
  theme(panel.border = element_blank(), # Set border
        panel.grid.major = element_blank(), #Set major gridlines
        panel.grid.minor = element_blank(), #Set minor gridlines
        axis.line = element_line(colour = "black"), #Set axes color
        plot.background=element_blank()) #Set the plot background #set title attributes

GO_plot_all_pval

ggsave("../../../output/Slope_Base/Frontloaded_GOSeq/FRONTs_P001_orderedpval.pdf", GO_plot_all_pval, width = 40, height = 500, units = c("in"), dpi=100, limitsize=FALSE)
```


### KEGG enrichment
```{r}
##Get a list of KEGG Terms for the 2631 frontloaded genes
KO.terms <- geneInfo %>%
  filter(gene_id %in% FRONTs$Gene) %>%
  dplyr::select(gene_id, KEGG)


#run goseq using Wallenius method for all categories of KEGG terms
KO.wall <-
  goseq(
    pwf,
    ID.vector,
    gene2cat = KO.terms,
    test.cats = c("KEGG"),
    method = "Wallenius",
    use_genes_without_cat = TRUE
  )


KO <- KO.wall[order(KO.wall$over_represented_pvalue), ]
colnames(KO)[1] <- "KEGGterm"

#adjust p-values
#KO$bh_adjust <- p.adjust(KO$over_represented_pvalue, method = "BH") #add adjusted p-values

#Filtering for p < 0.05
KO_05 <- KO %>%
  dplyr::filter(over_represented_pvalue < 0.05) %>%
  dplyr::arrange(., over_represented_pvalue)

head(KO_05)
#Write file of results 
write.csv(KO, "../../../output/Slope_Base/Frontloaded_GOSeq/KEGG_Frontloaded.csv")
```

If using non-adjusted p-value, we get 6 significant KEGG terms:


## Alt pipeline? Using all GO terms from all 9011 genes as input (instead of testing enrichment of GO terms from a list of frontloaded GO terms???????)

```{r}
#GO.terms_alt

##Get a list of GO Terms for ALL 9011 genes
GO.terms_alt <- geneInfo %>%
  #filter(gene_id %in% FRONTs$Gene) %>%
  dplyr::select(gene_id, Annotation.GO.ID)

##Format to have one goterm per row with gene ID repeated
split_alt <- strsplit(as.character(GO.terms_alt$Annotation.GO.ID), ";")
split2_alt <- data.frame(v1 = rep.int(GO.terms_alt$gene, sapply(split_alt, length)), v2 = unlist(split_alt))
colnames(split2_alt) <- c("gene", "Annotation.GO.ID")
GO.terms_alt <- split2_alt

goseq_res <- goseq(pwf, gene2cat = GO.terms_alt, method="Wallenius", use_genes_without_cat=TRUE)

#GO.wall <-   goseq(pwf, gene2cat = GO.terms,     method="Wallenius", use_genes_without_cat=TRUE)

head(goseq_res)

#adjusting the p-values here makes all of them p=1 ?????

GO <- goseq_res[order(goseq_res$over_represented_pvalue),]

colnames(GO)[1] <- "GOterm"
    
#adjust p-values 
GO$bh_adjust <-  p.adjust(GO$over_represented_pvalue, method="BH") #add adjusted p-values

#Write file of results 
write.csv(GO, "../../../output/Slope_Base/Frontloaded_GOSeq/GOseq_Frontloaded_ALT.csv")

#Filtering for p < 0.05
GO_05_ALT <- GO %>%
        dplyr::filter(over_represented_pvalue<0.05) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)

#Filtering for p < 0.05
GO_001_ALT <- GO %>%
        dplyr::filter(over_represented_pvalue<0.001) %>%
        dplyr::arrange(., ontology, over_represented_pvalue)
```

Plotting!

```{r}
ontologies<-c("BP", "CC", "MF")

for (category in ontologies) {
    
    go_results <- GO_05_ALT
    
    go_results<-go_results%>%
      filter(ontology==category)%>%
      filter(over_represented_pvalue != "NA") %>%
      filter(numInCat>10)%>%
      arrange(., over_represented_pvalue)
    
      #Reduce/collapse GO term set with the rrvgo package 
      simMatrix <- calculateSimMatrix(go_results$GOterm,
                                orgdb="org.Ce.eg.db", #c. elegans database
                                ont=category,
                                method="Rel")
    #calculate similarity 
    scores <- setNames(-log(go_results$over_represented_pvalue), go_results$GOterm)
    reducedTerms <- reduceSimMatrix(simMatrix,
                                scores,
                                threshold=0.7,
                                orgdb="org.Ce.eg.db")
    
    #keep only the goterms from the reduced list
    go_results<-go_results%>%
      filter(GOterm %in% reducedTerms$go)
    
    #add in parent terms to list of go terms 
    go_results$ParentTerm<-reducedTerms$parentTerm[match(go_results$GOterm, reducedTerms$go)]
   
    #plot significantly enriched GO terms by Slim Category faceted by slim term 
  GO.plot <-  ggplot2::ggplot(go_results, aes(x = ontology, y = term)) + 
    geom_point(aes(size=over_represented_pvalue)) + 
    scale_size(name="Over rep. p-value", trans="reverse", range=c(1,3))+
    facet_grid(ParentTerm ~ ., scales = "free", labeller = label_wrap_gen(width = 5, multi_line = TRUE))+
    theme_bw() + theme(panel.border = element_blank(), panel.grid.major = element_blank(),
    panel.grid.minor = element_blank(), axis.line = element_line(colour = "black"),
    strip.text.y = element_text(angle=0, size = 10),
    strip.text.x = element_text(size = 20),
    axis.text = element_text(size = 8),
    axis.title.x = element_blank(),
    axis.title.y = element_blank())
  
  print(GO.plot)
  
ggsave(filename=paste0("../../../output/Slope_Base/Frontloaded_GOSeq/FRONTs_P05_ALT", "_", category, ".png"), plot=GO.plot, dpi=100, width=12, height=100, units="in", limitsize=FALSE)
}
```

### KEGG enrichment
```{r}
##Get a list of KEGG Terms for all genes
KO.terms <- geneInfo %>%
  #filter(gene_id %in% FRONTs$Gene) %>%
  dplyr::select(gene_id, KEGG)

#run goseq using Wallenius method for all categories of KEGG terms
KO.wall <-
  goseq(
    pwf,
    ID.vector,
    gene2cat = KO.terms,
    test.cats = c("KEGG"),
    method = "Wallenius",
    use_genes_without_cat = TRUE
  )


KO <- KO.wall[order(KO.wall$over_represented_pvalue), ]
colnames(KO)[1] <- "KEGGterm"

#adjust p-values
#KO$bh_adjust <- p.adjust(KO$over_represented_pvalue, method = "BH") #add adjusted p-values

#Filtering for p < 0.05
KO_05 <- KO %>%
  dplyr::filter(over_represented_pvalue < 0.05) %>%
  dplyr::arrange(., over_represented_pvalue)

head(KO_05)
#Write file of results 
write.csv(KO, "../../../output/Slope_Base/Frontloaded_GOSeq/KEGG_Frontloaded_ALT.csv")
```
If using non-adjusted p-value, we get one significant KEGG term:
K17341 (HMCN)
hemicentin,	Age-related macular degeneration
