---
title: "glmmSeq, with pOverA filtering and outliers removed, Slope as Base Level"
author: "Zoe Dellaert"
date: "4/24/2023"
output: html_document
---

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

This is a Rmd file analyzing our raw count data by the glmmSeq package as described in the [vignette](https://cran.r-project.org/web/packages/glmmSeq/vignettes/glmmSeq.html) and [manual](https://cran.r-project.org/web/packages/glmmSeq/glmmSeq.pdf).

```{r}
sessionInfo() #provides list of loaded packages and version of R. I still have version 4.1 for now.
```

First, download the glmmSeq package. I had to install qvalue using BiocManager as described in the [vignette](https://cran.r-project.org/web/packages/glmmSeq/vignettes/glmmSeq.html)

```{r install packages, eval=FALSE}
BiocManager::install("qvalue", force = TRUE)
install.packages("rlang")
install.packages("glmmSeq") #from CRAN
#devtools::install_github("myles-lewis/glmmSeq") # from github
install.packages("plotly")
BiocManager::install("genefilter")
BiocManager::install("DESeq2")
```

Next, load the package 
  
```{r}
library(lme4)
library(glmmSeq)
library(dplyr) #load dplyr, which is required to run this tutorial but not included in the vignette
library(kableExtra) #load kableExtra, which is also required to run this tutorial but not included in the vignette
library(plotly) #library plotly to make interactive plots
sessionInfo()
```


and load in raw count data
```{r}
cts_raw <- read.csv("../../../TagSeq_Output/HeronPdam_gene_count_matrix.csv") #load in data
rownames(cts_raw) <- cts_raw[,1] #set first column that contains gene names as rownames
cts_raw <- cts_raw[,-1] #remove the column with gene names
```

Clean up sample names from "RF13B_S85_ALL.bam.gtf" to "RF13B"
```{r}
colnames(cts_raw)<-gsub("_[^_]+$", "",colnames(cts_raw)) #get rid of "_ALL.bam.gtf"
colnames(cts_raw)<-gsub("_[^_]+$", "",colnames(cts_raw)) #get rid of "_S85"
head(colnames(cts_raw)) #see first 6 clean sample names
```

Metadata from this dataset
```{r}
coldata <- read.csv("../../../../RAnalysis/Data/RNA Submission Sample List metadata.csv") #read in metadata file
coldata <- plyr::rename(coldata, c("Sample.Name"="Coral_ID")) #Make a column that represents the colonies
coldata$Colony <- gsub("A", "", coldata$Coral_ID) #ID which sample is from which colony by remobing letters after colony name
coldata$Colony <- gsub("B", "", coldata$Colony) #ID which sample is from which colony by remobing letters after colony name
coldata$Colony <- gsub("C", "", coldata$Colony) #ID which sample is from which colony by remobing letters after colony name
coldata$Colony <- gsub("D", "", coldata$Colony) #ID which sample is from which colony by remobing letters after colony name

coldata$Origin <- factor(coldata$Origin, levels = c("Slope","Flat")) #set variables to factors, with Slope as the baseline
coldata$Colony <- factor(coldata$Colony) #set variables to factors
coldata$Treatment <- factor(coldata$Treatment) #set variables to factors
head(coldata)
```

Data sanity checks:
```{r}
all(rownames(coldata$Coral_ID) %in% colnames(cts_raw)) #are all of the sample names (rows) in the metadata column names in the gene count matrix?
all(rownames(coldata$Coral_ID) == colnames(cts_raw)) #are they the same in the same order?
```

## Remmoving outlier "RF16A" and "RF16C"

```{r}
cts_raw_outrm <- cts_raw %>% select(-c("RF16A","RF16C")) #remove those columns from count matrix
coldata_outrm <- coldata %>% filter(Coral_ID != c("RF16A","RF16C")) #removed those rows from metadata
```

## Using filtered data instead of raw counts as input

```{r}
cts_filt_outrm <- cts_raw_outrm[rowSums(!as.matrix(cts_raw_outrm)) < ncol(cts_raw_outrm), ] #here we remove all genes that were not expressed in any of our samples- we are left with 24,220 genes.
```


## pOverA filtering

```{r}
library(genefilter)

#use pOverA filtering to reduce dataset

ffun<-filterfun(pOverA(0.25,10))  #set up filtering parameters
cts_filt_outrm_poa <- genefilter((cts_filt_outrm), ffun) #apply filter
sum(cts_filt_outrm_poa) #count number of genes left

cts_filt_outrm_poa <- cts_filt_outrm[cts_filt_outrm_poa,] #keep only rows that passed filter
```

After removing outliers, there are now 9011 genes in the filtered dataset.

Data sanity checks:
```{r}
all(rownames(coldata_outrm$Coral_ID) %in% colnames(cts_filt_outrm_poa)) #are all of the sample names (rows) in the metadata column names in the gene count matrix?
all(rownames(coldata_outrm$Coral_ID) == colnames(cts_filt_outrm_poa)) #are they the same in the same order?
```


### Time to calculate dispersion!

"Using negative binomial models requires gene dispersion estimates to be made. This can be achieved in a number of ways. A common way to calculate this for gene i is to use the equation:

Dispersioni = (variancei - meani)/meani2

Starting with raw counts, so using [DESeq2](http://bioconductor.org/packages/devel/bioc/vignettes/DESeq2/inst/doc/DESeq2.html), but could also import a normalized gene count matrix (i.e. transcripts per million) and calculate disperson manually as in the [vignette](https://cran.r-project.org/web/packages/glmmSeq/vignettes/glmmSeq.html).

```{r}
library(DESeq2) #library in DEseq to use to calculate dispersions

dds_filt_outrm <- DESeqDataSetFromMatrix(countData = cts_filt_outrm_poa,
                              colData = coldata_outrm,
                              design = ~1)

dds_filt_outrm <- DESeq(dds_filt_outrm) #apply DEseq dispersion calculation
dispersions_filt_outrm <- setNames(dispersions(dds_filt_outrm), rownames(cts_filt_outrm_poa)) #save dispersions as vector for each gene

dds_norm_counts <- as.data.frame(counts(dds_filt_outrm, normalized=TRUE)) #save normalized counts from DEseq object for downstream use

rm(dds_filt_outrm) #option to remove large DEseq object, we no longer need it once we have saved the dispersions
```


### Size Factors
There is also an option to include size factors for each gene.

```{r, eval=FALSE}
sizeFactors_filt <- estimateSizeFactorsForMatrix(cts_filt_outrm_poa)
```

## Fitting Models

In this case study we want to use time and response as fixed effects and the patients as random effects:

> gene expression ~ Origin * Treatment + (1 | Colony)

To fit this model for all genes we can use the glmmSeq function.

```{r, eval=FALSE}
results <- glmmSeq(~ Treatment * Origin + (1 | Colony),
                   countdata = cts_filt_outrm_poa,
                   metadata = coldata_outrm,
                   dispersion = dispersions_filt_outrm,
                   progress = TRUE)

#this fitting takes a while, so I save it as an RDS and load this in when needed instead of re-running everytime (unless something upstream of here changes)

saveRDS(results, file = "glmmSeq.rds")  #save RDS
```

Showed errors in one gene, Pocillopora_acuta_HIv2___RNAseq.g5950.t1.

```{r}
results <- readRDS(file = "glmmSeq.rds") #load in RDS from previous step / previous iteration

results@errors   # show detailed gene error
```

Interrogate that gene's counts to try to understand this error better.

```{r}
cts_filt_outrm_poa %>% filter(row.names(cts_filt_outrm_poa) %in% c('Pocillopora_acuta_HIv2___RNAseq.g5950.t1')) %>% unlist
```


## Hypothesis testing: How much does the interaction factor affect the fit of the model

```{r, eval=FALSE}
glmmLRT <- glmmSeq(~ Treatment * Origin + (1 | Colony),
                   reduced = ~ Treatment + Origin + (1 | Colony), #changed * to + to have reduced model without interaction term
                   countdata = cts_filt_outrm_poa,
                   metadata = coldata_outrm,
                   dispersion = dispersions_filt_outrm, verbose = FALSE)

#this fitting takes a while, so I save it as an RDS and load this in when needed instead of re-running everytime (unless something upstream of here changes)

saveRDS(glmmLRT, file = "glmmLRT.rds") #save RDS
```


```{r}
glmmLRT <- readRDS(file = "glmmLRT.rds") #load in RDS from previous step / previous iteration

summary(glmmLRT, "Pocillopora_acuta_HIv2___RNAseq.986_t") #see example statistics for one example gene

LRTstats <- summary(glmmLRT)

head(LRTstats[,13:15]) #view the chi squared values and p values comparing the full and reduced model. P value > 0.05 indicates that the reduced model is just as good as predicting the value of this gene as the full model.

LRTpvals <- unlist(LRTstats[,"P_LRT"])

sum(LRTpvals < 0.05, na.rm=TRUE) #count how many genes had a significant effect of using the full model compared to the reduced model

rm(glmmLRT) #remove this large object, it will not be used again (and you can always reload the saved RDS)
```

So, for only 300 of the 9011 genes, the full model performs better than the reduced. Maybe we should use the reduced? Unclear if we want to do so or stick with the same model as the original paper. Worth also testing a reduced model without treatment if that is even possible.


### Outputs
```{r}
results@modelData #show model data, basically the different conditions tested
```

### Q values

```{r}
results <- glmmQvals(results) # adds a matrix to the results object containing Q-vals
```


Model fit statistics, w/ p-value ordered by Origin:Treatment:
```{r}
stats <- summary(results)  #save the summary table from the model fit results

#using adjusted p-value
kable(stats[order(stats[, 'Treatment:Origin']), ]) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "400px") #View stats table as a scrollable markdown table


write.csv(stats, "../../../output/Slope_Base/signif_genes.csv") #save stats file as csv for downstream analysis
```

Summary stats for a specific gene. (basically a pretty way of looking at one row of the stats table via "stats["Pocillopora_acuta_HIv2___TS.g25814.t1",]")

```{r}
summary(results, gene = "Pocillopora_acuta_HIv2___TS.g25814.t1")
```

Estimated means based on each geneâ€™s fitted model to show fixed effects and their 95% confidence intervals can be seen in the @predict slot:

```{r}
predict = data.frame(results@predict) #estimated means based on fitted model with LCI (lower) and UCI (upper) confidence intervals

kable(predict) %>%
  kable_styling() %>%
  scroll_box(width = "100%", height = "400px")
  
write.csv(predict, "../../../output/Slope_Base/model_expression_prediction_allgenes.csv") #save stats file as csv for downstream analysis
```

## Model Plots

As written the plotting function from this package does not work when x1var, which in our case is Treatment, is a factor.

I changed the function to allow for discrete values on the X axis:

```{r}
source(file = "../Factor_ggmodelPlot.R")

#I changed their code a lot and need to document all the changes

formPlot(results,geneName = "Pocillopora_acuta_HIv2___RNAseq.g14176.t1",x1var = "Treatment",x2var="Origin",x2shift=NULL) #the plotting code is based of off these matrices, made using their function "formPlot"

plotColours <- c("skyblue","mediumseagreen")
modColours <- c("dodgerblue3","seagreen4")
```


### most significant gene by Treatment:Origin:
```{r}
Factor_ggmodelPlot(results,
            geneName = "Pocillopora_acuta_HIv2___RNAseq.g14176.t1",
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3)
```

To make these plots interactive, you can use ggplotly(_plot code_, tooltip="id")

### (second) most significant gene by Treatment:
```{r}
Factor_ggmodelPlot(results,
            geneName = "Pocillopora_acuta_HIv2___RNAseq.g5899.t2",
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3)
```

### most significant gene by Origin:
```{r}
Factor_ggmodelPlot(results,
            geneName = "Pocillopora_acuta_HIv2___RNAseq.g3772.t2",
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3)
```

## MA Plots

Fixed the script for the maplots , just added "scale_y_continuous(limits = c(-4.2,4.2), breaks = seq(-4,4, 2))" to make the scale the same on both plots.
```{r}
source("../fixed_ylim_maPlot.R")
maPlots_filt_outrm <- fixed_ylim_maPlot(results,
                  x1var="Treatment",
                  x2var="Origin",
                  x2Values=c("Flat", "Slope"),
                  graphics="ggplot")

maPlots_filt_outrm$combined
```

## Fold change plots

```{r}
fcPlot(results, x1var = "Treatment", x2var = "Origin", graphics = "plotly",
       pCutoff = 0.05, useAdjusted = TRUE)
```

```{r}
fcPlot(results, x1var = "Origin", x2var = "Treatment", graphics = "plotly",
       pCutoff = 0.05, useAdjusted = TRUE)
```


## But what is fold change, really? 

In this plotting script, fold change is defined as:

x = log2(y_Stable_Slope_predData + 1) - log2(y_Stable_Flat_predData + 1)
y = log2(y_Variable_Slope_predData +1) - log2(y_Variable_Flat_predData +1)

So, for each gene, the predicted mean of expression within a group based on the model fit (pred_Data) is normalized using log 2 and then this is subtracted from another log2-transformed predicted expression for the group to compare with. Lets make a fold change dataframe:

```{r}
predData <- predict[, 1:nrow(results@modelData)] #from the predicted data "results@predict", save the first four columns representing the predicted expression values per group (basically drop the confidence intervals)

colnames(predData)

#"y_Stable_Flat"    "y_Variable_Flat"  "y_Stable_Slope"   "y_Variable_Slope"

#taking the plotData dataframe straight from the fcPlot function code

    # x1var = "Origin"
    # x2var = "Treatment"

    # x1Values <- levels(factor(results@modelData[, "Origin"]))[1:2] # "Flat"  "Slope"
    # x2Values <- levels(factor(results@modelData[, "Treatment"]))[1:2] # "Stable"   "Variable"

    # xCols <- which(modelData[, x2var] == x2Values[1] & modelData[, x1var] %in% x1Values)
    # yCols <- which(modelData[, x2var] == x2Values[2] & modelData[, x1var] %in% x1Values)
 
    # xCols <- which(results@modelData[, "Treatment"] == "Stable"   & results@modelData[, "Origin"] %in% c("Flat","Slope")) # c(1,3) aka "y_Stable_Flat" and "y_Stable_Slope"
    # yCols <- which(results@modelData[, "Treatment"] == "Variable" & results@modelData[, "Origin"] %in% c("Flat","Slope")) # c(2,4) aka "y_Variable_Flat" and "y_Variable_Slope"
  
  
plotData <- data.frame(
    Stable_OriginFC = log2(predData[, "y_Stable_Slope"]+1) - log2(predData[, "y_Stable_Flat"]+1),
    Variable_OriginFC  = log2(predData[, "y_Variable_Slope"]+1) - log2(predData[, "y_Variable_Flat"]+1))
    
plotData$maxGroupFC <- ifelse(abs(plotData$Stable_OriginFC) > abs(plotData$Variable_OriginFC), "Stable", "Variable")

rownames(plotData) <- rownames(predData)
```

So, in this data frame, Stable_OriginFC is the log2 fold change comparing Slope vs. Flat origin within the stable treatment and Variable_OriginFC is the log2fold change comparingSlope vs. Flat origin within the variable treatment

```{r significance classification, broken down by fold change}
#Once again, all of this code is straight from the glmmSeq source code.

pCutoff <- 0.05

  # Set up the colour code
  colLevels <- c('Not Significant', paste0("q_", "Origin", ' < ', pCutoff),
                 paste0("q_", "Origin", ":", "Treatment", " < ", pCutoff,
                        " (biggest FC in ", "Variable", ")"),
                 paste0("q_", "Origin", ":", "Treatment", " < ", pCutoff,
                        " (biggest FC in ", "Stable", ")"))
  
  plotData$col <- colLevels[1] #set all genes as 'Not Significant'
  
  plotData$col[stats[, "Origin"] < pCutoff & ! is.na(stats[, "Origin"])] <- colLevels[2] #set all genes whose q_values (adjusted p-values) for "Origin" are less than 0.05 as "q_Origin < 0.05"

  plotData$col[stats[, "Treatment:Origin"] < pCutoff & !is.na(stats[, "Treatment"])] <- colLevels[3] #set all genes whose q_values (adjusted p-values) for the interaction "Treatment:Origin" are less than  0.05 as "q_Origin:Treatment < 0.05 (biggest FC in Variable)"

  plotData$col[plotData$col == colLevels[3] & plotData$maxGroupFC == "Stable"] <- colLevels[4] #for all of those that we just set as "q_Origin:Treatment < 0.05 (biggest FC in Variable)", change those whose maxGroupFC was Stable to "q_Origin:Treatment < 0.05 (biggest FC in Stable)"

  plotData$col[is.na(plotData$col)] <- 'Not Significant' #if any NAs were introduced, set those genes to Not significant
   
  plotData$col <- factor(plotData$col, levels = colLevels) #turn this new significance column into a factor

print(table(plotData$col))

#reminder that this signifcant list does not include a signficicance test by treatment alone, so any genes listed as "Not Significant" could still be significant but only by Treatment 
```

Now we can take these calculated fold changes and bring them into downstream analyses.


```{r}
stats_wFC <- merge(as.data.frame(stats), plotData, by="row.names", sort=FALSE)

rownames(stats_wFC) <- stats_wFC[,1] #set row names from first column

stats_wFC <- stats_wFC[,-1] #remove first column
```


## Alt Fold Change Plots: division fold change, no log2
```{r}
source(file = "../ALT_FCPlot.R")

head(plotData)

ALT_plotData <- data.frame(
    Stable_OriginFC =    (predData[, "y_Stable_Slope"]  /predData[, "y_Stable_Flat"  ]),
    Variable_OriginFC  = (predData[, "y_Variable_Slope"]/predData[, "y_Variable_Flat"]))

ALT_plotData$maxGroupFC <- ifelse(abs(ALT_plotData$Stable_OriginFC) > abs(ALT_plotData$Variable_OriginFC), "Stable", "Variable")

rownames(ALT_plotData) <- rownames(predData)

ALT_fcPlot(results, x1var = "Origin", x2var = "Treatment", graphics = "plotly",
       pCutoff = 0.05, useAdjusted = TRUE)
```

## Alt Fold Change Plots: division fold change, with log2
```{r}
source(file = "../ALT_LogFCPlot.R")

head(plotData)

ALT_Log_plotData <- data.frame(
    Stable_OriginFC =    log2(predData[, "y_Stable_Slope"]  /predData[, "y_Stable_Flat"  ]),
    Variable_OriginFC  = log2(predData[, "y_Variable_Slope"]/predData[, "y_Variable_Flat"]))

ALT_Log_plotData$maxGroupFC <- ifelse(abs(ALT_plotData$Stable_OriginFC) > abs(ALT_plotData$Variable_OriginFC), "Stable", "Variable")

rownames(ALT_Log_plotData) <- rownames(predData)

#worth noting that this changes a small few (26) of the Max Fold Change (so ones where the absolute value of fold change was largest in Stable now the absolute valye is largest in Variable or vice versa) 
which((ifelse(abs(plotData$Stable_OriginFC) > abs(plotData$Variable_OriginFC), "Stable", "Variable")) != (ifelse(abs(ALT_Log_plotData$Stable_OriginFC) > abs(ALT_Log_plotData$Variable_OriginFC), "Stable", "Variable")))

plotData[which((ifelse(abs(plotData$Stable_OriginFC) > abs(plotData$Variable_OriginFC), "Stable", "Variable")) != (ifelse(abs(ALT_Log_plotData$Stable_OriginFC) > abs(ALT_Log_plotData$Variable_OriginFC), "Stable", "Variable"))),]

ALT_Log_plotData[which((ifelse(abs(plotData$Stable_OriginFC) > abs(plotData$Variable_OriginFC), "Stable", "Variable")) != (ifelse(abs(ALT_Log_plotData$Stable_OriginFC) > abs(ALT_Log_plotData$Variable_OriginFC), "Stable", "Variable"))),]

ALT_LogfcPlot(results, x1var = "Origin", x2var = "Treatment", graphics = "plotly",
       pCutoff = 0.05, useAdjusted = TRUE)
```

## Downstream DEG list

For downstream use, I am saving the "stats_wFC" dataframe (containing significance and FC of each gene and model fit)
```{r}
stats_cts_merge_rawcts <- merge(stats_wFC, cts_filt_outrm_poa, by="row.names", sort=FALSE) ## Merge statistics (P-value info for each gene) with raw count data after pOverA filtering

names(stats_cts_merge_rawcts)[1] <- "Gene" #set first column column name to "Gene"

stats_cts_merge_cts <- merge(stats_wFC, as.data.frame(results@countdata), by="row.names", sort=FALSE) ## Merge statistics (P-value info for each gene) with count data from the glmmseq results object, which is the exact same as the raw count data (after pOverA filtering, cts_filt_outrm_poa) but the 0s have been replaced with 0.125 which will be useful for any future log-transforms

names(stats_cts_merge_cts)[1] <- "Gene" #set first column column name to "Gene"
 
stats_cts_merge_norm <- merge(stats_wFC, dds_norm_counts, by="row.names", sort=FALSE) ## Merge statistics (P-value info for each gene) with normalized count data from the DEseq2 object. This one does still have the 0s as 0s.

names(stats_cts_merge_norm)[1] <- "Gene" #set first column column name to "Gene"
 
head(stats_cts_merge_rawcts)
head(stats_cts_merge_cts)
head(stats_cts_merge_norm)

write.csv(stats_cts_merge_rawcts, "../../../output/Slope_Base/signif_genes_rawcts.csv") #save this as csv for downstream analysis
write.csv(stats_cts_merge_cts, "../../../output/Slope_Base/signif_genes_cts.csv") #save this as csv for downstream analysis
write.csv(stats_cts_merge_norm, "../../../output/Slope_Base/signif_genes_normcts.csv") #save this as csv for downstream analysis
```

```{r DEGs by variable list}
DEGs <- stats_wFC

DEGs$Gene <- rownames(DEGs)

Origin_DEGs <- DEGs %>%  dplyr::filter(Origin < 0.05) 
Treatment_DEGs <- DEGs %>%  dplyr::filter(Treatment < 0.05) 
Interaction_DEGs <- DEGs %>%  dplyr::filter(`Treatment:Origin` < 0.05) 

nrow(Origin_DEGs)

nrow(Treatment_DEGs)

#lets make a list of genes that are significant only by treatment
sum(Treatment_DEGs[,"Gene"] %in% Origin_DEGs[,"Gene"]) #how many of these genes are also significant by Origin? 11

#let's list the 7 genes that are significant by treatment but not by Origin

Treatment_DEGs_notOrigin <- Treatment_DEGs[-which(Treatment_DEGs[,"Gene"] %in% Origin_DEGs[,"Gene"]),"Gene"]
DEGs[Treatment_DEGs_notOrigin,22:28]

sum(Treatment_DEGs_notOrigin %in% Interaction_DEGs[,"Gene"]) #how many of these genes are also significant by Interaction? 3

# 3 of those are also significant by the interaction term! Let's look more into the 4 that are not significant by Origin or Interaction

Treatment_DEGs_ONLY <- Treatment_DEGs_notOrigin[-which(Treatment_DEGs_notOrigin %in% Interaction_DEGs[,"Gene"])]

DEGs[Treatment_DEGs_ONLY,22:28]

#lets make a list of genes that are significant only by interaction
nrow(Interaction_DEGs)

sum(Interaction_DEGs[,"Gene"] %in% Origin_DEGs[,"Gene"]) #how many of these genes are also significant by Origin? 25

#let's list the 5 genes that are significant by interaction but not by Origin

Interaction_DEGs_notOrigin <- Interaction_DEGs[-which(Interaction_DEGs[,"Gene"] %in% Origin_DEGs[,"Gene"]),"Gene"]
DEGs[Interaction_DEGs_notOrigin,22:28]

#Are any of these not significant by treatment?

sum(Interaction_DEGs_notOrigin %in% Treatment_DEGs[,"Gene"]) #how many of these genes are also significant by Treatment? 3

# 3 of those are also significant by the treatment! Let's look more into the 2 that are not significant by Origin or Treatment

Interaction_DEGs_ONLY <- Interaction_DEGs_notOrigin[-which(Interaction_DEGs_notOrigin %in% Treatment_DEGs[,"Gene"])]

DEGs[Interaction_DEGs_ONLY,22:28]
```

#Gene lists to look into for enrichment/dive into annotations of the small gene sets:

- Origin_DEGs (840 genes)
- Treatment_DEGs (18 genes)
- Interaction_DEGs(30 genes)
- Treatment_DEGs_ONLY (4 genes)
- Interaction_DEGs_ONLY (2 genes)

```{r}
write.csv(Origin_DEGs, "../../../output/Slope_Base/DEGs/Origin_DEGs.csv") #save this as csv for downstream analysis
write.csv(Treatment_DEGs, "../../../output/Slope_Base/DEGs/Treatment_DEGs.csv") #save this as csv for downstream analysis
write.csv(Interaction_DEGs, "../../../output/Slope_Base/DEGs/Interaction_DEGs.csv") #save this as csv for downstream analysis
write.csv(DEGs[Treatment_DEGs_ONLY,], "../../../output/Slope_Base/DEGs/Treatment_DEGs_ONLY.csv") #save this as csv for downstream analysis
write.csv(DEGs[Interaction_DEGs_ONLY,], "../../../output/Slope_Base/DEGs/Interaction_DEGs_ONLY.csv") #save this as csv for downstream analysis
```


```{r}
Treatment_DEGs_ONLY
Interaction_DEGs_ONLY
```


## This is a great place to extract these genes for GO and KEGG enrichment analysis.


Special DEG plots :)

```{r}
for (i in Treatment_DEGs_ONLY) {print(Factor_ggmodelPlot(results,
            geneName = i,
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            title = "Gene significant only by Treatment",
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3))}

for (i in Interaction_DEGs_ONLY) {print(Factor_ggmodelPlot(results,
            geneName = i,
            x1var = "Treatment",
            x2var="Origin", addBox = T,
            xlab = "Treatment and Origin",
            title = "Gene significant only by Interaction",
            colours = plotColours,
            lineColours = plotColours, 
            modelColours = modColours,
            modelSize = 3))}
```

