---
title: "glmmSeq GOMWU Analysis"
author: "Zoe Dellaert"
date: "7/21/2024"
output: html_document
---

# GlmmSeq GO Analysis, GO_MWU

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo() #provides list of loaded packages and version of R. I still have version 4.1 for now.
```

```{r install packages}
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")

library(dplyr)
library(ggplot2)
library(rtracklayer)
library(rrvgo)
library(GO.db)
library(tidyr)
library(forcats)
library(scales)
sessionInfo() #list of packages after library-ing these packages
```


```{r}
DEGs <- read.csv(file="../../../output/glmmseq/signif_genes_normcts.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

#NOTE! This is not a file only with differentially expressed genes, this contains all of the genes in our dataset but also contains p-value information and fold change information to help determine which genes are signficant DEGs based on our model in glmmSeq

rownames(DEGs) <- DEGs$Gene

dim(DEGs)

Origin_DEGs <- DEGs %>%  dplyr::filter(Origin < 0.05)

nrow(Origin_DEGs)
```

```{r}
genes <- rownames(DEGs)
```

## **Generate files for Functional Enrichment**  

Based off of [Ariana's script](https://github.com/AHuffmyer/EarlyLifeHistory_Energetics/blob/master/Mcap2020/Scripts/TagSeq/Genome_V3/1_WGCNA_Mcap_V3.Rmd)

Functional annotation file obtained from: http://cyanophora.rutgers.edu/Pocillopora_acuta/Pocillopora_acuta_HIv2.genes.EggNog_results.txt.gz on April 3, 2023.  


```{r}
Pacuta.annot <- read.delim("../../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt") %>% dplyr::rename("query" = X.query)

dim(Pacuta.annot)
head(Pacuta.annot)
```

```{r}
genes2annot = match(genes, Pacuta.annot$query) #match genes in DEGs (all genes after filtering) to genes in annotation file

sum(is.na(genes2annot)) #number of genes without EggNog annotation
missing<-as.data.frame(genes[which(is.na(genes2annot))]) #dataframe of genes without EggNog annotation
head(missing)
```

2812/9011 genes without eggnog annotation

```{r}
names(Pacuta.annot)

geneInfo0 = data.frame(gene_id = genes, #add gene id
  Accession = Pacuta.annot$seed_ortholog[genes2annot], #add accession number
  Bitscore = Pacuta.annot$score[genes2annot], #add bitscore
  eValue = Pacuta.annot$evalue[genes2annot], #add e value
  Description = Pacuta.annot$Description[genes2annot], #add description of gene
  Annotation.GO.ID = Pacuta.annot$GOs[genes2annot], #add GO ID's
  q_Origin = DEGs$Origin, #add Origin adjusted p-value
  q_Treatment = DEGs$Treatment, #add Treatment adjusted p-value
  q_Interaction = DEGs$Treatment.Origin, #add Treatment:Origin adjusted p-value
  Stable_OriginFC = DEGs$Stable_OriginFC, #add fold change for Slope vs Flat in the stable treatment
  Variable_OriginFC = DEGs$Variable_OriginFC, #add fold change for Slope vs Flat in the variable treatment
  maxGroupFC = DEGs$maxGroupFC, #add max group fold change (was FC bigger in stable of variable treatment)
  col = DEGs$col) #add qualitative significance info

dim(geneInfo0)
head(geneInfo0,2)
```

Order geneInfo0 by significance of Origin, q_Origin (adjusted p value)
```{r}
geneInfo <- geneInfo0[order(geneInfo0[, 'q_Origin']), ]
```

Format GO terms to remove dashes and quotes in  GOs column
```{r}
geneInfo$Annotation.GO.ID <- gsub('"', "", geneInfo$Annotation.GO.ID)
geneInfo$Annotation.GO.ID <- gsub("-", NA, geneInfo$Annotation.GO.ID)
```

## Input for gene_to_go.tab, if you haven't run the WGCNA code yet.

```{r}
##Get a list of GO Terms for the all 9011 genes
 # GO.terms <- geneInfo %>%
 #   dplyr::select(gene_id, Annotation.GO.ID) %>% dplyr::rename("GO.terms" = Annotation.GO.ID)
 # 
 # GO.terms_only <- GO.terms %>% na.omit()
 # 
 # write.table(GO.terms_only, "../../GO_MWU/gene_to_go.tab",row.names = FALSE, sep = "\t", quote = FALSE)
```

### DE genes by origin

Make the Go_MWU input csv file

```{r}
### Generate vector with 1 for the 840 significant DEGs by Origin, and 0 for the other genes
DE_Origin <- geneInfo %>%
    dplyr::select(gene_id, Annotation.GO.ID, q_Origin) %>%
    na.omit() %>%
    mutate(DE = ifelse(q_Origin < 0.05, 1, 0)) %>%
    dplyr::select(gene_id, DE) 
  
print(sum(DE_Origin$DE)) #only 341 of these have GO annotations
  
write.csv(DE_Origin, "../../GO_MWU/input_files/DE_Origin.csv", row.names = FALSE, quote = FALSE)
```




### What about the direction of FC: Genes significant by Origin and upregulated in Slope origin (positive FC) for both treatments

Make the Go_MWU input csv file

```{r}
### Generate vector with 1 for the 346 significant DEGs by Origin that are upregulated in the Slope origin in both treatments, and 0 for the other genes
DE_Origin_upSlope <- geneInfo %>%
    dplyr::select(gene_id, Annotation.GO.ID, q_Origin, Stable_OriginFC, Variable_OriginFC) %>%
    na.omit() %>%
    mutate(DE = ifelse(q_Origin < 0.05 & Stable_OriginFC > 0 & Variable_OriginFC > 0, 1, 0)) %>%
    dplyr::select(gene_id, DE) 

print(sum(DE_Origin_upSlope$DE))
  
# Only 136 of those have GO annotations

write.csv(DE_Origin_upSlope, "../../GO_MWU/input_files/DE_Origin_upSlope.csv", row.names = FALSE, quote = FALSE)
```

### What about the direction of FC: Genes significant by Origin and upregulated in Flat origin (negative FC) for both treatments

Make the Go_MWU input csv file

```{r}
### Generate vector with 1 for the 481 significant DEGs by Origin that are upregulated in the Flat origin in both treatments, and 0 for the other genes
DE_Origin_upFlat <- geneInfo %>%
    dplyr::select(gene_id, Annotation.GO.ID, q_Origin, Stable_OriginFC, Variable_OriginFC) %>%
    na.omit() %>%
    mutate(DE = ifelse(q_Origin < 0.05 & Stable_OriginFC < 0 & Variable_OriginFC < 0, 1, 0)) %>%
    dplyr::select(gene_id, DE) 

print(sum(DE_Origin_upFlat$DE))
  
# Only 171 of those have GO annotations

write.csv(DE_Origin_upFlat, "../../GO_MWU/input_files/DE_Origin_upFlat.csv", row.names = FALSE, quote = FALSE)
```

### DE genes by Treatment (18 genes)

Make the Go_MWU input csv file

```{r}
### Generate vector with 1 for the 18 significant DEGs by Treatment, and 0 for the other genes
DE_Treatment <- geneInfo %>%
    dplyr::select(gene_id, Annotation.GO.ID, q_Treatment) %>%
    na.omit() %>%
    mutate(DE = ifelse(q_Treatment < 0.05, 1, 0)) %>%
    dplyr::select(gene_id, DE) 
  
print(sum(DE_Treatment$DE)) #5 of the 18 have GO annotations
  
write.csv(DE_Treatment, "../../GO_MWU/input_files/DE_Treatment.csv", row.names = FALSE, quote = FALSE)
```


### DE genes by interaction of origin and treatment (30 genes)

Make the Go_MWU input csv file

```{r}
### Generate vector with 1 for the 30 significant DEGs by Interaction, and 0 for the other genes
DE_Interaction <- geneInfo %>%
    dplyr::select(gene_id, Annotation.GO.ID, q_Interaction) %>%
    na.omit() %>%
    mutate(DE = ifelse(q_Interaction < 0.05, 1, 0)) %>%
    dplyr::select(gene_id, DE) 
  
print(sum(DE_Interaction$DE)) #11 of the 30 have GO annotations
  
write.csv(DE_Interaction, "../../GO_MWU/input_files/DE_Interaction.csv", row.names = FALSE, quote = FALSE)
```

### Frontloaded genes

```{r}
FRONTs <- read.csv(file="../../../output/glmmseq/frontloaded_genes.csv", sep=',', header=TRUE)  %>% dplyr::select(!c('X'))

### Generate vector with 1 for the 2631 frontloaded genes, and 0 for the other genes
frontloaded <- geneInfo %>%
  dplyr::select(gene_id, Annotation.GO.ID) %>%
  na.omit() %>%
  mutate(front = ifelse(gene_id %in% FRONTs$Gene, 1, 0)) %>%
  dplyr::select(gene_id, front)

print(sum(frontloaded$front)) #1343 of the 2631 have GO annotations

write.csv(frontloaded, "../../GO_MWU/input_files/frontloaded.csv", row.names = FALSE, quote = FALSE)
```

