---
title: "WGCNA GOMWU Analysis"
author: "Zoe Dellaert"
date: "7/16/2024"
output: html_document
---

# WGCNA GO Analysism GO_MWU

```{r setup, include=FALSE}
knitr::opts_chunk$set(echo = TRUE)
```

```{r}
sessionInfo() #provides list of loaded packages and version of R. 
```

First, load the necessary packages.

```{r load_libraries, include = TRUE}
if ("rtracklayer" %in% rownames(installed.packages()) == 'FALSE') BiocManager::install("rtracklayer")

library(dplyr)
library(ggplot2)
library(rtracklayer)
library(rrvgo)
library(GO.db)
library(tidyr)
library(forcats)
library(scales)
sessionInfo() #list of packages after library-ing these packages
```

## Load in data
```{r}
gff<-rtracklayer::import("../../../data/Pocillopora_acuta_HIv2.genes_fixed.gff3")
gff<-as.data.frame(gff) %>% dplyr::select(-Parent)
dim(gff) # 478988     9
names(gff) 

transcripts <- subset(gff, type == "transcript")
transcripts_gr <- makeGRangesFromDataFrame(transcripts, keep.extra.columns=TRUE) #extract length information
transcript_lengths <- width(transcripts_gr) #isolate length of each gene
seqnames<-transcripts_gr$ID #extract list of gene id 
lengths<-cbind(seqnames, transcript_lengths)
lengths<-as.data.frame(lengths) #convert to data frame

dim(transcripts) #33730    13
```

```{r}
kegg <- read.delim("../../../data/Pocillopora_acuta_HIv2.genes.KEGG_results.txt",header = FALSE)
kegg <- as.data.frame(kegg)
colnames(kegg)[1] <- "gene_id" 
colnames(kegg)[2] <- "KEGG_new"
head(kegg)
```


```{r}
eggnog<-read.delim("../../../data/Pocillopora_acuta_HIv2.genes.EggNog_results.txt")#this file contains all of the go terms, descriptions, kegg, etc
eggnog<- plyr::rename(eggnog, c("X.query"="gene_id"))
head(eggnog,2)
```

```{r}
gogene <- merge(transcripts, eggnog, by=c("gene_id"), all=T)
gogene <- merge(gogene, kegg, by=c("gene_id"), all=T)
head(gogene,2)
dim(gogene)
```


```{r}
geneInfo <- read.csv("../../../output/WGCNA/WGCNA_ModuleMembership.csv") #this file was generated from the WGCNA analyses and contains the modules of interest
geneInfo<- plyr::rename(geneInfo, c("X"="gene_id"))
dim(geneInfo) # there are 9012 genes in our gene info file
```

```{r}
geneInfo <- merge(gogene, geneInfo, by=c("gene_id")) #merging the GO and Kegg info to module membership for the 9012 genes
```

Format GO terms to remove dashes and quotes and separate by semicolons (replace , with ;) in  GOs column
```{r}
#geneInfo$GOs <- gsub(",", ";", geneInfo$GOs)
geneInfo$GOs <- gsub('"', "", geneInfo$GOs)
geneInfo$GOs <- gsub("-", NA, geneInfo$GOs)

geneInfo$KEGG_new[geneInfo$KEGG_new == ""] <- NA
```

```{r}
unique(geneInfo$moduleColor)
```

```{r}
geneInfo$Length<-lengths$transcript_lengths[match(geneInfo$gene_id, lengths$seqnames)]
dim(geneInfo)
write.csv(geneInfo, file = "../../../output/WGCNA/topGO/geneInfo_WGCNA.csv") #gene info for reference/supplement
```


## Modules

```{r}
calc_up_mods <- c("brown", "red", "black", "pink", "salmon", "blue")

nrow(geneInfo %>% dplyr::filter(moduleColor=="brown")) #942
nrow(geneInfo %>% dplyr::filter(moduleColor=="red")) #425
nrow(geneInfo %>% filter(moduleColor=="black")) #396
nrow(geneInfo %>% filter(moduleColor=="pink")) #220
nrow(geneInfo %>% filter(moduleColor=="salmon")) #154
nrow(geneInfo %>% filter(moduleColor=="blue")) #1989

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="brown")), nrow(geneInfo %>% dplyr::filter(moduleColor=="red")), nrow(geneInfo %>% filter(moduleColor=="black")), nrow(geneInfo %>% filter(moduleColor=="pink")), nrow(geneInfo %>% filter(moduleColor=="salmon")), nrow(geneInfo %>% filter(moduleColor=="blue")))

# 4126

calc_down_mods <- c("turquoise","magenta","lightcyan")

nrow(geneInfo %>% dplyr::filter(moduleColor=="turquoise")) #2558
nrow(geneInfo %>% dplyr::filter(moduleColor=="magenta")) #219
nrow(geneInfo %>% filter(moduleColor=="lightcyan")) #65

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="turquoise")), nrow(geneInfo %>% dplyr::filter(moduleColor=="magenta")), nrow(geneInfo %>% filter(moduleColor=="lightcyan")))

# 2842

other_mods <- c("green","yellow", "purple", "midnightblue","cyan","tan")

sum(nrow(geneInfo %>% dplyr::filter(moduleColor=="green")), nrow(geneInfo %>% dplyr::filter(moduleColor=="yellow")), nrow(geneInfo %>% filter(moduleColor=="purple")), nrow(geneInfo %>% filter(moduleColor=="midnightblue")), nrow(geneInfo %>% filter(moduleColor=="cyan")),nrow(geneInfo %>% filter(moduleColor=="tan")))

# 2044

# 4126 + 2842 + 2044 = 9012, which represents all of our genes
```



## GO_MWU Setup, how to manipulate the data into the correct file format needed to run GoMWU

```{r}
##Get a list of GO Terms for the all 9012 genes
GO.terms <- geneInfo %>%
  dplyr::select(gene_id, GOs) %>% dplyr::rename("GO.terms" = GOs)

GO.terms_only <- GO.terms %>% na.omit()

write.table(GO.terms_only, "../../GO_MWU/gene_to_go.tab",row.names = FALSE, sep = "\t", quote = FALSE)
```

```{r}
modules_of_interest <- c("brown", "red", "blue", "turquoise", "magenta")

for (mod in modules_of_interest) {
  
  module_name <- paste0(mod, "_module")
  column_name <- paste0("A.", mod)
  
  #this one is just genes that have go term and just zero or one if in the module or not
  module_data <- geneInfo %>%
    dplyr::select(gene_id, GOs, moduleColor, column_name) %>%
    na.omit() %>%
    mutate(column_name = ifelse(moduleColor == mod, 1, 0)) %>%
    dplyr::select(gene_id, column_name) 
  
  print(sum(module_data$column_name))
  
  file_path <- paste0("../../GO_MWU/input_files/", mod, "_module.csv")
  
  write.csv(module_data, file_path, row.names = FALSE, quote = FALSE)
}
```

